<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Comparative Genomics with SynExtend and DECIPHER • CompGenomicsBioc2022</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Comparative Genomics with SynExtend and DECIPHER">
<meta property="og:description" content="In the past decade, the number of sequenced proteins with unknown functions has grown exponentially while the number of 
             experimentally analyzed proteins has increased at a relatively constant rate. To assign functions to more proteins, many 
             in silico methods have been produced that predict protein function purely from gene sequence data. These methods rely on 
             a ‘guilt-by-association’ analysis to detect genes that are likely involved in common functional pathways. We have 
             developed two Bioconductor packages with many functions for comparative genomics. This workshop will focus on one 
             application of comparative genomics: predicting the functional association between proteins to formulate hypotheses about 
             the unknown role of new proteins. Along the way, we will cover several topics:
             - Finding and importing genomes into a sequence database with the DECIPHER package
             - Gene calling and annotation using the DECIPHER package
             - Identification of clusters of orthologous genes using the SynExtend package
             - Construction of alignments and phylogenetic trees using the new TreeLine function in the DECIPHER package
             - Prediction of functional associations using ProtWeaver in the SynExtend package
             The talk will highlight our newest functionality, ProtWeaver, which implements several methods commonly used in the literature 
             to prediction protein functional association. We will briefly show how each algorithm works, then apply them to a real set of 
             proteins of unknown function. This workshop will teach participants how to extract useful information from large biological 
             sequence data with comparative genomics and predict the function of uncharacterized proteins.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">CompGenomicsBioc2022</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/workshop_example.html">Comparative Genomics with SynExtend and DECIPHER</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ahl27/CompGenomicsBioc2022" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="comparative-genomics-analyses-with-synextend-and-decipher">Comparative Genomics Analyses with <code>SynExtend</code> and <code>DECIPHER</code>
<a class="anchor" aria-label="anchor" href="#comparative-genomics-analyses-with-synextend-and-decipher"></a>
</h1></div>
<p>The <code>SynExtend</code> and <code>DECIPHER</code> packages for R incorporate a wealth of easy to use functions for comparative genomics analyses. This workshop will introduce users to these packages by walking them through a complete workflow of identifying co-evolving genes from a dataset of genome sequences.</p>
<p>This workflow comprises several steps, each of which are detailed below. More detail will be added later; this is currently a work in progress.</p>
<div class="section level2">
<h2 id="loading-genome-data-with-decipher">Loading Genome Data with <code>DECIPHER</code>
<a class="anchor" aria-label="anchor" href="#loading-genome-data-with-decipher"></a>
</h2>
<p>The first step in analyzing genomics data is loading the data itself. Here we will download sequencing data from NCBI as a <code>.fasta</code>, load it into R, then visualize and align the sequences.</p>
</div>
<div class="section level2">
<h2 id="gene-calling-and-annotation-with-decipher">Gene Calling and Annotation with <code>DECIPHER</code>
<a class="anchor" aria-label="anchor" href="#gene-calling-and-annotation-with-decipher"></a>
</h2>
<p>Once we have the data, we will identify genetic regions with <code>FindGenes()</code> from <code>DECIPHER</code>, then annotate them with <code>IdTaxa()</code>.</p>
</div>
<div class="section level2">
<h2 id="annotation-of-cogs-with-synextend">Annotation of COGs with <code>SynExtend</code>
<a class="anchor" aria-label="anchor" href="#annotation-of-cogs-with-synextend"></a>
</h2>
<p>Now that we have annotated gene calls, we can combine them into clusters of orthologous genes (COGs) with <code>DisjointSets()</code> from <code>SynExtend</code>.</p>
</div>
<div class="section level2">
<h2 id="constructing-gene-trees-with-decipher">Constructing Gene Trees with <code>DECIPHER</code>
<a class="anchor" aria-label="anchor" href="#constructing-gene-trees-with-decipher"></a>
</h2>
<p>We will then build phylogenetic trees from each COG using the new <code>TreeLine()</code> function from <code>DECIPHER</code>.</p>
</div>
<div class="section level2">
<h2 id="identifying-co-evolving-gene-collectives-with-synextend">Identifying Co-evolving Gene Collectives with <code>SynExtend</code>
<a class="anchor" aria-label="anchor" href="#identifying-co-evolving-gene-collectives-with-synextend"></a>
</h2>
<p>With COGs and gene trees in hand, we can identify gene clusters under shared evolutionary pressure with the <code>ProtWeaver</code> class in <code>SynExtend</code>. Co-evolutionary signal implies functional association, so the resulting pairwise associations are useful for finding functionally associated genes/proteins.</p>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>At the conclusion of this workshop, users will be able to perform the following tasks in R: * Visualize sequence data * Identify and annotate genes from sequence data * Identify COGs from a set of gene calls * Build phylogenies at the species and gene level * Predict COGs under shared evolutionary pressure</p>
</div>
<div class="section level2">
<h2 id="useful-links">Useful Links<a class="anchor" aria-label="anchor" href="#useful-links"></a>
</h2>
<ul>
<li><a href="http://bioconductor.org/packages/release/bioc/html/DECIPHER.html" class="external-link">DECIPHER</a></li>
<li><a href="http://bioconductor.org/packages/release/bioc/html/SynExtend.html" class="external-link">SynExtend</a></li>
<li><a href="http://www2.decipher.codes/Tutorials.html" class="external-link">Related Tutorials</a></li>
<li><a href="https://www.wrightlabscience.com/p/index.html" class="external-link">Our Lab!</a></li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/ahl27/CompGenomicsBioc2022/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/ahl27/CompGenomicsBioc2022/issues/new/choose" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing CompGenomicsBioc2022</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Aidan Lakshman <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-9465-6785" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
