<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="In the past decade, the number of sequenced proteins with unknown functions has grown exponentially while the number of 
             experimentally analyzed proteins has increased at a relatively constant rate. To assign functions to more proteins, many 
             in silico methods have been produced that predict protein function purely from gene sequence data. These methods rely on 
             a ‘guilt-by-association’ analysis to detect genes that are likely involved in common functional pathways. We have 
             developed two Bioconductor packages with many functions for comparative genomics. This workshop will focus on one 
             application of comparative genomics: predicting the functional association between proteins to formulate hypotheses about 
             the unknown role of new proteins. Along the way, we will cover several topics:
             - Finding and importing genomes into a sequence database with the DECIPHER package
             - Gene calling and annotation using the DECIPHER package
             - Identification of clusters of orthologous genes using the SynExtend package
             - Construction of alignments and phylogenetic trees using the new TreeLine function in the DECIPHER package
             - Prediction of functional associations using ProtWeaver in the SynExtend package
             The talk will highlight our newest functionality, ProtWeaver, which implements several methods commonly used in the literature 
             to prediction protein functional association. We will briefly show how each algorithm works, then apply them to a real set of 
             proteins of unknown function. This workshop will teach participants how to extract useful information from large biological 
             sequence data with comparative genomics and predict the function of uncharacterized proteins.">
<title>Comparative Genomics with SynExtend and DECIPHER • CompGenomicsBioc2022</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Comparative Genomics with SynExtend and DECIPHER">
<meta property="og:description" content="In the past decade, the number of sequenced proteins with unknown functions has grown exponentially while the number of 
             experimentally analyzed proteins has increased at a relatively constant rate. To assign functions to more proteins, many 
             in silico methods have been produced that predict protein function purely from gene sequence data. These methods rely on 
             a ‘guilt-by-association’ analysis to detect genes that are likely involved in common functional pathways. We have 
             developed two Bioconductor packages with many functions for comparative genomics. This workshop will focus on one 
             application of comparative genomics: predicting the functional association between proteins to formulate hypotheses about 
             the unknown role of new proteins. Along the way, we will cover several topics:
             - Finding and importing genomes into a sequence database with the DECIPHER package
             - Gene calling and annotation using the DECIPHER package
             - Identification of clusters of orthologous genes using the SynExtend package
             - Construction of alignments and phylogenetic trees using the new TreeLine function in the DECIPHER package
             - Prediction of functional associations using ProtWeaver in the SynExtend package
             The talk will highlight our newest functionality, ProtWeaver, which implements several methods commonly used in the literature 
             to prediction protein functional association. We will briefly show how each algorithm works, then apply them to a real set of 
             proteins of unknown function. This workshop will teach participants how to extract useful information from large biological 
             sequence data with comparative genomics and predict the function of uncharacterized proteins.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Coevolution%20Networks.html">Coevolutionary Networks</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com" aria-label="My Website">
    <span class="fa fa-potato"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="welcome">Welcome!<a class="anchor" aria-label="anchor" href="#welcome"></a>
</h1></div>
<p>The <code>SynExtend</code> and <code>DECIPHER</code> packages for R incorporate a wealth of easy to use functions for comparative genomics analyses. This interactive tutorial series will introduce users to these packages by walking them through a complete workflow of identifying co-evolving genes from a dataset of genome sequences. This webpage was created for presentation at <a href="https://bioc2022.bioconductor.org/" class="external-link">Bioconductor 2022</a>, but the content will be freely available forever.</p>
<p>I’ve summarized on this page all the skills you can expect to learn by working through the tutorials on this site. See the <a href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a> page for more information, along with code examples you can run for yourself!</p>
<p>(Note: I’m still working on the tutorials, but they will be posted by mid-June! Apologies for the delay.)</p>
<div class="section level3">
<h3 id="loading-genome-data-with-decipher">
<a href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Loading Genome Data with <code>DECIPHER</code></a><a class="anchor" aria-label="anchor" href="#loading-genome-data-with-decipher"></a>
</h3>
<p>The first step in analyzing genomics data is loading the data itself. Here we will download sequencing data from NCBI as a <code>.fasta</code>, load it into R, then perform some basic operations with the data. Users will learn to efficiently work with large scale genomics data, including visualization and alignment of sequencing data.</p>
<p><a href="https://www.ahl27.com/CompGenomicsBioc2022/reference/index.html#loading-genome-data"><sup>Function Reference</sup></a></p>
</div>
<div class="section level3">
<h3 id="gene-calling-and-annotation-with-decipher">
<a href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation with <code>DECIPHER</code></a><a class="anchor" aria-label="anchor" href="#gene-calling-and-annotation-with-decipher"></a>
</h3>
<p>A natural next step is identifying what elements comprise each genome in our dataset. Users will learn to programmatically identify coding and non-coding regions of genomes, and annotate them with predicted KEGG orthology groups.</p>
<p><a href="https://www.ahl27.com/CompGenomicsBioc2022/reference/index.html#gene-calling-and-annotation"><sup>Function Reference</sup></a></p>
</div>
<div class="section level3">
<h3 id="annotation-of-cogs-with-synextend">
<a href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Annotation of COGs with <code>SynExtend</code></a><a class="anchor" aria-label="anchor" href="#annotation-of-cogs-with-synextend"></a>
</h3>
<p>Annotated genetic regions can be mapped across organisms into clusters of orthologous genes (COGs). Users will learn how to identify COGs at scale using the data generated in the previous step.</p>
<p><a href="https://www.ahl27.com/CompGenomicsBioc2022/reference/index.html#constructing-cogs"><sup>Function Reference</sup></a></p>
</div>
<div class="section level3">
<h3 id="constructing-gene-trees-with-decipher">
<a href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Constructing Gene Trees with <code>DECIPHER</code></a><a class="anchor" aria-label="anchor" href="#constructing-gene-trees-with-decipher"></a>
</h3>
<p>Each COG comprises sets of conserved orthologs across species. These data, combined with sequencing data for each ortholog, allow us to reconstruct the evolutionary history of each COG. Users will learn how to construct, visualize, and save phylogenetic trees from sets of genomes.</p>
<p><a href="https://www.ahl27.com/CompGenomicsBioc2022/reference/index.html#constructing-phylogenies"><sup>Function Reference</sup></a></p>
</div>
<div class="section level3">
<h3 id="identifying-co-evolving-gene-collectives-with-synextend">
<a href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">Identifying Co-evolving Gene Collectives with <code>SynExtend</code></a><a class="anchor" aria-label="anchor" href="#identifying-co-evolving-gene-collectives-with-synextend"></a>
</h3>
<p>With these data, we can analyze patterns in evolutionary signal across COGs. Co-evolutionary signal between genes implies functional association, so finding COGs under shared selective pressure aids us in uncovering the mechanisms of intracellular pathways. Users will learn to tease out subtle evidence of correlated evolutionary pressure in order to create co-evolutionary networks.</p>
<p><a href="https://www.ahl27.com/CompGenomicsBioc2022/reference/index.html#finding-co-evolving-gene-collectives"><sup>Function Reference</sup></a></p>
</div>
<div class="section level3">
<h3 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h3>
<p>At the conclusion of this workshop, users will be able to perform the following tasks in R:</p>
<ul>
<li>Visualize sequence data</li>
<li>Identify and annotate genes from sequence data</li>
<li>Identify COGs from a set of gene calls</li>
<li>Build phylogenies at the species and gene level</li>
<li>Predict COGs under shared evolutionary pressure</li>
</ul>
</div>
<div class="section level3">
<h3 id="useful-links">Useful Links<a class="anchor" aria-label="anchor" href="#useful-links"></a>
</h3>
<ul>
<li><a href="http://bioconductor.org/packages/release/bioc/html/DECIPHER.html" class="external-link">DECIPHER</a></li>
<li><a href="http://bioconductor.org/packages/release/bioc/html/SynExtend.html" class="external-link">SynExtend</a></li>
<li><a href="http://www2.decipher.codes/Tutorials.html" class="external-link">Related Tutorials</a></li>
<li><a href="https://www.wrightlabscience.com/p/index.html" class="external-link">Our Lab!</a></li>
</ul>
<p> </p>
<p> </p>
<p> </p>
<p> </p>
<p><a href="http://creativecommons.org/licenses/by-sa/4.0/" class="external-link"><img src="https://img.shields.io/badge/License-CC%20BY--SA%204.0-lightgrey.svg" alt="CC BY-SA 4.0"></a></p>
<p>This work is licensed under a <a href="http://creativecommons.org/licenses/by-sa/4.0/" class="external-link">Creative Commons Attribution-ShareAlike 4.0 International License</a>.</p>
<p><a href="http://creativecommons.org/licenses/by-sa/4.0/" class="external-link"><img src="https://licensebuttons.net/l/by-sa/4.0/88x31.png" alt="CC BY-SA 4.0"></a></p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/ahl27/CompGenomicsBioc2022/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/ahl27/CompGenomicsBioc2022/issues/new/choose" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing CompGenomicsBioc2022</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Aidan Lakshman <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-9465-6785" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
