---
title: "Conclusion"
author: Aidan Lakshman^[ahl27@pitt.edu]
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Conclusion}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
    wrap: 72
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This tutorial has walked through a complete pipeline of comparative genomic
analysis. By using this tool, we've been able to generate hypotheses of 
functional associations between genetic regions using only raw sequence data
as input. Along the way, we've covered the following:

- Working with sequencing data, including alignment and visualization
- Finding and annotating genes from sequencing data
- Finding Clusters of Orthologous Genes (COGs)
- Reconstructing the evolutionary history of COGs
- Analyzing coevolutionary signal between COGs

This coevolutionary signal allows us to generate hypotheses about possible unknown
functional associations between proteins. In the previous example, we were able
to identify that all ureases are under shared selective pressure.

This pipeline works on both genes that have been previously characterized and those that have not, meaning we can guide future wet lab investigation with these predictions. Future work will investigate ways to improve our predictions, so stay tuned to the `SynExtend` package as it develops!

## Thank you!

If you've made it through this entire tutorial, thank you for following along!
I hope this series was informative and useful to your analyses. All code showcased
here is actively being worked on by members of our lab, especially the `ProtWeaver`
and `ProtWeb` functionalities. If you have any comments, suggestions, or feature requests
for `ProtWeaver`, `ProtWeb`, or this tutorial, please feel free to either email me at 
ahl27@pitt.edu or open an issue on GitHub.

## Other Resources

If you're interested in learning more about me, our lab, or phylogenetics, check
out these resources:

- [My website](https://ahl27.com)
- [The Wright Lab](https://wrightlabscience.com)
- [Intro to Phylogenetics](https://www.ahl27.com/IntroPhylogenetics/)
- [More DECIPHER Tutorials](http://www2.decipher.codes/)


```{css, echo=FALSE}
.pagination {
    display: flex;
    display: -webkit-flex;
    padding-left: 0;
    list-style: none;
    justify-content: center
}
```

<div class="center">
  <ul class="pagination pagination-lg">
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">&laquo;</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">1</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">2</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">3</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">4</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">5</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">6</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html">7</a>
    </li>
    <li class="page-item disabled">
      <a class="page-link" href="#">&raquo;</a>
    </li>
  </ul>
</div>