---
title: "Constructing Phylogenies with DECIPHER"
author: Aidan Lakshman^[ahl27@pitt.edu]
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{ConstructingPhylogenies}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
    wrap: 72
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, out.width='1000px', dpi=200)
library(DECIPHER)
library(SynExtend)
library(ape)

newTreeLine <- system.file('extdata', 'TreeLine.R', 
                            package='CompGenomicsBioc2022')
source(newTreeLine, local=knitr::knit_global())
```

## Phylogenetic Reconstruction

We've now learned how to find Clusters of Orthologous Genes (COGs) from a set of 
sequences. The last thing we need for our final analysis are phylogenetic 
reconstructions of each gene cluster. In this step, we'll build phylogenies for
our COGs using the new `TreeLine()` function introduced in the latest update 
of `DECIPHER`.

In the last step, we constructed COGs for a small subset of our data to learn the
algorithmic flow of that portion of the pipeline. Since analyzing the entire
dataset is time-prohibitive for this workshop, we'll work with a pre-calculated
`RData` file containing the results we would have gotten if we had run the same
calculation on the entire dataset.

*Small note:* the setup for this page includes a helper function to plot the
phylogenies as a circular plot. You can plot them just as well with `plot`, I
just prefer this visualization. You can also use `plottype='unrooted'` for a 
different visualization, though labels are often hard to read in that visualization.

```{r eval=FALSE}
library(DECIPHER)
source("TreeLine.R") # Shouldn't be necessary if using dev version
```

```{r}
datafile <- system.file('extdata', 'phyloInputData.RData', 
                        package='CompGenomicsBioc2022')
load(datafile, verbose=TRUE) # Should load 'AlignedSequences'

# Helper plotting function
plot_tree_unrooted <- function(dend, main, plottype='fan'){
  if(require(ape)){
    tf <- tempfile()
    WriteDendrogram(dend, file=tf, quoteLabels=FALSE)
    predTree <- read.tree(tf)
    plot(predTree, plottype, main=main)
  } else {
    plot(dend, main=main)
  }
}

testSeqs <- AlignedSequences

treeCOG1 <- TreeLine(testSeqs, method='MP', reconstruct=TRUE, verbose=FALSE)
plot_tree_unrooted(treeCOG1, main='MP')
```

That's all we need to construct a quick phylogenenetic tree in R! I've set
`reconstruct=TRUE` to have `TreeLine` automatically reconstruct ancestral states 
at each node.

However, `TreeLine` incorporates a wealth of functionality past what is detailed here. 
In fact, this tree isn't even the best tree we can make! Let's take a look at all 
the new features included in the `TreeLine()` function.

## Tree-Building Methods

Our first example used `method='MP'`, meaning it constructed a phyletic tree
using a Maximum Parsimony method. However, many more methods are available. 
`TreeLine()` implements Maximum Parsimony (`MP`), Neighbor-Joining (`NJ`), 
Ultrametric (`complete`, `single`, `UPGMA`, `WPGMA`), and Maximum Likelihood (`ML`)
methods. Each of these have different strengths, weaknesses, and assumptions. While
an in-depth look at the difference between these methods is outside the scope of
this tutorial, I recently published [another tutorial](https://ahl27.com/IntroPhylogenetics)
that goes into the mathematics of how these methods work.

Example code for each of these:
```{r results='hide', fig.keep='all'}
# Maximum Parsimony
treeMP <- TreeLine(myXStringSet=testSeqs, method='MP')
plot_tree_unrooted(treeMP, main='MP')
```
```{r results='hide', fig.keep='all'}
# Neighbor-Joining
distMatrix <- DistanceMatrix(testSeqs, correction='Jukes-Cantor')
treeNJ <- TreeLine(myDistMatrix=distMatrix, method='NJ')
plot_tree_unrooted(treeNJ, main='NJ')
```
```{r results='hide', fig.keep='all'}
# UPGMA tree
distMatrix <- DistanceMatrix(testSeqs, correction='Jukes-Cantor')
treeUltra <- TreeLine(myDistMatrix=distMatrix, method='UPGMA')
plot(treeUltra, main='UPGMA')
```

Maximum-Likelihood trees are the most accurate, but also the slowest to create. 
This method iteratively maximizes the likelihood of the tree under a given sequence
evolution model for a set of aligned sequences. In the interest of time, this 
demo will set the `maxTime` argument to prevent the algorithm from taking too long.

```{r results='hide', fig.keep='all'}
# Maximum Likehood
#   - Max runtime is set here to 30sec, default is as long as it takes
#   - maxTime expresses time in HOURS, not sec/min
#   - Note that method='ML' is the default setting
treeML <- TreeLine(testSeqs, maxTime=(30/3600))
plot_tree_unrooted(treeML, main='Maximum-Likelihood')
```

## Sequence Evolution Models

One question you're probably asking is, "What is this 'given sequence evolution
model'?" That's an excellent question--by default, `TreeLine()` will test a variety
of sequence evolution models and pick the one that works best for your data. 
This means that you shouldn't typically have to worry about which model to use.

However, what if we wanted to explicitly pick a certain model? What if we wanted
to exclude a handful of models? Or what if we're just curious what models are even
being tested?

Fret not, for there is a solution. Models are passed to `TreeLine()` as a list
with two named entries, `$Nucleotide` and `$Protein`. To look at the default 
models tested, simply print out the `MODELS` variable included from `DECIPHER`:

```{r}
DECIPHER::MODELS
```

Nucleotide models include classic names like Jukes-Cantor (`JC69`) and Felsenstein
1981 (`F81`), as well as many others. The amino acid substitution models contain
a mixture of general models (e.g. `BLOSUM62`, `Dayhoff`), as well as models tailored
towards specific organisms (e.g. `Q.insect`, `HIVb`).

To use a single model, simply create a list matching the structure of `MODELS`, 
just with the models you want to include. To exclude certain models, copy `MODELS` 
and remove the entries you don't want. 

```{r results='hide',fig.keep='all'}
# Using a specific set of models
mySpecificModel <- list(Nucleotide=c('JC69', 'HKY85'))
specTree <- TreeLine(testSeqs, model=mySpecificModel, maxTime=(30/3600))
plot_tree_unrooted(specTree, main='Specific Set ML')

# Excluding Specific Models
myExcludedModel <- DECIPHER::MODELS
myExcludedModel$Protein <- NULL # Remove all protein models
myExcludedModel$Nucleotide <- myExcludedModel$Nucleotide[-1] # Remove JC69
exclTree <- TreeLine(testSeqs, model=myExcludedModel, maxTime=(30/3600))
plot_tree_unrooted(exclTree, main='Excluded Set ML')
```

## Conclusion

That's all we need to know how to do to generate phylogenies for our dataset. 
In order to conduct our final coevolutionary analysis, we're going to need 
to build a gene tree for all of our COGs. I've precomputed these trees for us,
so we can load those in in the next step without having to worry about long runtimes.

There are a few other parameters I didn't mention in this writeup. The most significant
one for our use is `reconstruct=TRUE`, as this reconstructs ancestral states (which
will be important for later analyses). There are also additional arguments for 
multiprocessing (`processors=1`), using Laguerre quadrature for likelihoods 
(`quadrature=T/F`), switching the type of information criterion for ML trees
(`informationCriterion=c('AICc', 'BIC')`), and many others. See the documentation
page for more information on these--for now, we'll continue on to our final goal.

```{css, echo=FALSE}
.pagination {
    display: flex;
    display: -webkit-flex;
    padding-left: 0;
    list-style: none;
    justify-content: center
}
```

<div class="center">
  <ul class="pagination pagination-lg">
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">&laquo;</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">1</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">2</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">3</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">4</a>
    </li>
    <li class="page-item active">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">5</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">6</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html">7</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">&raquo;</a>
    </li>
  </ul>
</div>