---
title: "Finding COGs with SynExtend"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{FindingCOGs}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
    wrap: 72
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(DECIPHER)
library(SynExtend)
```

We've now learned some ways to load genomic data into R, as well as ways to find
and annotate genomic sequences. Once we have annotated sequence data, we'll want 
to find genes that are orthologous. Orthologous genes are genes that derive
from some common ancestral gene in the past. This is how we can "match" up genes 
from different organisms. It isn't guaranteed that these genes have preserved
function since diverging from their ancestral state, but it does give us insight
into the evolution of genes over time. Sets of orthologous genes will 
referred to as COGs (Clusters of Orthologous Genes).

## Building Our Dataset

We're going to continue using our *Micrococcus* genomes from NCBI, this time on a 
subset of 5 genomes. As mentioned in previous sections, the complete data are available [here](https://www.ncbi.nlm.nih.gov/data-hub/genome/?taxon=1269&annotated_only=true&refseq_annotation=true&assembly_level=3%3A3),
and you are more than welcome to try these analyses out with more genomes at any time!

All the code in this section will work on larger datasets, you may just have to wait
a little while. See the [Conclusions](https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html)
page for more information on running these analyses at scale.

For this analysis, we're downloading the genomic data as `.fasta` files, but also
precalculated annotations as `.gff` files. We could have also called genes and 
annotated them in `DECIPHER` using the method in the previous page, this just
provides an example of using prebuilt annotations for a more thorough overview
of different usecases.


```{css echo=FALSE}
.hvr-grow-shadow {
  display: inline-block;
  vertical-align: middle;
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  box-shadow: 0 0 1px rgba(0, 0, 0, 0);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: box-shadow, transform;
  transition-property: box-shadow, transform;
}
.hvr-grow-shadow:hover, .hvr-grow-shadow:focus, .hvr-grow-shadow:active {
  box-shadow: 0 10px 10px -10px rgba(0, 0, 0, 0.5);
  -webkit-transform: scale(1.1);
  transform: scale(1.1);
}
```
```{r echo=FALSE}
dir <- system.file('extdata', 'COGs',
                             'micrococcus_wgff',
                             package='CompGenomicsBioc2022')
downloadthis::download_dir(
  path = dir,
  output_name = "micrococcus_wgff",
  button_label = "Download Genomes and Annotations",
  button_type = "success",
  has_icon = TRUE,
  icon = "fa fa-download",
  self_contained = FALSE,
  class = "hvr-grow-shadow"
)
```



```{r eval=FALSE}
library(DECIPHER)
library(SynExtend)
```

```{r results='hide'}
COGExampleDir <- system.file('extdata', 'COGs',
                             'micrococcus_wgff',
                             package='CompGenomicsBioc2022')

# TODO: Better explanation here
# We're going to be using the database API for DECIPHER since this is designed for
# larger scale analyses. See code below for how we do this!
DBPATH <- tempfile()

genomedirs <- list.files(COGExampleDir, full.names = TRUE)
genomedirs <- genomedirs[grep('json', genomedirs, fixed=T, invert=T)]

# Initializing our GeneCalls list
GeneCalls <- vector('list', length=length(genomedirs))

for (i in seq_along(genomedirs)){
  subfiles <- list.files(genomedirs[i], full.names = TRUE)
  
  # Find the FASTA file and the GFF annotations file
  fna_file <- subfiles[which(grepl('.*fna$', subfiles))]
  gff_file <- subfiles[which(grepl('.*gff$', subfiles))]
  
  # Read in sequence to database
  Seqs2DB(seqs = fna_file,
          type = "FASTA",
          dbFile = DBPATH,
          identifier = as.character(i), # Sequences must be identified by number
          verbose = TRUE)
  
  # Read in annotations
  GeneCalls[[i]] <- gffToDataFrame(GFF = gff_file,
                                    Verbose = TRUE)
}
names(GeneCalls) <- seq_along(GeneCalls) # Must have number IDs here too
```

## Finding Orthologous Pairs

Now we have all of our data read in successfully. Next, we'll have to find
pairs of orthologous genes. This is accomplished by means of the `NucleotideOverlap()`
and `PairSummaries()` functions from `SynExtend`. `NucleotideOverlap()` uses
a `Synteny` object and determines where genomic features are connected by syntenic hits.
`PairSummaries` determines pairs of genes that are orthologous by parsing these
connected regions.

*Note:* Several methods here are commented out. This is to save time within the 
workshop, since we have a lot to cover in a relatively short time. Running the
output of `PairSummaries()` through `BlockExpansion()` and `BlockReconciliation()`
improves accuracy of our final identified orthologous regions at the cost of runtime. 
I encourage readers to try out this functionality on their own in the absence
of tight time constraints.

```{r results='hide'}
Syn <- FindSynteny(dbFile = DBPATH,
                   verbose = TRUE)

Overlaps <- NucleotideOverlap(SyntenyObject = Syn,
                               GeneCalls = GeneCalls,
                               LimitIndex = FALSE,
                               AcceptContigNames = TRUE,
                               Verbose = TRUE)
```
```{r}
head(Overlaps)
```
```{r results='hide'}
Pairs <- PairSummaries(SyntenyLinks = Overlaps,
                     GeneCalls = GeneCalls,
                     DBPATH = DBPATH,
                     PIDs = FALSE, # Set to TRUE for better accuracy (slower) 
                     Score = FALSE, # Set to TRUE for better accuracy (slower)
                     Verbose = TRUE)

# These methods only work if we set PIDs and Score to TRUE
# Unfortunately we don't have time in this workshop to use these
# Feel free to try them out on your own with a larger dataset!

# P02 <- BlockExpansion(Pairs = P01,
#                       DBPATH = DBPATH,
#                       Verbose = TRUE,
#                       NewPairsOnly = FALSE)
# P03 <- BlockReconciliation(Pairs = P02,
#                            PIDThreshold = 0.75,
#                            SCOREThreshold = 200,
#                            Verbose = TRUE)
# Pairs <- P03[P03$PID > 0.4, ]
```
```{r}
head(Pairs)
```

## Finding COGs

From these pairwise orthologous regions, we can finally determine COGs using the
`DisjointSet()` function from `SynExtend`. This function analyzes pairs to find
determine which orthologs are (dis)connected. Future work will look into smarter
ways to determine COGs from pairwise orthologies, but this analysis is already highly
accurate. We also use `ExtractBy` to pull out the sequences corresponding to each
genomic region in each COG.
```{r results='hide'}
COGSets <- DisjointSet(Pairs = Pairs,
                        Verbose = TRUE)

# Extract sequences for COGs with orthologs in at least 5 (all) genomes
Sequences <- ExtractBy(x = Pairs,
                       y = DBPATH,
                       z = COGSets[lengths(COGSets) >= 5],
                       Verbose = TRUE)

# These come back in different orders, so let's match them up
allnames <- lapply(Sequences, names)
COGMapping <- sapply(COGSets, \(x){
  which(sapply(allnames, \(y) setequal(x,y)))
})
COGMapping <- COGMapping[sapply(COGMapping, \(x) length(x) > 0)]

MatchedCOGSets <- COGSets[names(COGMapping)]
MatchedSequences <- Sequences[unlist(COGMapping)]
names(MatchedSequences) <- names(COGMapping)
```
```{r}
MatchedCOGSets[1:3]
MatchedSequences[1:3]
```

## Conclusion

Now we know how to generate COGs from a dataset of genomes and gene calls. We 
could have also generated gene calls ourselves, but if high quality gene annotations
are already available (e.g. on NCBI), it makes sense to use them. Remember that this
example is intentionally small so it can fit into our workshop within the time 
constraints--I highly encourage experimenting with other, larger datasets!

```{css, echo=FALSE}
.pagination {
    display: flex;
    display: -webkit-flex;
    padding-left: 0;
    list-style: none;
    justify-content: center
}
```

<div class="center">
  <ul class="pagination pagination-lg">
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">&laquo;</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">1</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">2</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">3</a>
    </li>
    <li class="page-item active">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">4</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">5</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">6</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html">7</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">&raquo;</a>
    </li>
  </ul>
</div>