---
title: "Gene Calling and Annotation with DECIPHER"
author: Aidan Lakshman^[ahl27@pitt.edu]
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{GeneCallingAnnotation}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
    wrap: 72
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(DECIPHER)
newTreeLine <- system.file('extdata', 'TreeLine.R', 
                            package='CompGenomicsBioc2022')
source(newTreeLine, local=knitr::knit_global())
```

```{r echo=FALSE, out.width='100%'}
knitr::include_graphics('images/PipelineGeneCalling.pdf')
```

## Gene Calling and Annotation

At this point, we've learned how to read in some genomic data, and have
gained some basic familiarity working with it. The next step in our
pipeline is to take a set of genomes, identify the coding regions in
them, and predict the function of as many genetic regions as we can.
We'll start off by identifying the genes themselves.

## Finding Genes

We're going to keep using our *Micrococcus* dataset, but in the interest of time we'll focus on finding genes in a single sequence. We'll begin by reading in the data from a `.fasta` file, as we did in the previous section.


```{css echo=FALSE}
.hvr-grow-shadow {
  display: inline-block;
  vertical-align: middle;
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  box-shadow: 0 0 1px rgba(0, 0, 0, 0);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: box-shadow, transform;
  transition-property: box-shadow, transform;
}
.hvr-grow-shadow:hover, .hvr-grow-shadow:focus, .hvr-grow-shadow:active {
  box-shadow: 0 10px 10px -10px rgba(0, 0, 0, 0.5);
  -webkit-transform: scale(1.1);
  transform: scale(1.1);
}
```
```{r echo=FALSE}
f <- system.file('extdata', 'GeneCalling', 'SingleSeq.fa.gz', 
                        package='CompGenomicsBioc2022')
downloadthis::download_file(
  path = f,
  button_label = "Download Example Sequence",
  button_type = "success",
  has_icon = TRUE,
  icon = "fa fa-download",
  self_contained = FALSE,
  class = "hvr-grow-shadow"
)
```

&nbsp;

```{r eval=FALSE}
library(DECIPHER)

# This file is downloadable at the above link
datafile <- '/path/to/SingleSeq.fa.gz'
dnaGenome <- readDNAStringSet(datafile)
aaGenome <- translate(dnaGenome) 
```

```{r echo=FALSE}
# Load a single complete Micrococcus genome
datafile <- system.file('extdata', 'GeneCalling', 'SingleSeq.fa.gz', 
                        package='CompGenomicsBioc2022')
dnaGenome <- readDNAStringSet(datafile)
aaGenome <- translate(dnaGenome) 
```

Next, we're going to identify the genes in our sequence  using `FindGenes()` from the `DECIPHER` package.

`FindGenes()` returns a `Genes` object with information on where genes
start and end in the genome. We can then extract the sequences
corresponding to each gene using the `ExtractGenes()` function.

```{r results='hide'}
geneLocs <- FindGenes(dnaGenome)
genes <- ExtractGenes(geneLocs, dnaGenome, type='AAStringSet')
```

```{r}
# `Genes` object
geneLocs

# Sequences corresponding to each gene
genes
```

## Removing Non-Coding Regions

`FindGenes()` finds the genes themselves, but these may include non-coding regions. We're more interested in the regions that are actually translated into proteins, since these are what we'll try to annotate later. For this, we'll use the `FindNonCoding()` function.
Using `FindGenes()` with `FindNonCoding()` in this way greatly improves
the accuracy of `FindGenes()`.

`FindNonCoding()` is used with three main datafiles depending on the
data to analyze:

-   `data("NonCodingRNA_Archaea")` for Archaeal data
-   `data("NonCodingRNA_Bacteria")` for Bacterial data
-   `data("NonCodingRNA_Eukarya")` for Eukaryotic data

These include pretrained models with common non-coding patterns for the
relevant domain of life. If these pretrained models are insuffient, you
can train your own dataset using `LearnNonCoding()`, though this is
outside the scope of this workshop.

```{r results='hide'}
data("NonCodingRNA_Bacteria")
ncRNA <- NonCodingRNA_Bacteria

geneticRegions <- FindNonCoding(ncRNA, dnaGenome)

## Find annotations of noncoding regions
annotations <- attr(geneticRegions, "annotations")
geneMatches <- match(geneticRegions[,"Gene"], annotations)
noncodingAnnots <- sort(table(names(annotations)[geneMatches]))
```
```{r}
# What noncoding regions have we found and annotated?
noncodingAnnots
```

Once we've run `FindNonCoding()`, we can use `ExtractGenes()` as before
to pull out the coding regions. This will result in more accurate gene
calling than just running `FindGenes()` directly.

```{r results='hide'}
# Find Genes
genes <- FindGenes(dnaGenome, includeGenes=geneticRegions)
```

```{r}
# Genes in the genome
genes 
```

## Classification with `IDTAXA`

We now have a set of coding regions. Our last step for this section is
to try to annotate their function. This functionality is done with
`IdTaxa()` from the `DECIPHER` package. 

`IdTaxa()` requires a training set, which can be obtained in two ways.
The first is to download them from [DECIPHER's downloads page](http://www2.decipher.codes/Downloads.html), which contains prebuilt training sets for a variety of organisms. We'll be using an Actinobacteria dataset obtained from this website.

The other method is to build a training set yourself using 
`LearnTaxa()`. This will not be covered in this workshop, but more 
information is available on the `DECIPHER` documentation for people
that are interested.

In the interest of time, we'll just classify the first 10 genes. Note 
that our training set for `IdTaxa()` is trained on amino acids, 
so we have to first call `translate()` on our DNA sequences to be able
to provide `IdTaxa()` with amino acid sequences.

```{r include=FALSE}
makeTrainingSet <- system.file('extdata', 'GeneCalling', 
                               'RecombineIdTaxaTraining.R',
                            package='CompGenomicsBioc2022')
source(makeTrainingSet, local=knitr::knit_global())
```

```{r echo=FALSE}
downloadthis::download_link(
  link='http://www2.decipher.codes/Classification/TrainingSets/KEGG_Prokaryotes_r95.RData',
  button_label = "Download Training Set",
  button_type = "success",
  has_icon = TRUE,
  icon = "fa fa-download",
  self_contained = FALSE,
  class = "hvr-grow-shadow"
)
```

&nbsp;

```{r eval=FALSE}
# RData training set file is downloadable from the above button.
# You can also download it directly from the DECIPHER website.

load('/path/to/TrainingSet.RData')
trainingSet <- ExTrainList$TrainingSet
```

```{r results='hide', warning=FALSE}
# Find amino acid sequences corresponding to found genes
geneSeqs <- ExtractGenes(genes, dnaGenome, type="DNAStringSet")
geneSeqs <- translate(geneSeqs)

# Classify!
ids <- IdTaxa(geneSeqs[1:10], trainingSet)
```

Once we've finished calculating, we can either view the annotations
directly, or plot them as a taxonomy.

```{r}
# Looking at all results
ids

# Looking at a specific entry
ids[[1]]

# Plot the distribution of results
plot(ids, trainingSet)
```

```{css, echo=FALSE}
.pagination {
    display: flex;
    display: -webkit-flex;
    padding-left: 0;
    list-style: none;
    justify-content: center
}
```

::: center
<ul class="pagination pagination-lg">

<li class="page-item">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">«</a>

</li>

<li class="page-item">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">1</a>

</li>

<li class="page-item">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">2</a>

</li>

<li class="page-item active">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">3</a>

</li>

<li class="page-item">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">4</a>

</li>

<li class="page-item">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">5</a>

</li>

<li class="page-item">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">6</a>

</li>

<li class="page-item">
  <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html">7</a>
</li>

<li class="page-item">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">»</a>

</li>

</ul>
:::
