---
title: "Gene Calling and Annotation with DECIPHER"
author: Aidan Lakshman^[ahl27@pitt.edu]
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{GeneCallingAnnotation}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
    wrap: 72
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(DECIPHER)
newTreeLine <- system.file('extdata', 'TreeLine.R', 
                            package='CompGenomicsBioc2022')
source(newTreeLine, local=knitr::knit_global())
```

## Gene Calling and Annotation

At this point, we've learned how to read in some genomic data, and have
gained some basic familiarity working with it. The next step in our
pipeline is to take a set of genomes, identify the coding regions in
them, and predict the function of as many genetic regions as we can.
We'll start off by identifying the genes themselves.

## Finding Genes

In the interest of time, we're going to be using plasmid sequences from
a dataset of Archaeal genomes. Plasmids are small enough that we can
quickly execute our queries. We'll begin by reading in the data from a
`.fasta` file, as we did in the previous section.

```{r eval=FALSE}
library(DECIPHER)
source("TreeLine.R") # Shouldn't be necessary if using dev version
```

```{r}
# Load a single complete Micrococcus genome
datafile <- system.file('extdata', 'GeneCalling', 'SingleSeq.fa.gz', 
                        package='CompGenomicsBioc2022')
dnaGenome <- readDNAStringSet(datafile)
aaGenome <- translate(dnaGenome) 
```

Next, we're going to identify the genes in the first sequence in our
sequence set. We'll mark this as our 'genome' (although it's really a
plasmid), and predict genes within it using `FindGenes()` from the
`DECIPHER` package.

`FindGenes()` returns a `Genes` object with information on where genes
start and end in the genome. We can then extract the sequences
corresponding to each gene using the `ExtractGenes()` function.

```{r results='hide'}
geneLocs <- FindGenes(dnaGenome)
genes <- ExtractGenes(geneLocs, dnaGenome, type='AAStringSet')
```

```{r}
# `Genes` object
geneLocs

# Sequences corresponding to each gene
genes
```

## Removing Non-Coding Regions

`FindGenes()` finds the genes themselves, but these may not encode
coding regions. We're more interested in the regions that are actually
translated into proteins, since these are what we'll try to annotate
later. For this, we'll use the `FindNonCoding()` function.

`FindNonCoding()` is used with three main datafiles depending on the
data to analyze:

-   `data("NonCodingRNA_Archaea")` for Archaeal data
-   `data("NonCodingRNA_Bacteria")` for Bacterial data
-   `data("NonCodingRNA_Eukarya")` for Eukaryotic data

These include pretrained models with common non-coding patterns for the
relevant domain of life. If these pretrained models are insuffient, you
can train your own dataset using `LearnNonCoding()` (though this is
outside the scope of this workshop).

Once we've run `FindNonCoding()`, we can use `ExtractGenes()` as before
to pull out the coding regions. This will result in more accurate gene
calling than just running `FindGenes()` directly.

```{r results='hide'}
data("NonCodingRNA_Bacteria")
ncRNA <- NonCodingRNA_Bacteria

geneticRegions <- FindNonCoding(ncRNA, dnaGenome)

## Find annotations of noncoding regions
annotations <- attr(geneticRegions, "annotations")
geneMatches <- match(geneticRegions[,"Gene"], annotations)
noncodingAnnots <- sort(table(names(annotations)[geneMatches]))

# Find Genes
genes <- FindGenes(dnaGenome, includeGenes=geneticRegions)
```

```{r}
# What noncoding regions have we found and successfully annotated?
noncodingAnnots

# Genes in the genome
genes 
```

## Classification with `IDTAXA`

We now have a set of coding regions. Our last step for this section is
to try to annotate their function. This functionality is done with
`IdTaxa()` from the `DECIPHER` package. 

`IdTaxa()` requires a training set, which can be obtained in two ways.
The first is to download them from [DECIPHER's downloads page](http://www2.decipher.codes/Downloads.html), which contains prebuilt training sets for a variety of organisms. We'll be using an Actinobacteria dataset obtained from this website.

The other method is to build a training set yourself using 
`LearnTaxa()`. This will not be covered in this workshop, but more 
information is available on the `DECIPHER` documentation for people
that are interested.

In the interest of time, we'll just classify the first 10 genes. Note 
that our training set for `IdTaxa()` is trained on amino acids, 
so we have to provide it with amino acid sequences. This is why we 

```{r include=FALSE}
makeTrainingSet <- system.file('extdata', 'GeneCalling', 
                               'RecombineIdTaxaTraining.R',
                            package='CompGenomicsBioc2022')
source(makeTrainingSet, local=knitr::knit_global())
```

```{r results='hide', warning=FALSE}
# Find amino acid sequences corresponding to found genes
geneSeqs <- ExtractGenes(genes, dnaGenome, type="DNAStringSet")
geneSeqs <- translate(geneSeqs)

# Load training set
# Note: this training set is built on the fly due to GH file constraints
# You can download this directly from decipher.codes
datafile <- system.file('extdata', 'GeneCalling', 
                        'IdTaxaActinobacteriaTrainingSet.RData', 
                        package='CompGenomicsBioc2022')
load(datafile)

# Classify!
ids <- IdTaxa(geneSeqs[1:10], trainingSet)
```

Once we've finished calculating, we can either view the annotations
directly, or plot them as a taxonomy.

```{r}
# Looking at all results
ids

# Looking at a specific entry
ids[[1]]

# Plot the distribution of results
plot(ids, trainingSet)
```

```{css, echo=FALSE}
.pagination {
    display: flex;
    display: -webkit-flex;
    padding-left: 0;
    list-style: none;
    justify-content: center
}
```

::: center
<ul class="pagination pagination-lg">

<li class="page-item">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">«</a>

</li>

<li class="page-item">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">1</a>

</li>

<li class="page-item">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">2</a>

</li>

<li class="page-item active">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">3</a>

</li>

<li class="page-item">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">4</a>

</li>

<li class="page-item">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">5</a>

</li>

<li class="page-item">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">6</a>

</li>

<li class="page-item">
  <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html">7</a>
</li>

<li class="page-item">

<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">»</a>

</li>

</ul>
:::
