---
title: "Loading in Genome Data with DECIPHER"
author: Aidan Lakshman^[ahl27@pitt.edu]
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{LoadingGenomeData}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
    wrap: 72
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(DECIPHER)
```

```{r echo=FALSE, out.width='100%'}
knitr::include_graphics('images/PipelineGettingData.pdf')
```

## Reading in Sequencing Data

In order to work with sequencing data, we first have to get it into R in a format
that allows us to work with it. The most commonly used formats for genomic data are
the `XString` and `XStringSet` classes, which are available through the `Biostrings`
package.

`XStrings` come in 4 distinct flavors, depending on the characters allowed:

- `DNAString`, for DNA data (`ATGC`, plus gaps and ambiguity codes)
- `RNAString`, for RNA data (`AUGC`, plus gaps and ambiguity codes)
- `AAString`, for amino acid data (20 amino acids plus `U`, `O`, ambiguity codes, and unknown/gaps)
- `BString`, for any combination of any letters

When `XString` objects of the same type are combined, they become an `XStringSet`. 
This provides an easy way to store and work with genomics data. Below is an
example of manually creating an `XStringSet`:
```{r eval=FALSE}
library(DECIPHER) # Auto-imports Biostrings
```

```{r}

sequences <- c('AGACTCGCA',
               'AGACGGTCA',
               'TCATTAGTT',
               'TGCTGAAAA',
               'AGCTGTTGC')

sequenceSet <- DNAStringSet(sequences)
sequenceSet
```

We can also translate DNA sequences to their amino acid counterparts with `translate()`.
```{r warning=FALSE}
translate(sequenceSet)
```

Manually typing in sequences obviously isn't a great system. Most sequencing data 
are downloadable as `.fasta` or `.fastq` files, so let's look at a more realistic
use-case that reads in a data from a `.fasta`. We're going to be using an example
set of 17 complete *Micrococcus* genomes obtained from [NCBI GenBank](https://www.ncbi.nlm.nih.gov/data-hub/genome/?taxon=1269&annotated_only=true&refseq_annotation=true&assembly_level=3%3A3). 
I've predownloaded these and repackaged them into a single `.FASTA` file for this 
workshop (the file extension is `.fa.gz` since it's been compressed for size).
See the [Setup page](https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html) for more information on how to download this data yourself.

These sequences have been trimmed to facilitate completing analysis in a single 
workshop slot. Feel free to experiment with the complete sequences later.
You can also build this object yourself by downloading the `_genomic.fasta` files
from GenBank and then reading them in individually.

```{css echo=FALSE}
.hvr-grow-shadow {
  display: inline-block;
  vertical-align: middle;
  -webkit-transform: perspective(1px) translateZ(0);
  transform: perspective(1px) translateZ(0);
  box-shadow: 0 0 1px rgba(0, 0, 0, 0);
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: box-shadow, transform;
  transition-property: box-shadow, transform;
}
.hvr-grow-shadow:hover, .hvr-grow-shadow:focus, .hvr-grow-shadow:active {
  box-shadow: 0 10px 10px -10px rgba(0, 0, 0, 0.5);
  -webkit-transform: scale(1.1);
  transform: scale(1.1);
}
```
```{r echo=FALSE}
f <- system.file('extdata', 'LoadingData', 'MicroFASTAtrimmed.fa.gz', 
                            package='CompGenomicsBioc2022')
downloadthis::download_file(
  path = f,
  button_label = "Download Example Sequences",
  button_type = "success",
  has_icon = TRUE,
  icon = "fa fa-download",
  self_contained = FALSE,
  class = "hvr-grow-shadow"
)
```


```{r, echo=FALSE}
exampledata <- system.file('extdata', 'LoadingData', 'MicroFASTAtrimmed.fa.gz', 
                            package='CompGenomicsBioc2022')

exStringSet <- readDNAStringSet(exampledata, format="fasta")
```

&nbsp;

```{r, eval=FALSE}
# MicroFASTAtrimmed.fa.gz is loaded from the above button
# also available at extdata/LoadingData/ within this package

exampleSeqs <- '/path/to/MicroFASTAtrimmed.fa.gz'

exStringSet <- readDNAStringSet(exampleSeqs, format="fasta")
# we could also use format='fastq' for FASTQ datasets
```
Success! Now we have a large example dataset to work with.

If we wanted to visualize these sequences, we can open them in a web browser
using `BrowseSeqs()` from `DECIPHER`:

```{r, eval=FALSE}
BrowseSeqs(exStringSet)
```

```{r echo=FALSE, out.width='100%'}
knitr::include_graphics('images/browseSeqs.png')
```

## Aligning Sequences

Now that we have some sequences, let's explore some of the ways we can manipulate
them. A complete demo of `Biostrings` is outside the scope of this workshop, so
we'll just focus on functionality added via `DECIPHER`. 

A common analysis in 
bioinformatics is aligning sequences. This is easily achievable with either the 
`AlignSeqs()` function or the `AlignTranslation()` functions from `DECIPHER`. 
`AlignTranslation()` aligns sequences based on their translated amino acid 
sequences, and is significantly more accurate for coding sequences.

```{r}
# Align the sequences
aliNoTranslate <- AlignSeqs(exStringSet, verbose=FALSE)

# Aligning using translated amino acid sequences
aliTranslate <- AlignTranslation(exStringSet, verbose=FALSE)

# Visualize through R 
aliNoTranslate
```

Alignments tend to start with a lot of gaps, and as a result viewing them through
R isn't always super informative.
As before, we can visualize this alignment in a much better way using `BrowseSeqs()`:

```{r, eval=FALSE}
BrowseSeqs(aliNoTranslate)
```

The output should resemble the following:

```{r echo=FALSE, out.width='100%'}
knitr::include_graphics('images/picAlignment.png')
```

## Conclusion

Now we've managed to read in sequence data and perform some basic analyses. 
With this knowledge, we can proceed to the next step of the pipeline: finding
and annotating genes.

<!-- ## Alignment-Free Analysis -->

<!-- Another common analysis is to look at syntenic hits between sequences without  -->
<!-- aligning them. The result allows us to easily visualize differences between  -->
<!-- two sequences without having to align them. Let's look at an example of finding -->
<!-- syntenic hits between the first 6 sequences in our set. -->

<!-- ```{r results='hide', fig.keep='all'} -->
<!-- to_select <- c(4,6,8,9) -->
<!-- synData <- exStringSet[to_select] -->
<!-- names(synData) <- as.character(to_select) -->
<!-- dbConn <- dbConnect(SQLite(), ":memory:") -->
<!-- Seqs2DB(synData, "XStringSet", dbConn, identifier=as.character(to_select)) -->

<!-- syn <- FindSynteny(dbConn, verbose=FALSE) -->
<!-- dbDisconnect(dbConn) -->

<!-- pairs(syn) -->
<!-- ``` -->

<!-- What are we looking at with this plot? Each plot compares two genomes, and each -->
<!-- dot is a syntenic hit between the two genomes. The X position of the dot is its -->
<!-- location on the first genome, and the Y position is its position on the second genome. -->
<!-- If we had identical genomes, we would expect to see a diagonal line $y=x$, indicating -->
<!-- that all bases occur at the same place in both genomes. Gaps indicate areas that  -->
<!-- do not match, and points off the diagonal indicate matching areas in different places on  -->
<!-- each genomes. -->

<!-- You may also notice that this example creates a SQLite table prior to running `FindSynteny()`. -->
<!-- `DECIPHER` is built around working with sequences via SQLite tables to allow for  -->
<!-- analysis at scale. When working with hundreds of thousands of genomes, we can't always -->
<!-- just store the files within memory on our computer. This won't be covered in this -->
<!-- tutorial, but advanced users can refer to the `DECIPHER` docs to find out more about -->
<!-- using database files. I've also included more information on these in the Function -->
<!-- Reference page for this section. -->

```{css, echo=FALSE}
.pagination {
    display: flex;
    display: -webkit-flex;
    padding-left: 0;
    list-style: none;
    justify-content: center
}
```

<div class="center">
  <ul class="pagination pagination-lg">
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">&laquo;</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">1</a>
    </li>
    <li class="page-item active">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">2</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">3</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">4</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">5</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">6</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html">7</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">&raquo;</a>
    </li>
  </ul>
</div>