---
title: "Loading in Genome Data with DECIPHER"
author: Aidan Lakshman^[ahl27@pitt.edu]
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{LoadingGenomeData}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
    wrap: 72
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Coming Soon!
This tutorial will be coming soon! It will cover the following topics:

* Importing `.fasta` sequences into `R` as an `XStringSet`
* Basic operations with `XString` objects (ex. `reverseComplement`)
* Alignment-free analysis with `FindSynteny()`
* Aligning sequences with `AlignSeqs()`, `AlignTranslation()`, and `AlignSynteny()`
* Exporting sequences to SQLite databases for working at scale
* Retrieving sequences from SQLite databases
* Visualizing sequences with `BrowseSeqs()` and `BrowseDB()`

At the end of this section, we will have successfully imported sequencing data from standard formats into a format conducive to analyses in R. We will also have gained some basic information on the data in our dataset.

Depending on timing, it may be a good idea to skip databases entirely and just
focus on working with `XStringSets`. We can just use `BrowseSeqs()` for visualization, rather than using `BrowseDB()`.

```{r eval=FALSE}
library(DECIPHER)

dna
dbConn <- dbConnect(SQLite(), ":memory:")
Seqs2DB(dna, "FASTA", dbConn, "Seq")
BrowseDB(dbConn)
```

```{css, echo=FALSE}
.pagination {
    display: flex;
    display: -webkit-flex;
    padding-left: 0;
    list-style: none;
    justify-content: center
}
```

<div class="center">
  <ul class="pagination pagination-lg">
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">&laquo;</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">1</a>
    </li>
    <li class="page-item active">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">2</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">3</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">4</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">5</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">6</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">&raquo;</a>
    </li>
  </ul>
</div>