---
title: "Loading in Genome Data with DECIPHER"
author: Aidan Lakshman^[ahl27@pitt.edu]
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{LoadingGenomeData}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
    wrap: 72
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

```

## Reading in Sequencing Data

In order to work with sequencing data, we first have to get it into R in a format
that allows us to work with it. The most commonly used format for genomic data is
the `XString` and `XStringSet` classes, which are available through the `Biostrings`
package.

`XString`s come in 4 distinct flavors, depending on the characters allowed:
- `DNAString`, for DNA data (ATGC, plus gaps and ambiguity codes)
- `RNAString`, for RNA data (AUGC, plus gaps and ambiguity codes)
- `AAString`, for amino acid data (20 amino acids plus ambiguity codes, U, O, and unknown/gaps)
- `BString`, for any combination of any letters

When `XString` objects of the same type are combined, they become an `XStringSet`. 
This provides an easy way to store and work with genomics data. Below is an
example of manually creating an `XStringSet`:
```{r eval=FALSE}
library(DECIPHER)

sequences <- c('AGACT',
               'AGACG',
               'TCATT',
               'TGCTG',
               'AGCTG')

sequenceSet <- DNAStringSet(sequences)
sequenceSet
```

Manually typing in sequences obviously isn't a great system. Most sequencing data 
are downloadable as `.fasta` or `.fastq` files, so let's look at a more realistic
use-case that reads in a data from a `.fasta`.

```{r}
exampledata <- system.file('extdata', '50S_ribosomal_protein_L2.fas', 
                            package='CompGenomicsBioc2022', mustWork=TRUE)

# note that we can also use format='fastq' for FASTQ datasets
exStringSet <- readDNAStringSet(exampledata, format="fasta")
exSSet <- exStringSet[1:100]
exSSet
```

Success! Now we have a large example dataset to work with.

If we wanted to visualize these sequences, we can open them in a web browser
using `BrowseSeqs()` from `DECIPHER`:

```{r, eval=FALSE}
BrowseSeqs(exSSet)
```

## Aligning Sequences

Now that we have some sequences, let's explore some of the ways we can manipulate
them. A complete demo of `Biostrings` is outside the scope of this workshop, so
we'll just focus on functionality added via `DECIPHER`. A common analysis in 
bioinformatics is aligning sequences. This is easily achievable with either the 
`AlignSeqs()` function or the `AlignTranslation()` functions from `DECIPHER`. 
`AlignTranslation()` aligns sequences based on their translated amino acid sequences, 
and is significantly more accurate for DNA or RNA sequences.

```{r}
ali1 <- AlignSeqs(exSSet, verbose=FALSE)
ali2 <- AlignTranslation(exSSet, verbose=FALSE)

ali1
ali2
```

As before, we can visualize this alignment sequences using `BrowseSeqs()`:

```{r, eval=FALSE}
BrowseSeqs(ali2)
```

## Alignment-Free Analysis

Another common analysis is to look at syntenic hits between sequences without 
aligning them. The result allows us to easily visualize differences between 
two sequences without having to align them. Let's look at an example of finding
syntenic hits between the first 6 sequences in our set.

```{r}
synData <- exSSet[1:6]
names(synData) <- 1:6
dbConn <- dbConnect(SQLite(), ":memory:")
Seqs2DB(synData, "XStringSet", dbConn, "example")

syn <- FindSynteny(dbConn, verbose=FALSE)
syn
pairs(syn)
```

What are we looking at with this plot? Each plot compares two genomes, and each
dot is a syntenic hit between the two genomes. The X position of the dot is its
location on the first genome, and the Y position is its position on the second genome.
If we had identical genomes, we would expect to see a diagonal line $y=x$, indicating
that all bases occur at the same place in both genomes. 

You may also notice that this example creates a SQLite table prior to running `FindSynteny()`.
`DECIPHER` is built around working with sequences via SQLite tables to allow for 
analysis at scale. When working with hundreds of thousands of genomes, we can't always
just store the files within memory on our computer. This won't be covered in this
tutorial, but advanced users can refer to the `DECIPHER` docs to find out more about
using database files. I've also included more information on these in the Function
Reference page for this section.

```{css, echo=FALSE}
.pagination {
    display: flex;
    display: -webkit-flex;
    padding-left: 0;
    list-style: none;
    justify-content: center
}
```

<div class="center">
  <ul class="pagination pagination-lg">
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">&laquo;</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">1</a>
    </li>
    <li class="page-item active">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">2</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">3</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">4</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">5</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">6</a>
    </li>
    <li class="page-item">
      <a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">&raquo;</a>
    </li>
  </ul>
</div>