<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Performs profile-to-profile alignment of multiple unaligned sequences following a guide tree."><title>Align a Set of Unaligned Sequences — AlignSeqs • CompGenomicsBioc2022</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Align a Set of Unaligned Sequences — AlignSeqs"><meta property="og:description" content="Performs profile-to-profile alignment of multiple unaligned sequences following a guide tree."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/workshop_syllabus.html">Comparative Genomics with SynExtend and DECIPHER</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Align a Set of Unaligned Sequences</h1>
      
      <div class="d-none name"><code>AlignSeqs.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Performs profile-to-profile alignment of multiple unaligned sequences following a guide tree.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">AlignSeqs</span><span class="op">(</span><span class="va">myXStringSet</span>,
         guideTree <span class="op">=</span> <span class="cn">NULL</span>,
         iterations <span class="op">=</span> <span class="fl">2</span>,
         refinements <span class="op">=</span> <span class="fl">1</span>,
         gapOpening <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">18</span>, <span class="op">-</span><span class="fl">14</span><span class="op">)</span>,
         gapExtension <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="op">-</span><span class="fl">2</span><span class="op">)</span>,
         useStructures <span class="op">=</span> <span class="cn">TRUE</span>,
         structures <span class="op">=</span> <span class="cn">NULL</span>,
         FUN <span class="op">=</span> <span class="va">AdjustAlignment</span>,
         levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.9</span>, <span class="fl">0.7</span>, <span class="fl">0.7</span>, <span class="fl">0.4</span>, <span class="fl">10</span>, <span class="fl">5</span>, <span class="fl">5</span>, <span class="fl">2</span><span class="op">)</span>,
         alphabet <span class="op">=</span> <span class="va">AA_REDUCED</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
         processors <span class="op">=</span> <span class="fl">1</span>,
         verbose <span class="op">=</span> <span class="cn">TRUE</span>,
         <span class="va">...</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>myXStringSet</dt>
<dd><p>An <code>AAStringSet</code>, <code>DNAStringSet</code>, or <code>RNAStringSet</code> object of unaligned sequences.</p></dd>
<dt>guideTree</dt>
<dd><p>Either <code>NULL</code> or a <code>dendrogram</code> giving the ordered tree structure in which to align profiles.  If <code>NULL</code> then a guide tree will be automatically constructed based on the order of shared k-mers.</p></dd>
<dt>iterations</dt>
<dd><p>Number of iteration steps to perform.  During each iteration step the guide tree is regenerated based on the alignment and the sequences are realigned.</p></dd>
<dt>refinements</dt>
<dd><p>Number of refinement steps to perform.  During each refinement step groups of sequences are realigned to rest of the sequences, and the best of these two alignments (before and after realignment) is kept.</p></dd>
<dt>gapOpening</dt>
<dd><p>Single numeric giving the cost for opening a gap in the alignment, or two numbers giving the minimum and maximum costs.  In the latter case the cost will be varied depending upon whether the groups of sequences being aligned are nearly identical or maximally distant.</p></dd>
<dt>gapExtension</dt>
<dd><p>Single numeric giving the cost for extending an open gap in the alignment, or two numbers giving the minimum and maximum costs.  In the latter case the cost will be varied depending upon whether the groups of sequences being aligned are nearly identical or maximally distant.</p></dd>
<dt>useStructures</dt>
<dd><p>Logical indicating whether to use secondary structure predictions during alignment.  If <code>TRUE</code> (the default), secondary structure probabilities will be automatically calculated for amino acid and RNA sequences if they are not provided (i.e., when <code>structures</code> is <code>NULL</code>).</p></dd>
<dt>structures</dt>
<dd><p>Either a list of secondary structure probabilities matching the <code>structureMatrix</code>, such as that output by <code>PredictHEC</code> or <code>PredictDBN</code>, or <code>NULL</code> to generate the structures automatically.  Only applicable if <code>myXStringSet</code> is an <code>AAStringSet</code> or <code>RNAStringSet</code>.</p></dd>
<dt>FUN</dt>
<dd><p>A function to be applied after each profile-to-profile alignment.  (See details section below.)</p></dd>
<dt>levels</dt>
<dd><p>Numeric with eight elements specifying the levels at which to trigger events.  (See details section below.)</p></dd>
<dt>alphabet</dt>
<dd><p>Character vector of amino acid groupings used to reduce the 20 standard amino acids into smaller groups.  Alphabet reduction helps to find more distant homologies between sequences.  A non-reduced amino acid alphabet can be used by setting <code>alphabet</code> equal to <code>AA_STANDARD</code>.  Only applicable if <code>myXStringSet</code> is an <code>AAStringSet</code>.</p></dd>
<dt>processors</dt>
<dd><p>The number of processors to use, or <code>NULL</code> to automatically detect and use all available processors.</p></dd>
<dt>verbose</dt>
<dd><p>Logical indicating whether to display progress.</p></dd>
<dt>...</dt>
<dd><p>Further arguments to be passed directly to <code>AlignProfiles</code>, including <code>perfectMatch</code>, <code>misMatch</code>, <code>gapPower</code>, <code>terminalGap</code>, <code>restrict</code>, <code>anchor</code>, <code>normPower</code>, <code>standardize</code>, <code>substitutionMatrix</code>, and <code>structureMatrix</code>.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The profile-to-profile method aligns a sequence set by merging profiles along a guide tree until all the input sequences are aligned.  This process has three main steps:  (1)  If <code>guideTree=NULL</code>, an initial single-linkage guide tree is constructed based on a distance matrix of shared k-mers.  Alternatively, a <code>dendrogram</code> can be provided as the initial <code>guideTree</code>.  (2)  If <code>iterations</code> is greater than zero, then a UPGMA guide tree is built based on the initial alignment and the sequences are re-aligned along this tree.  This process repeated <code>iterations</code> times or until convergence.  (3)  If <code>refinements</code> is greater than zero, then subsets of the alignment are re-aligned to the remainder of the alignment.  This process generates two alignments, the best of which is chosen based on its sum-of-pairs score.  This refinement process is repeated <code>refinements</code> times, or until convergence.</p>
<p>The purpose of <code>levels</code> is to speed-up the alignment process by not running time consuming processes when they are unlikely to change the outcome.  The first four <code>levels</code> control when <code>refinements</code> occur and the function <code>FUN</code> is run on the alignment.  The default <code>levels</code> specify that these events should happen when above 0.9 (AA; <code>levels[1]</code>) or 0.7 (DNA/RNA; <code>levels[3]</code>) average dissimilarity on the initial tree, when above 0.7 (AA; <code>levels[2]</code>) or 0.4 (DNA/RNA; <code>levels[4]</code>) average dissimilarity on the iterative tree(s), and after every tenth improvement made during refinement.  The sixth element of levels (<code>levels[6]</code>) prevents <code>FUN</code> from being applied at any point to less than 5 sequences.</p>
<p>The <code>FUN</code> function is always applied before returning the alignment so long as there are at least <code>levels[6]</code> sequences.  The default <code>FUN</code> is <code>AdjustAlignment</code>, but <code>FUN</code> can be any function that takes in an <code>XStringSet</code> as its first argument, as well as <code>weights</code>, <code>processors</code>, and <code>substitutionMatrix</code> as optional arguments.  For example, the default <code>FUN</code> could be altered to not perform any changes by setting it equal to <code>function(x, ...) return(x)</code>, where <code>x</code> is an <code>XStringSet</code>.</p>
<p>Secondary structures are automatically computed for amino acid and RNA sequences unless <code>structures</code> are provided or <code>useStructures</code> is <code>FALSE</code>.  The default <code>structureMatrix</code> is used unless an alternative is provided.  For RNA sequences, secondary structures are only computed when the total length of the initial guide tree is at least 5 (<code>levels[7]</code>) or the length of subsequent trees is at least 2 (<code>levels[8]</code>).</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>An <code>XStringSet</code> of aligned sequences.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Wright, E. S. (2015). DECIPHER: harnessing local sequence context to improve protein multiple sequence alignment. BMC Bioinformatics, 16, 322. http://doi.org/10.1186/s12859-015-0749-z</p>
<p>Wright, E. S. (2020). RNAconTest: comparing tools for noncoding RNA multiple sequence alignment based on structural consistency. RNA 2020, 26, 531-540.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Erik Wright <a href="mailto:eswright@pitt.edu">eswright@pitt.edu</a></p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code>AdjustAlignment</code>, <code>AlignDB</code>, <code>AlignProfiles</code>, <code>AlignSynteny</code>, <code><a href="AlignTranslation.html">AlignTranslation</a></code>, <code>ReadDendrogram</code>, <code><a href="TreeLine.html">TreeLine</a></code>, <code>StaggerAlignment</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">db</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Bacteria_175seqs.sqlite"</span>, package<span class="op">=</span><span class="st">"DECIPHER"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">dna</span> <span class="op">&lt;-</span> <span class="fu"><a href="SearchDB.html">SearchDB</a></span><span class="op">(</span><span class="va">db</span>, remove<span class="op">=</span><span class="st">"all"</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Search Expression:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> select row_names, sequence from _Seqs where row_names in (select row_names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> from Seqs)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet of length: 175</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.04 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="va">alignedDNA</span> <span class="op">&lt;-</span> <span class="fu">AlignSeqs</span><span class="op">(</span><span class="va">dna</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on shared 9-mers:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 1.28 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Clustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.06 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Aligning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 2.54 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iteration 1 of 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on alignment:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.08 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reclustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.06 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Realigning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 1.91 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iteration 2 of 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on alignment:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.08 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reclustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.06 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Realigning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.67 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="fu"><a href="BrowseSeqs.html">BrowseSeqs</a></span><span class="op">(</span><span class="va">alignedDNA</span>, highlight<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># use secondary structure with RNA sequences</span></span>
<span class="r-in"><span class="va">alignedRNA</span> <span class="op">&lt;-</span> <span class="fu">AlignSeqs</span><span class="op">(</span><span class="fu">RNAStringSet</span><span class="op">(</span><span class="va">dna</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on shared 9-mers:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 1.25 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Clustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.04 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Aligning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 3.2 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iteration 1 of 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Predicting RNA Secondary Structures:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 2.9 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on alignment:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.08 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reclustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.06 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Realigning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 2.93 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iteration 2 of 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Predicting RNA Secondary Structures:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 2.91 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on alignment:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.08 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reclustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.05 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Realigning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.65 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="fu"><a href="BrowseSeqs.html">BrowseSeqs</a></span><span class="op">(</span><span class="va">alignedRNA</span>, highlight<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>

