<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Performs alignment of a set of DNA or RNA sequences by aligning their corresponding amino acid sequences."><title>Align Sequences By Their Amino Acid Translation — AlignTranslation • CompGenomicsBioc2022</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Align Sequences By Their Amino Acid Translation — AlignTranslation"><meta property="og:description" content="Performs alignment of a set of DNA or RNA sequences by aligning their corresponding amino acid sequences."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Coevolution%20Networks.html">Coevolutionary Networks</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com" aria-label="My Website">
    <span class="fa fa-potato"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Align Sequences By Their Amino Acid Translation</h1>
      
      <div class="d-none name"><code>AlignTranslation.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Performs alignment of a set of DNA or RNA sequences by aligning their corresponding amino acid sequences.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">AlignTranslation</span><span class="op">(</span><span class="va">myXStringSet</span>,
                 sense <span class="op">=</span> <span class="st">"+"</span>,
                 direction <span class="op">=</span> <span class="st">"5' to 3'"</span>,
                 readingFrame <span class="op">=</span> <span class="cn">NA</span>,
                 type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">myXStringSet</span><span class="op">)</span>,
                 geneticCode <span class="op">=</span> <span class="va">GENETIC_CODE</span>,
                 <span class="va">...</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>myXStringSet</dt>
<dd><p>A <code>DNAStringSet</code> or <code>RNAStringSet</code> object of unaligned sequences.</p></dd>
<dt>sense</dt>
<dd><p>Single character specifying sense of the input sequences, either the positive (<code>"+"</code>) coding strand or negative (<code>"-"</code>) non-coding strand.</p></dd>
<dt>direction</dt>
<dd><p>Direction of the input sequences, either <code>"5' to 3'"</code> or <code>"3' to 5'"</code>.</p></dd>
<dt>readingFrame</dt>
<dd><p>Numeric vector giving a single reading frame for all of the sequences, or an individual reading frame for each sequence in <code>myXStringSet</code>.  The <code>readingFrame</code> can be either <code>1</code>, <code>2</code>, <code>3</code> to begin translating on the first, second, and third nucleotide position, or <code>NA</code> (the default) to guess the reading frame.  (See details section below.)</p></dd>
<dt>type</dt>
<dd><p>Character string indicating the type of output desired.  This should be (an abbreviation of) one of <code>"DNAStringSet"</code>, <code>"RNAStringSet"</code>, <code>"AAStringSet"</code>, or <code>"both"</code>.  (See value section below.)</p></dd>
<dt>geneticCode</dt>
<dd><p>Either a character vector giving the genetic code in the same format as <code>GENETIC_CODE</code> (the default), or a list containing one genetic code for each sequence in <code>myXStringSet</code>.</p></dd>
<dt>...</dt>
<dd><p>Further arguments to be passed directly to <code><a href="AlignSeqs.html">AlignSeqs</a></code>, including <code>gapOpening</code>, <code>gapExtension</code>, <code>gapPower</code>, <code>terminalGap</code>, <code>restrict</code>, <code>anchor</code>, <code>normPower</code>, <code>substitutionMatrix</code>, <code>structureMatrix</code>, <code>standardize</code>, <code>alphabet</code>, <code>guideTree</code>, <code>iterations</code>, <code>refinements</code>, <code>useStructures</code>, <code>structures</code>, <code>FUN</code>, and <code>levels</code>.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Alignment of proteins is often more accurate than alignment of their coding nucleic acid sequences.  This function aligns the input nucleic acid sequences via aligning their translated amino acid sequences.  First, the input sequences are translated according to the specified <code>sense</code>, <code>direction</code>, and <code>readingFrame</code>.  The resulting amino acid sequences are aligned using <code>AlignSeqs</code>, and this alignment is (conceptually) reverse translated into the original sequence type, <code>sense</code>, and <code>direction</code>.  Not only is alignment of protein sequences generally more accurate, but aligning translations will ensure that the reading frame is maintained in the nucleotide sequences.</p>
<p>If the <code>readingFrame</code> is <code>NA</code> (the default) then an attempt is made to guess the reading frame of each sequence based on the number of stop codons in the translated amino acids.  For each sequence, the first reading frame will be chosen (either <code>1</code>, <code>2</code>, or <code>3</code>) without stop codons, except in the final position.  If the number of stop codons is inconclusive for a sequence then the reading frame will default to <code>1</code>.  The entire length of each sequence is translated in spite of any stop codons identified.  Note that this method is only constructive in circumstances where there is a substantially long coding sequence with at most a single stop codon expected in the final position, and therefore it is preferable to specify the reading frame of each sequence if it is known.</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>An <code>XStringSet</code> of the class specified by <code>type</code>, or a list of two components (nucleotides and amino acids) if <code>type</code> is <code>"both"</code>.  Note that incomplete starting and ending codons will be translated into the mask character ("+") if the result includes an <code>AAStringSet</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Wright, E. S. (2015). DECIPHER: harnessing local sequence context to improve protein multiple sequence alignment. BMC Bioinformatics, 16, 322. http://doi.org/10.1186/s12859-015-0749-z</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Erik Wright <a href="mailto:eswright@pitt.edu">eswright@pitt.edu</a></p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code>AlignDB</code>, <code>AlignProfiles</code>, <code><a href="AlignSeqs.html">AlignSeqs</a></code>, <code>AlignSynteny</code>, <code>CorrectFrameshifts</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co"># first three sequences translate to  MFITP*</span></span>
<span class="r-in"><span class="co"># and the last sequence translates as MF-TP*</span></span>
<span class="r-in"><span class="va">rna</span> <span class="op">&lt;-</span> <span class="fu">RNAStringSet</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AUGUUCAUCACCCCCUAA"</span>, <span class="st">"AUGUUCAUAACUCCUUGA"</span>,</span>
<span class="r-in">  <span class="st">"AUGUUCAUUACACCGUAG"</span>, <span class="st">"AUGUUUACCCCAUAA"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="va">RNA</span> <span class="op">&lt;-</span> <span class="fu"><a href="AlignSeqs.html">AlignSeqs</a></span><span class="op">(</span><span class="va">rna</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-in"><span class="va">RNA</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> RNAStringSet object of length 4:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]    18 AUGUUCAUCACCCCCUAA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2]    18 AUGUUCAUAACUCCUUGA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3]    18 AUGUUCAUUACACCGUAG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [4]    18 AUGU---UUACCCCAUAA</span>
<span class="r-in"></span>
<span class="r-in"><span class="va">RNA</span> <span class="op">&lt;-</span> <span class="fu">AlignTranslation</span><span class="op">(</span><span class="va">rna</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-in"><span class="va">RNA</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> RNAStringSet object of length 4:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]    18 AUGUUCAUCACCCCCUAA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2]    18 AUGUUCAUAACUCCUUGA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3]    18 AUGUUCAUUACACCGUAG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [4]    18 ---AUGUUUACCCCAUAA</span>
<span class="r-in"></span>
<span class="r-in"><span class="va">AA</span> <span class="op">&lt;-</span> <span class="fu">AlignTranslation</span><span class="op">(</span><span class="va">rna</span>, type<span class="op">=</span><span class="st">"AAStringSet"</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-in"><span class="va">AA</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> AAStringSet object of length 4:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]     6 MFITP*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2]     6 MFITP*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3]     6 MFITP*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [4]     6 -MFTP*</span>
<span class="r-in"></span>
<span class="r-in"><span class="va">BOTH</span> <span class="op">&lt;-</span> <span class="fu">AlignTranslation</span><span class="op">(</span><span class="va">rna</span>, type<span class="op">=</span><span class="st">"both"</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-in"><span class="va">BOTH</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [[1]]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> RNAStringSet object of length 4:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]    18 AUGUUCAUCACCCCCUAA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2]    18 AUGUUCAUAACUCCUUGA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3]    18 AUGUUCAUUACACCGUAG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [4]    18 ---AUGUUUACCCCAUAA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [[2]]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> AAStringSet object of length 4:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     width seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1]     6 MFITP*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2]     6 MFITP*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3]     6 MFITP*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [4]     6 -MFTP*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># example of aligning many protein coding sequences:</span></span>
<span class="r-in"><span class="va">fas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"50S_ribosomal_protein_L2.fas"</span>, package<span class="op">=</span><span class="st">"DECIPHER"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">dna</span> <span class="op">&lt;-</span> <span class="fu">readDNAStringSet</span><span class="op">(</span><span class="va">fas</span><span class="op">)</span></span>
<span class="r-in"><span class="va">DNA</span> <span class="op">&lt;-</span> <span class="fu">AlignTranslation</span><span class="op">(</span><span class="va">dna</span><span class="op">)</span> <span class="co"># align the translation then reverse translate</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on shared 5-mers:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.55 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Clustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.05 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Aligning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 1.48 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iteration 1 of 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on alignment:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.09 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reclustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.07 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Realigning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.65 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iteration 2 of 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on alignment:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.09 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reclustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.08 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Realigning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.09 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="va">DNA</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet object of length 317:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       width seq                                             names               </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]  1011 ----------------------...---------------------- Rickettsia prowaz...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]  1011 ----------------------...---------------------- Porphyromonas gin...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]  1011 ----------------------...---------------------- Porphyromonas gin...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]  1011 ----------------------...---------------------- Porphyromonas gin...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]  1011 ----------------------...---------------------- Pasteurella multo...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ...   ... ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [313]  1011 ----------------------...---------------------- Pectobacterium at...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [314]  1011 ----------------------...---------------------- Acinetobacter sp....</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [315]  1011 ----------------------...GGGTCGCGGTGGTCGTCAGTCT Thermosynechococc...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [316]  1011 ----------------------...---------------------- Bradyrhizobium ja...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [317]  1011 ----------------------...AGGT------------------ Gloeobacter viola...</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># using a mixture of standard and non-standard genetic codes</span></span>
<span class="r-in"><span class="va">gC1</span> <span class="op">&lt;-</span> <span class="fu">getGeneticCode</span><span class="op">(</span>id_or_name2<span class="op">=</span><span class="st">"1"</span>, full.search<span class="op">=</span><span class="cn">FALSE</span>, as.data.frame<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-in"><span class="co"># Mollicutes use an alternative genetic code</span></span>
<span class="r-in"><span class="va">gC2</span> <span class="op">&lt;-</span> <span class="fu">getGeneticCode</span><span class="op">(</span>id_or_name2<span class="op">=</span><span class="st">"4"</span>, full.search<span class="op">=</span><span class="cn">FALSE</span>, as.data.frame<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-in"><span class="va">w</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"Mycoplasma|Ureaplasma"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dna</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="va">gC</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">vector</a></span><span class="op">(</span><span class="st">"list"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">dna</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="va">gC</span><span class="op">[</span><span class="op">-</span><span class="va">w</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">gC1</span><span class="op">)</span></span>
<span class="r-in"><span class="va">gC</span><span class="op">[</span><span class="va">w</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">gC2</span><span class="op">)</span></span>
<span class="r-in"><span class="va">AA</span> <span class="op">&lt;-</span> <span class="fu">AlignTranslation</span><span class="op">(</span><span class="va">dna</span>, geneticCode<span class="op">=</span><span class="va">gC</span>, type<span class="op">=</span><span class="st">"AAStringSet"</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on shared 5-mers:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.56 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Clustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.05 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Aligning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 1.47 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iteration 1 of 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on alignment:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.09 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reclustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.07 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Realigning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.65 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iteration 2 of 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on alignment:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.09 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reclustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.1 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Realigning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.09 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="fu"><a href="BrowseSeqs.html">BrowseSeqs</a></span><span class="op">(</span><span class="va">AA</span><span class="op">)</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>

