<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Opens an html file in a web browser to show the sequences in an XStringSet."><title>View Sequences in a Web Browser — BrowseSeqs • CompGenomicsBioc2022</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="View Sequences in a Web Browser — BrowseSeqs"><meta property="og:description" content="Opens an html file in a web browser to show the sequences in an XStringSet."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">Coevolutionary Networks</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com" aria-label="My Website">
    <span class="fa fa-potato"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>View Sequences in a Web Browser</h1>
      
      <div class="d-none name"><code>BrowseSeqs.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Opens an html file in a web browser to show the sequences in an <code>XStringSet</code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">myXStringSet</span>,
           htmlFile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"/myXStringSet.html"</span>, sep <span class="op">=</span> <span class="st">""</span><span class="op">)</span>,
           openURL <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span>,
           colorPatterns <span class="op">=</span> <span class="cn">TRUE</span>,
           highlight <span class="op">=</span> <span class="cn">NA</span>,
           patterns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"-"</span>, <span class="fu">alphabet</span><span class="op">(</span><span class="va">myXStringSet</span>, baseOnly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>,
           colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substring</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">rainbow</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">patterns</span><span class="op">)</span>,
                              v<span class="op">=</span><span class="fl">0.8</span>, start<span class="op">=</span><span class="fl">0.9</span>, end<span class="op">=</span><span class="fl">0.7</span><span class="op">)</span>, <span class="fl">1</span>, <span class="fl">7</span><span class="op">)</span>,
           colWidth <span class="op">=</span> <span class="cn">Inf</span>,
           <span class="va">...</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>myXStringSet</dt>
<dd><p>A <code>XStringSet</code> object of sequences.</p></dd>
<dt>htmlFile</dt>
<dd><p>Character string giving the location where the html file should be written.</p></dd>
<dt>openURL</dt>
<dd><p>Logical indicating whether the <code>htmlFile</code> should be opened in a web browser.</p></dd>
<dt>colorPatterns</dt>
<dd><p>Logical specifying whether to color matched <code>patterns</code>, or an integer vector providing pairs of start and stop boundaries for coloring.</p></dd>
<dt>highlight</dt>
<dd><p>Numeric specifying which sequence in the set to use for comparison or <code>NA</code> to color all sequences (default).  If <code>highlight</code> is <code>0</code> then positions differing from the consensus sequence are highlighted.</p></dd>
<dt>patterns</dt>
<dd><p>Either an <code>AAStringSet</code>, <code>DNAStringSet</code>, or <code>RNAStringSet</code> object, a character vector containing regular expressions, a list of numeric matrices, or <code>NULL</code>.  (See details section below.)</p></dd>
<dt>colors</dt>
<dd><p>Character vector providing the color for each of the matched <code>patterns</code>.  Typically a character vector with elements of 7 characters:  ``#'' followed by the red, blue, green values in hexadecimal (after rescaling to 0 ... 255).  Ignored when <code>patterns</code> is a list of matrices.</p></dd>
<dt>colWidth</dt>
<dd><p>Integer giving the maximum number of nucleotides wide the display can be before starting a new page.  Must be a multiple of <code>20</code> (e.g., <code>100</code>), or <code>Inf</code> (the default) to display all the sequences in one set of rows.</p></dd>
<dt>...</dt>
<dd><p>Additional arguments to adjust the appearance of the consensus sequence at the base of the display.  Passed directly to <code>ConsensusSequence</code> for an <code>AAStringSet</code>, <code>DNAStringSet</code>, or <code>RNAStringSet</code>, or to <code>consensusString</code> for a <code>BStringSet</code>.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p><code>BrowseSeqs</code> converts an <code>XStringSet</code> into html format for viewing in a web browser.  The sequences are colored in accordance with the <code>patterns</code> that are provided, or left uncolored if <code>colorPatterns</code> is <code>FALSE</code> or <code>patterns</code> is <code>NULL</code>.  Character or <code>XStringSet</code> patterns are matched as regular expressions and colored according to <code>colors</code>.  If <code>patterns</code> is a list of matrices, then it must contain one element per sequence.  Each matrix is interpreted as providing the fraction red, blue, and green for each letter in the sequence.  Thus, <code>colors</code> is ignored when <code>patterns</code> is a list.  (See examples section below.)</p>
<p>Patterns are not matched across column breaks, so multi-character <code>patterns</code> should be carefully considered when <code>colWidth</code> is less than the maximum sequence length.  Patterns are matched sequentially in the order provided, so it is feasible to use nested <code>patterns</code> such as <code>c("ACCTG", "CC")</code>.  In this case the ``CC'' could be colored differently inside the previously colored ``ACCTG''.  Note that <code>patterns</code> overlapping the boundaries of a previously matched pattern will not be matched.  For example, ``ACCTG'' would not be matched if <code>patterns=c("CC", "ACCTG")</code>.</p>
<p>Some web browsers cannot quickly display a large amount colored text, so it is recommended to use <code>colorPatterns = FALSE</code> or to <code>highlight</code> a sequence when viewing a large <code>XStringSet</code>.  Highlighting will only show all of the characters in the highlighted sequence, and convert all matching positions in the other sequences into dots without <code>color</code>.  Also, note that some web browsers display small shifts between fixed-width characters that may become noticeable as color offsets between the ends of long sequences.</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Creates an html file containing sequence data and (if <code>openURL</code> is <code>TRUE</code>) opens it in a web browser for viewing.  The layout has the sequence name on the left, position legend on the top, cumulative number of nucleotides on the right, and consensus sequence on the bottom.
Returns <code>htmlFile</code> if the html file was written successfully.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>ES Wright (2016) "Using DECIPHER v2.0 to Analyze Big Biological Sequence Data in R". The R Journal, <b>8(1)</b>, 352-359.
Kunzmann P., et al. (2020) "Substitution matrix based color schemes for sequence alignment visualization". BMC Bioinformatics, <b>21(1):209</b>.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Erik Wright <a href="mailto:eswright@pitt.edu">eswright@pitt.edu</a></p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>Some web browsers do not display colored characters with equal widths.  If positions do not align across sequences then try opening the <code>htmlFile</code> with a different web browser.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="BrowseDB.html">BrowseDB</a></code>, <code>ConsensusSequence</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co"># load the example DNA sequences</span></span>
<span class="r-in"><span class="va">db</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Bacteria_175seqs.sqlite"</span>, package<span class="op">=</span><span class="st">"DECIPHER"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">dna</span> <span class="op">&lt;-</span> <span class="fu"><a href="SearchDB.html">SearchDB</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span> <span class="co"># non-coding ribosomal RNA gene sequences</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Search Expression:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> select row_names, sequence from _Seqs where row_names in (select row_names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> from Seqs)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet of length: 175</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.01 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># example of using the defaults with DNA sequences</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">dna</span><span class="op">)</span> <span class="co"># view the XStringSet</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># color only "ACTG" and "CSC" patterns (where S is C or G)</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">dna</span>, patterns<span class="op">=</span><span class="fu">DNAStringSet</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ACTG"</span>, <span class="st">"CSC"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># highlight (i.e., only fully-color) the first sequence</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">dna</span>, highlight<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="co"># other sequences are dots where matching</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># highlight the consensus sequence at the bottom</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">dna</span>, highlight<span class="op">=</span><span class="fl">0</span><span class="op">)</span> <span class="co"># other sequences are dots where matching</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># split the wide view into multiple vertical pages (for printing)</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">dna</span>, colWidth<span class="op">=</span><span class="fl">100</span>, highlight<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># specify an alternative color scheme for -, A, C, G, T</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">dna</span>, colors<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#1E90FF"</span>, <span class="st">"#32CD32"</span>, <span class="st">"#9400D3"</span>, <span class="st">"black"</span>, <span class="st">"#EE3300"</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># only color the positions within certain positional ranges (100-200 &amp; 250-500)</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">dna</span>, colorPatterns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">200</span>, <span class="fl">250</span>, <span class="fl">500</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># example of calling attention to letters by coloring gaps black</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">dna</span>, patterns<span class="op">=</span><span class="st">"-"</span>, colors<span class="op">=</span><span class="st">"black"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># color according to base-pairing by supplying the fraction RGB for every position</span></span>
<span class="r-in"><span class="va">dbn</span> <span class="op">&lt;-</span> <span class="fu">PredictDBN</span><span class="op">(</span><span class="va">dna</span>, type<span class="op">=</span><span class="st">"structures"</span><span class="op">)</span> <span class="co"># calculate the secondary structures</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on alignment:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.11 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining sequence weights:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.03 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Computing Free Energies:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 14.06 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Predicting RNA Secondary Structures:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 3.14 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="co"># dbn now contains the scores for whether a base is paired (left/right) or unpaired</span></span>
<span class="r-in"><span class="va">dbn</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span> <span class="co"># the scores for the first position in the first sequence</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            .            (            ) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9.999881e-01 1.185217e-05 0.000000e+00 </span>
<span class="r-in"><span class="va">dbn</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="fl">10</span><span class="op">]</span> <span class="co"># the scores for the tenth position in the second sequence</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            .            (            ) </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9.903051e-01 9.694895e-03 1.531456e-12 </span>
<span class="r-in"><span class="co"># these positional scores can be used as shades of red, green, and blue:</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">dna</span>, patterns<span class="op">=</span><span class="va">dbn</span><span class="op">)</span> <span class="co"># red = unpaired, green = left-pairing, blue = right</span></span>
<span class="r-in"><span class="co"># positions in black are not part of the consensus secondary structure</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># color all restriction sites</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">RESTRICTION_ENZYMES</span><span class="op">)</span> <span class="co"># load dataset containing restriction enzyme sequences</span></span>
<span class="r-in"><span class="va">sites</span> <span class="op">&lt;-</span> <span class="va">RESTRICTION_ENZYMES</span></span>
<span class="r-in"><span class="va">sites</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"[^A-Z]"</span>, <span class="st">""</span>, <span class="va">sites</span><span class="op">)</span> <span class="co"># remove non-letters</span></span>
<span class="r-in"><span class="va">sites</span> <span class="op">&lt;-</span> <span class="fu">DNAStringSet</span><span class="op">(</span><span class="va">sites</span><span class="op">)</span> <span class="co"># convert the character vector to a DNAStringSet</span></span>
<span class="r-in"><span class="va">rc_sites</span> <span class="op">&lt;-</span> <span class="fu">reverseComplement</span><span class="op">(</span><span class="fu">DNAStringSet</span><span class="op">(</span><span class="va">sites</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"><span class="va">w</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">sites</span> <span class="op">!=</span> <span class="va">rc_sites</span><span class="op">)</span> <span class="co"># find non-palindromic restriction sites</span></span>
<span class="r-in"><span class="va">sites</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">sites</span>, <span class="va">rc_sites</span><span class="op">[</span><span class="va">w</span><span class="op">]</span><span class="op">)</span> <span class="co"># append their reverse complement</span></span>
<span class="r-in"><span class="va">sites</span> <span class="op">&lt;-</span> <span class="va">sites</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nchar</a></span><span class="op">(</span><span class="va">sites</span><span class="op">)</span><span class="op">)</span><span class="op">]</span> <span class="co"># match shorter sites first</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">dna</span>, patterns<span class="op">=</span><span class="va">sites</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># color bases by quality score</span></span>
<span class="r-in"><span class="va">fastq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"s_1_sequence.txt"</span>, package<span class="op">=</span><span class="st">"Biostrings"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">reads</span> <span class="op">&lt;-</span> <span class="fu">readQualityScaledDNAStringSet</span><span class="op">(</span><span class="va">fastq</span>, quality.scoring<span class="op">=</span><span class="st">"solexa"</span><span class="op">)</span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>metadata columns on input DNAStringSet object were dropped</span>
<span class="r-in"><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html" class="external-link">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"yellow"</span>, <span class="st">"green"</span><span class="op">)</span><span class="op">)</span><span class="op">(</span><span class="fl">42</span><span class="op">)</span></span>
<span class="r-in"><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/col2rgb.html" class="external-link">col2rgb</a></span><span class="op">(</span><span class="va">colors</span><span class="op">)</span><span class="op">/</span><span class="fl">255</span></span>
<span class="r-in"><span class="va">quals</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="fu">quality</span><span class="op">(</span><span class="va">reads</span><span class="op">)</span>, <span class="st">"IntegerList"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">quals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">quals</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">colors</span><span class="op">[</span>, <span class="va">x</span><span class="op">]</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="fu">DNAStringSet</span><span class="op">(</span><span class="va">reads</span><span class="op">)</span>, patterns<span class="op">=</span><span class="va">quals</span><span class="op">)</span> <span class="co"># green = high quality, red = low quality</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># load the example protein coding sequences</span></span>
<span class="r-in"><span class="va">fas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"50S_ribosomal_protein_L2.fas"</span>, package<span class="op">=</span><span class="st">"DECIPHER"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">dna</span> <span class="op">&lt;-</span> <span class="fu">readDNAStringSet</span><span class="op">(</span><span class="va">fas</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># example of using the defaults with amino acid sequences</span></span>
<span class="r-in"><span class="va">aa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu">translate</span><span class="op">(</span><span class="va">dna</span><span class="op">)</span><span class="op">)</span> <span class="co"># the unique amino acid sequences</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">aa</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># example of highlighting the consensus amino acid sequence</span></span>
<span class="r-in"><span class="va">AA</span> <span class="op">&lt;-</span> <span class="fu"><a href="AlignSeqs.html">AlignSeqs</a></span><span class="op">(</span><span class="va">aa</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on shared 5-mers:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.09 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Clustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.03 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Aligning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.87 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iteration 1 of 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on alignment:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.02 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reclustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.03 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Realigning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.55 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iteration 2 of 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on alignment:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.02 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reclustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.03 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Realigning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.2 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">AA</span>, highlight<span class="op">=</span><span class="fl">0</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># example of highlighting positions that differ from the majority consensus</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">AA</span>, highlight<span class="op">=</span><span class="fl">0</span>, threshold<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># specify an alternative color scheme for amino acids (from Kunzmann et al.)</span></span>
<span class="r-in"><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>`-`<span class="op">=</span><span class="st">"#000000"</span>, `A`<span class="op">=</span><span class="st">"#BDB1E8"</span>, `R`<span class="op">=</span><span class="st">"#EFA2C5"</span>, `N`<span class="op">=</span><span class="st">"#F6602F"</span>,</span>
<span class="r-in">    `D`<span class="op">=</span><span class="st">"#FD5559"</span>, `C`<span class="op">=</span><span class="st">"#12C7FE"</span>, `Q`<span class="op">=</span><span class="st">"#DDACB4"</span>, `E`<span class="op">=</span><span class="st">"#FEA097"</span>, `G`<span class="op">=</span><span class="st">"#F46802"</span>,</span>
<span class="r-in">    `H`<span class="op">=</span><span class="st">"#FCA708"</span>, `I`<span class="op">=</span><span class="st">"#369BD9"</span>, `L`<span class="op">=</span><span class="st">"#2E95EC"</span>, `K`<span class="op">=</span><span class="st">"#CF7690"</span>, `M`<span class="op">=</span><span class="st">"#4B8EFE"</span>,</span>
<span class="r-in">    `F`<span class="op">=</span><span class="st">"#76997D"</span>, `P`<span class="op">=</span><span class="st">"#FD2AE3"</span>, `S`<span class="op">=</span><span class="st">"#A08A9A"</span>, `T`<span class="op">=</span><span class="st">"#9A84D5"</span>, `W`<span class="op">=</span><span class="st">"#74C80D"</span>,</span>
<span class="r-in">    `Y`<span class="op">=</span><span class="st">"#9BB896"</span>, `V`<span class="op">=</span><span class="st">"#89B9F9"</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">AA</span>, colors<span class="op">=</span><span class="va">colors</span>, patterns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">colors</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># example of coloring in a reduced amino acid alphabet</span></span>
<span class="r-in"><span class="va">alpha</span> <span class="op">&lt;-</span> <span class="va">AA_REDUCED</span><span class="op">[[</span><span class="fl">15</span><span class="op">]</span><span class="op">]</span></span>
<span class="r-in"><span class="va">alpha</span> <span class="co"># clustering of amino acids based on similarity</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "ACGPSTWY" "DEHKNQR"  "FILMV"   </span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">AA</span>, patterns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"-"</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"["</span>, <span class="va">alpha</span>, <span class="st">"]"</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># color amino acids according to their predicted secondary structure</span></span>
<span class="r-in"><span class="va">hec</span> <span class="op">&lt;-</span> <span class="fu">PredictHEC</span><span class="op">(</span><span class="va">AA</span>, type<span class="op">=</span><span class="st">"probabilities"</span><span class="op">)</span> <span class="co"># calculate the secondary structures</span></span>
<span class="r-in"><span class="co"># hec now contains the probability that a base is in an alpha-helix or beta-sheet</span></span>
<span class="r-in"><span class="va">hec</span><span class="op">[[</span><span class="fl">3</span><span class="op">]</span><span class="op">]</span><span class="op">[</span>, <span class="fl">18</span><span class="op">]</span> <span class="co"># the 18th position in sequence 3 is likely part of a beta-sheet (E)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          H          E          C </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 0.10249191 0.80619493 0.09131316 </span>
<span class="r-in"><span class="co"># the positional probabilities can be used as shades of red, green, and blue:</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">AA</span>, patterns<span class="op">=</span><span class="va">hec</span><span class="op">)</span> <span class="co"># red = alpha-helix, green = beta-sheet, blue = coil</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># color codons according to their corresponding amino acid</span></span>
<span class="r-in"><span class="va">DNA</span> <span class="op">&lt;-</span> <span class="fu"><a href="AlignTranslation.html">AlignTranslation</a></span><span class="op">(</span><span class="va">dna</span><span class="op">)</span> <span class="co"># align the translation then reverse translate</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on shared 5-mers:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.78 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Clustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.06 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Aligning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 1.9 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iteration 1 of 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on alignment:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.13 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reclustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.09 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Realigning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.85 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iteration 2 of 2:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Determining distance matrix based on alignment:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.13 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reclustering into groups by similarity:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.1 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Realigning Sequences:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.11 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">rainbow</a></span><span class="op">(</span><span class="fl">21</span>, v<span class="op">=</span><span class="fl">0.8</span>, start<span class="op">=</span><span class="fl">0.9</span>, end<span class="op">=</span><span class="fl">0.7</span><span class="op">)</span> <span class="co"># codon colors</span></span>
<span class="r-in"><span class="va">m</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">GENETIC_CODE</span>, <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">GENETIC_CODE</span><span class="op">)</span><span class="op">)</span> <span class="co"># corresponding amino acid</span></span>
<span class="r-in"><span class="va">codonBounds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu">width</span><span class="op">(</span><span class="va">DNA</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="fl">3</span><span class="op">)</span>, <span class="co"># start of codons</span></span>
<span class="r-in">  <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fu">width</span><span class="op">(</span><span class="va">DNA</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>, <span class="co"># end of codons</span></span>
<span class="r-in">  nrow<span class="op">=</span><span class="fl">2</span>,</span>
<span class="r-in">  byrow<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">BrowseSeqs</span><span class="op">(</span><span class="va">DNA</span>,</span>
<span class="r-in">  colorPatterns<span class="op">=</span><span class="va">codonBounds</span>,</span>
<span class="r-in">  patterns<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"---"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">GENETIC_CODE</span><span class="op">)</span><span class="op">)</span>, <span class="co"># codons to color</span></span>
<span class="r-in">  colors<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substring</a></span><span class="op">(</span><span class="va">colors</span><span class="op">[</span><span class="va">m</span><span class="op">]</span>, <span class="fl">1</span>, <span class="fl">7</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>

