<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Finds syntenic blocks between groups of sequences in a database."><title>Finds Synteny in a Sequence Database — FindSynteny • CompGenomicsBioc2022</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Finds Synteny in a Sequence Database — FindSynteny"><meta property="og:description" content="Finds syntenic blocks between groups of sequences in a database."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">Coevolutionary Networks</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com" aria-label="My Website">
    <span class="fa fa-potato"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Finds Synteny in a Sequence Database</h1>
      
      <div class="d-none name"><code>FindSynteny.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Finds syntenic blocks between groups of sequences in a database.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">FindSynteny</span><span class="op">(</span><span class="va">dbFile</span>,
            tblName <span class="op">=</span> <span class="st">"Seqs"</span>,
            identifier <span class="op">=</span> <span class="st">""</span>,
            useFrames <span class="op">=</span> <span class="cn">TRUE</span>,
            alphabet <span class="op">=</span> <span class="va">AA_REDUCED</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
            geneticCode <span class="op">=</span> <span class="va">GENETIC_CODE</span>,
            sepCost <span class="op">=</span> <span class="fl">0</span>,
            gapCost <span class="op">=</span> <span class="op">-</span><span class="fl">0.01</span>,
            shiftCost <span class="op">=</span> <span class="fl">0</span>,
            codingCost <span class="op">=</span> <span class="fl">0</span>,
            maxSep <span class="op">=</span> <span class="fl">2000</span>,
            maxGap <span class="op">=</span> <span class="fl">5000</span>,
            minScore <span class="op">=</span> <span class="fl">30</span>,
            dropScore <span class="op">=</span> <span class="op">-</span><span class="fl">100</span>,
            maskRepeats <span class="op">=</span> <span class="cn">TRUE</span>,
            allowOverlap <span class="op">=</span> <span class="cn">TRUE</span>,
            storage <span class="op">=</span> <span class="fl">0.5</span>,
            processors <span class="op">=</span> <span class="fl">1</span>,
            verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>dbFile</dt>
<dd><p>A SQLite connection object or a character string specifying the path to the database file.</p></dd>
<dt>tblName</dt>
<dd><p>Character string specifying the table where the sequences are located.</p></dd>
<dt>identifier</dt>
<dd><p>Optional character string used to narrow the search results to those matching a specific identifier.  If <code>""</code> then all identifiers are selected.  Repeated identifiers will find synteny between a sequence and itself, while blocking identical positions from matching in both sequences.</p></dd>
<dt>useFrames</dt>
<dd><p>Logical specifying whether to use 6-frame amino acid translations to help find more distant hits.  Using the <code>alphabet</code> is helpful when the genome is largely composed of coding DNA.  If <code>FALSE</code> then faster but less sensitive to distant homology.</p></dd>
<dt>alphabet</dt>
<dd><p>Character vector of amino acid groupings used to reduce the 20 standard amino acids into smaller groups.  Alphabet reduction helps to find more distant homologies between sequences.  A non-reduced amino acid alphabet can be used by setting <code>alphabet</code> equal to <code>AA_STANDARD</code>.</p></dd>
<dt>geneticCode</dt>
<dd><p>Either a character vector giving the genetic code to use in translation, or a list containing one genetic code for each identifier.  If a list is provided then it must be named by the corresponding identifiers in the database.</p></dd>
<dt>sepCost</dt>
<dd><p>Cost per nucleotide separation between hits to apply when chaining hits into blocks.</p></dd>
<dt>gapCost</dt>
<dd><p>Cost for gaps between hits to apply when chaining hits into blocks.</p></dd>
<dt>shiftCost</dt>
<dd><p>Cost for shifting between different reading frames when chaining reduced amino acid hits into blocks.</p></dd>
<dt>codingCost</dt>
<dd><p>Cost for switching between coding and non-coding hits when chaining hits into blocks.</p></dd>
<dt>maxSep</dt>
<dd><p>Maximal separation (in nucleotides) between hits in the same block.</p></dd>
<dt>maxGap</dt>
<dd><p>The maximum number of gaps between hits in the same block.</p></dd>
<dt>minScore</dt>
<dd><p>The minimum score required for a chain of hits to become a block.  Higher values of <code>minScore</code> are less likely to yield false positives.</p></dd>
<dt>dropScore</dt>
<dd><p>The change from maximal score required to stop extending blocks.</p></dd>
<dt>maskRepeats</dt>
<dd><p>Logical specifying whether to ``soft'' mask repeats when searching for hits.</p></dd>
<dt>allowOverlap</dt>
<dd><p>Logical specifying whether to permit blocks to overlap on the same sequence.</p></dd>
<dt>storage</dt>
<dd><p>Excess gigabytes available to store objects so that they do not need to be recomputed in later steps.  This should be a number between zero and a (modest) fraction of the available system memory.  Note that more than <code>storage</code> gigabytes may be required, but will not be stored for later reuse.</p></dd>
<dt>processors</dt>
<dd><p>The number of processors to use, or <code>NULL</code> to automatically detect and use all available processors.</p></dd>
<dt>verbose</dt>
<dd><p>Logical indicating whether to display progress.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Long nucleotide sequences, such as genomes, are often not collinear or may be composed of many smaller segments (e.g., contigs).  <code>FindSynteny</code> searches for ``hits'' between sequences that can be chained into collinear ``blocks'' of synteny.  Hits are defined as k-mer exact nucleotide matches or k-mer matches in a reduced amino acid alphabet (if <code>useFrames</code> is <code>TRUE</code>).  Hits are chained into blocks as long as they are:  (1) within the same sequence, (2) within <code>maxSep</code> and <code>maxGap</code> distance, and (3) help maintain the score above <code>minScore</code>.  Blocks are extended from their first and last hit until their score drops below <code>dropScore</code> from the maximum that was reached.  This process results in a set of hits and blocks stored in an object of class ``Synteny''.</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>An object of class ``Synteny''.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Erik Wright <a href="mailto:eswright@pitt.edu">eswright@pitt.edu</a></p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p><code>FindSynteny</code> is intended to be used on sets of sequences with up to ~100 million nucleotides total per identifier.  For this reason, better performance can sometimes be achieved by assigning a unique identifier to each chromosome belonging to a large genome.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="AlignSynteny.html">AlignSynteny</a></code>, <code><a href="Synteny-class.html">Synteny-class</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">db</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Influenza.sqlite"</span>, package<span class="op">=</span><span class="st">"DECIPHER"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">synteny</span> <span class="op">&lt;-</span> <span class="fu">FindSynteny</span><span class="op">(</span><span class="va">db</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 1.36 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="va">synteny</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          H9N2     H5N1     H2N2     H7N9     H1N1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> H9N2   8 seqs 74% hits 78% hits 76% hits 73% hits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> H5N1 8 blocks   8 seqs 70% hits 74% hits 84% hits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> H2N2 8 blocks 8 blocks   8 seqs 73% hits 71% hits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> H7N9 8 blocks 8 blocks 8 blocks   8 seqs 74% hits</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> H1N1 8 blocks 8 blocks 8 blocks 8 blocks   8 seqs</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/graphics/pairs.html" class="external-link">pairs</a></span><span class="op">(</span><span class="va">synteny</span><span class="op">)</span> <span class="co"># scatterplot matrix</span></span>
<span class="r-plt img"><img src="FindSynteny-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>

