<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Predicts the start and stop positions of protein coding genes in a genome."><title>Find Genes in a Genome — FindGenes • CompGenomicsBioc2022</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Find Genes in a Genome — FindGenes"><meta property="og:description" content="Predicts the start and stop positions of protein coding genes in a genome."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">Coevolutionary Networks</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com" aria-label="My Website">
    <span class="fa fa-potato"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Find Genes in a Genome</h1>
      
      <div class="d-none name"><code>FindGenes.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Predicts the start and stop positions of protein coding genes in a genome.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">FindGenes</span><span class="op">(</span><span class="va">myDNAStringSet</span>,
          geneticCode <span class="op">=</span> <span class="fu">getGeneticCode</span><span class="op">(</span><span class="st">"11"</span><span class="op">)</span>,
          minGeneLength <span class="op">=</span> <span class="fl">60</span>,
          includeGenes <span class="op">=</span> <span class="cn">NULL</span>,
          allowEdges <span class="op">=</span> <span class="cn">TRUE</span>,
          allScores <span class="op">=</span> <span class="cn">FALSE</span>,
          showPlot <span class="op">=</span> <span class="cn">FALSE</span>,
          processors <span class="op">=</span> <span class="fl">1</span>,
          verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>myDNAStringSet</dt>
<dd><p>A <code>DNAStringSet</code> object of unaligned sequences representing a genome.</p></dd>
<dt>geneticCode</dt>
<dd><p>A named character vector defining the translation from codons to amino acids.  Optionally, an <code>"alt_init_codons"</code> attribute can be used to specify alternative initiation codons.  By default, the bacterial and archael genetic code is used, which has seven possible initiation codons: ATG, GTG, TTG, CTG, ATA, ATT, and ATC.</p></dd>
<dt>minGeneLength</dt>
<dd><p>Integer specifying the minimum length of genes to find in the genome.</p></dd>
<dt>includeGenes</dt>
<dd><p>A <code>Genes</code> object to include as potential genes or <code>NULL</code> (the default) to predict all genes <code>de novo</code>.</p></dd>
<dt>allowEdges</dt>
<dd><p>Logical determining whether to allow genes that run off the edge of the sequences.  If <code>TRUE</code> (the default), genes can be identified with implied starts or ends outside the boundaries of <code>myDNAStringSet</code>, although the boundary will be set to the last possible codon position.  This is useful when genome sequences are circular or incomplete.</p></dd>
<dt>allScores</dt>
<dd><p>Logical indicating whether to return information about all possible open reading frames or only the predicted genes (the default).</p></dd>
<dt>showPlot</dt>
<dd><p>Logical determining whether a plot is displayed with the distribution of gene lengths and scores.  (See details section below.)</p></dd>
<dt>processors</dt>
<dd><p>The number of processors to use, or <code>NULL</code> to automatically detect and use all available processors.</p></dd>
<dt>verbose</dt>
<dd><p>Logical indicating whether to print information about the predictions on each iteration.  (See details section below.)</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Protein coding genes are identified by learning their characteristic signature directly from the genome, i.e., <em>ab initio</em> prediction.  Gene signatures are derived from the content of the open reading frame and surrounding signals that indicate the presence of a gene.  Genes are assumed to not contain introns or frame shifts, making the function best suited for prokaryotic genomes.</p>
<p>If <code>showPlot</code> is <code>TRUE</code> then a plot is displayed with four panels.  The upper left panel shows the fitted distribution of background open reading frame lengths.  The upper right panel shows this distribution on top of the fitted distribution of predicted gene lengths.  The lower left panel shows the fitted distribution of scores for the intergenic spacing between genes on the same and opposite genome strands.  The bottom right panel shows the total score of open reading frames and predicted genes by length.</p>
<p>If <code>verbose</code> is <code>TRUE</code>, information is shown about the predictions at each iteration of gene finding.  The mean score difference between genes and non-genes is updated at each iteration, unless it is negative, in which case the score is dropped and a <code>"-"</code> is displayed.  The columns denote the number of iterations (<code>"Iter"</code>), number of codon scoring models (<code>"Models"</code>), start codon scores (<code>"Start"</code>), upstream k-mer motif scores (<code>"Motif"</code>), mRNA folding scores (<code>"Fold"</code>), initial codon bias scores (<code>"Init"</code>), upstream nucleotide bias scores (<code>"UpsNt"</code>), termination codon bias scores (<code>"Term"</code>), ribosome binding site scores (<code>"RBS"</code>), codon autocorrelation scores (<code>"Auto"</code>), stop codon scores (<code>"Stop"</code>), and number of predicted genes (<code>"Genes"</code>).</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>An object of class <code>Genes</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Erik Wright <a href="mailto:eswright@pitt.edu">eswright@pitt.edu</a></p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code>ExtractGenes</code>, <code><a href="FindNonCoding.html">FindNonCoding</a></code>, <code>Genes-class</code>, <code>WriteGenes</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co"># import a test genome</span></span>
<span class="r-in"><span class="va">fas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span>
<span class="r-in">  <span class="st">"Chlamydia_trachomatis_NC_000117.fas.gz"</span>,</span>
<span class="r-in">  package<span class="op">=</span><span class="st">"DECIPHER"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">genome</span> <span class="op">&lt;-</span> <span class="fu">readDNAStringSet</span><span class="op">(</span><span class="va">fas</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">z</span> <span class="op">&lt;-</span> <span class="fu">FindGenes</span><span class="op">(</span><span class="va">genome</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iter  Models Start Motif  Init  Fold UpsNt  Term   RBS  Auto  Stop Genes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1          1 18.40                 _                                 8861          1 18.40                 \                                 8861          1 18.40                 |                                 8861          1 18.40                 /                                 8861          1 18.40                 _                                 8861          1 18.40                 \                                 8861          1 18.40                 |                                 8861          1 18.40                 /                                 8861          1 18.40                 _                                 8862         12 18.40  0.48  0.90                                       8883         15 18.38  0.53  1.16  0.40  1.41                           8914         14 15.59  0.52  1.27  0.46  1.76  0.24                     8945         10 15.62  0.52  1.25  0.49  1.84  0.25  0.98  0.10  0.07   8976         11 15.61  0.51  1.24  0.50  1.92  0.25  1.04  0.10  0.07   8977         13 15.59  0.51  1.25  0.51  1.94  0.25  1.05  0.10  0.07   8988         12 15.59  0.51  1.25  0.50  1.94  0.26  1.03  0.10  0.07   8979         12 15.59  0.51  1.25  0.51  1.95  0.26  1.08  0.10  0.07   89710        12 15.58  0.51  1.25  0.49  1.96  0.25  1.03  0.10  0.07   89711        12 15.57  0.51  1.25  0.50  1.96  0.26  1.04  0.10  0.07   89812        13 15.57  0.52  1.25  0.52  1.97  0.26  1.06  0.10  0.07   89713        13 15.57  0.52  1.25  0.51  1.98  0.25  1.08  0.10  0.07   89914        13 15.57  0.52  1.25  0.51  1.98  0.25  1.08  0.10  0.07   897</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 31.79 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="va">z</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Genes object of size 897 specifying:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 897 protein coding genes from 66 to 5,361 nucleotides.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Index Strand Begin  End TotalScore ... Gene</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1     1      0     1 1176      77.84 ...    1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2     1      1  1321 1593      21.00 ...    1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3     1      0  1794 2096      34.43 ...    1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4     1      0  2108 3583     139.23 ...    1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5     1      0  3585 5051     124.83 ...    1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6     1      1  5150 6241      78.66 ...    1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ... with 891 more rows.</span>
<span class="r-in"><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu">ExtractGenes</span><span class="op">(</span><span class="va">z</span>, <span class="va">genome</span><span class="op">)</span></span>
<span class="r-in"><span class="va">genes</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet object of length 897:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       width seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]  1176 GCGGCCGCCCGGGAAATTGCTAAAAGATGGGAG...GGCGTTGGAATAGAGAAGTCGATAGGGAATAA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]   273 ATGCTTTGTAAAGTTTGTAGAGGATTATCTTCT...GATACCACCACCATCATATGGATAGAGAATAA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]   303 ATGACAGAGTCATATGTAAACAAAGAAGAAATC...GATTAGTCAAAGTCCCTACAGTTATCAAATAG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]  1476 ATGTATCGTAAGAGTGCTTTAGAATTAAGAGAT...TGAATGGACTTTTTGACGGAGGAATAGAATAA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]  1467 ATGGGCATAGCACATACTGAATGGGAGTCTGTG...AATTGCTATTAGCAGCTATGCGAGATATGTAA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ...   ... ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [893]  3051 ATGCCTTTTTCTTTGAGATCTACATCATTTTGT...TTGTATCCATGGGCTTGAATAGAATCTTTTAA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [894]   303 ATGCTGGCAACAATTAAAAAAATTACTGTGTTG...GAGATTCTCGCCTAGAATGCAAGAAGATATAA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [895]  2637 ATGCGACCTGATCATATGAACTTCTGTTGTCTA...CCCTGGATCTGGGGACCACTTACAGGTTCTAG</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [896]    96 ATGTCAAAAAAAAGTAATAATTTACAGACTTTT...ATGAAGAGTTAAGGAAGATTTTTGGATTGTGA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [897]   600 ATGAGCATCAGGGGAGTAGGAGGCAACGGGAAT...AAGCGAACAAGTTGGCTGGCCCTGACGGGGTA</span>
<span class="r-in"><span class="va">proteins</span> <span class="op">&lt;-</span> <span class="fu">ExtractGenes</span><span class="op">(</span><span class="va">z</span>, <span class="va">genome</span>, type<span class="op">=</span><span class="st">"AAStringSet"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">proteins</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> AAStringSet object of length 897:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       width seq</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [1]   392 AAAREIAKRWEQRVRDLQDKGAARKLLNDPLGR...QVEGILRDMLTNGSQTFRDLMRRWNREVDRE*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [2]    91 MLCKVCRGLSSLIVVLGAINTGILGVTGYKVNL...CLNFLKCCFKKRHGDCCSSKGGYHHHHMDRE*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [3]   101 MTESYVNKEEIISLAKNAALELEDAHVEEFVTS...DMVTSDFTQEEFLSNVPVSLGGLVKVPTVIK*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [4]   492 MYRKSALELRDAVVNRELSVTAITEYFYHRIES...ICQVGYSFQEHSQIKQLYPKAVNGLFDGGIE*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [5]   489 MGIAHTEWESVIGLEVHVELNTESKLFSPARNH...GFLVGQIMKRTEGKAPPKRVNELLLAAMRDM*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ...   ... ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [893]  1017 MPFSLRSTSFCFLACLCSYSYGFASSPQVLTPN...HHFGRAYMNYSLDARRRQTAHFVSMGLNRIF*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [894]   101 MLATIKKITVLLLSKRKAGIRIDYCALALDAVE...LDASLESAQVRLAGLMLDYWDGDSRLECKKI*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [895]   879 MRPDHMNFCCLCAAILSSTAVLFGQDPLGETAL...LHRLQTLLNVSYVLRGQSHSYSLDLGTTYRF*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [896]    32 MSKKSNNLQTFSSRALFHVFQDEELRKIFGL*</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [897]   200 MSIRGVGGNGNSRIPSHNGDGSNRRSQNTKGNN...NLDVNEARLMAAYTSECADHLEANKLAGPDGV</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>

