<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Groups the sequences into approximate clusters of similarity."><title>Cluster Sequences By Similarity — IdClusters • CompGenomicsBioc2022</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Cluster Sequences By Similarity — IdClusters"><meta property="og:description" content="Groups the sequences into approximate clusters of similarity."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">Coevolutionary Networks</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com" aria-label="My Website">
    <span class="fa fa-potato"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Cluster Sequences By Similarity</h1>
      
      <div class="d-none name"><code>IdClusters.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Groups the sequences into approximate clusters of similarity.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">IdClusters</span><span class="op">(</span><span class="va">myXStringSet</span>,
           cutoff <span class="op">=</span> <span class="fl">0</span>,
           processors <span class="op">=</span> <span class="fl">1</span>,
           verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>myXStringSet</dt>
<dd><p>If <code>method</code> is <code>"ML"</code> or <code>reconstruct</code> is not <code>FALSE</code>, the <code>DNAStringSet</code> or <code>RNAStringSet</code> used in the creation of <code>myDistMatrix</code>.  If <code>method</code> is <code>"inexact"</code>, the <code>DNAStringSet</code>, <code>RNAStringSet</code>, or <code>AAStringSet</code> to cluster.</p></dd>
<dt>cutoff</dt>
<dd><p>A vector with the maximum edge length separating the sequences in the same cluster.  Multiple cutoffs may be provided in ascending or descending order.  (See details section below.)</p></dd>
<dt>processors</dt>
<dd><p>The number of processors to use, or <code>NULL</code> to automatically detect and use all available processors.</p></dd>
<dt>verbose</dt>
<dd><p>Logical indicating whether to display progress.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p><code>IdClusters</code> groups the input sequences into approximate clusters using a heuristic algorithm.  First the sequences are ordered by length and the longest sequence becomes the first cluster seed.  If the second sequence is less than <code>cutoff</code> k-mer distance then it is added to the cluster, otherwise it becomes a new cluster representative.  The remaining sequences are matched to cluster representatives in a similar fashion until all sequences belong to a cluster.  In the majority of cases, this process results in clusters with members separated by less than <code>cutoff</code> distance.</p>
<p>Multiple cutoffs may be provided in sorted order.  If the <code>cutoff</code>s are provided in <em>descending</em> order then clustering at each new value of <code>cutoff</code> is continued within the prior <code>cutoff</code>'s clusters.  In this way clusters at lower values of <code>cutoff</code> are completely contained within their umbrella clusters at higher values of <code>cutoff</code>.  This is useful for accelerating the clustering process, because each subsequent group is only clustered within the previous group.  If multiple cutoffs are provided in <em>ascending</em> order then clustering at each level of <code>cutoff</code> is independent of the prior level.</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A data.frame is returned with dimensions \(N*M\), where each one of \(N\) sequences is assigned to a cluster at the \(M\)-level of cutoff.  The row.names of the data.frame correspond to the <em>names</em> of <code>myXStingSet</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Erik Wright <a href="mailto:eswright@pitt.edu">eswright@pitt.edu</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">fas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"50S_ribosomal_protein_L2.fas"</span>, package<span class="op">=</span><span class="st">"DECIPHER"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">dna</span> <span class="op">&lt;-</span> <span class="fu">readDNAStringSet</span><span class="op">(</span><span class="va">fas</span><span class="op">)</span></span>
<span class="r-in"><span class="va">aa</span> <span class="op">&lt;-</span> <span class="fu">translate</span><span class="op">(</span><span class="va">dna</span><span class="op">)</span></span>
<span class="r-in"><span class="va">inexact</span> <span class="op">&lt;-</span> <span class="fu">IdClusters</span><span class="op">(</span><span class="va">aa</span>, cutoff<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.7</span>, <span class="fl">0.1</span>, <span class="op">-</span><span class="fl">0.1</span><span class="op">)</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.07 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">inexact</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                     cluster_7 cluster_6 cluster_5 cluster_4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Rickettsia prowazekii str. Dachau           8        11        23        21</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Porphyromonas gingivalis W83               12         5         3         6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Porphyromonas gingivalis TDC60             12         5         3         6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Porphyromonas gingivalis ATCC 33277        12         5         3         6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Pasteurella multocida 671/90               13         4         0         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Pasteurella multocida 36950                13         4         0         2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                                     cluster_3 cluster_2 cluster_1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Rickettsia prowazekii str. Dachau          27        24        30</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Porphyromonas gingivalis W83                3         7         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Porphyromonas gingivalis TDC60              3         7         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Porphyromonas gingivalis ATCC 33277         3         7         3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Pasteurella multocida 671/90               15         2        94</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Pasteurella multocida 36950                15         2        94</span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">inexact</span>, <span class="fl">2</span>, <span class="va">max</span><span class="op">)</span> <span class="co"># number of clusters per cutoff</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> cluster_7 cluster_6 cluster_5 cluster_4 cluster_3 cluster_2 cluster_1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        15        33        53        71        85        94        98 </span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>

