<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference â€¢ CompGenomicsBioc2022</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Coevolution%20Networks.html">Coevolutionary Networks</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com" aria-label="My Website">
    <span class="fa fa-potato"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="loading-genome-data">Loading Genome Data<a class="anchor" aria-label="anchor" href="#loading-genome-data"></a></h2>
      
      <p class="section-desc"></p><p>Functions for loading and visualizing genomic data</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="AlignSeqs.html">AlignSeqs()</a></code> 
        </dt>
        <dd>Align a Set of Unaligned Sequences</dd>
      </dl><dl><dt>
          
          <code><a href="AlignTranslation.html">AlignTranslation()</a></code> 
        </dt>
        <dd>Align Sequences By Their Amino Acid Translation</dd>
      </dl><dl><dt>
          
          <code><a href="BrowseSeqs.html">BrowseSeqs()</a></code> 
        </dt>
        <dd>View Sequences in a Web Browser</dd>
      </dl><dl><dt>
          
          <code><a href="BrowseDB.html">BrowseDB()</a></code> 
        </dt>
        <dd>View a Database Table in a Web Browser</dd>
      </dl><dl><dt>
          
          <code><a href="SearchDB.html">SearchDB()</a></code> 
        </dt>
        <dd>Obtain Specific Sequences from a Database</dd>
      </dl><dl><dt>
          
          <code><a href="Seqs2DB.html">Seqs2DB()</a></code> 
        </dt>
        <dd>Add Sequences from Text File to Database</dd>
      </dl><dl><dt>
          
          <code><a href="DB2Seqs.html">DB2Seqs()</a></code> 
        </dt>
        <dd>Export Database Sequences to a FASTA or FASTQ File</dd>
      </dl></div><div class="section level2">
      <h2 id="gene-calling-and-annotation">Gene Calling and Annotation<a class="anchor" aria-label="anchor" href="#gene-calling-and-annotation"></a></h2>
      
      <p class="section-desc"></p><p>Functions for calling and annotating genetic regions</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="FindGenes.html">FindGenes()</a></code> 
        </dt>
        <dd>Find Genes in a Genome</dd>
      </dl><dl><dt>
          
          <code><a href="FindNonCoding.html">FindNonCoding()</a></code> 
        </dt>
        <dd>Find Non-Coding RNAs in a Genome</dd>
      </dl><dl><dt>
          
          <code><a href="LearnNonCoding.html">LearnNonCoding()</a></code> 
        </dt>
        <dd>Learn a Non-Coding RNA Model</dd>
      </dl><dl><dt>
          
          <code><a href="IdTaxa.html">IdTaxa()</a></code> 
        </dt>
        <dd>Assign Sequences a Taxonomic Classification</dd>
      </dl><dl><dt>
          
          <code><a href="Taxa-class.html">plot(<i>&lt;Taxa&gt;</i>)</a></code> <code><a href="Taxa-class.html">print(<i>&lt;Taxa&gt;</i>)</a></code> <code><a href="Taxa-class.html">`[`(<i>&lt;Taxa&gt;</i>)</a></code> 
        </dt>
        <dd>Taxa training and testing objects</dd>
      </dl><dl><dt>
          
          <code><a href="LearnTaxa.html">LearnTaxa()</a></code> 
        </dt>
        <dd>Train a Classifier for Assigning Taxonomy</dd>
      </dl></div><div class="section level2">
      <h2 id="constructing-cogs">Constructing COGs<a class="anchor" aria-label="anchor" href="#constructing-cogs"></a></h2>
      
      <p class="section-desc"></p><p>Functions to construct COGs from annotated genetic regions</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="DisjointSet.html">DisjointSet()</a></code> 
        </dt>
        <dd>Return single linkage clusters from <code>PairSummaries</code> objects.</dd>
      </dl><dl><dt>
          
          <code><a href="ExtractBy.html">ExtractBy()</a></code> 
        </dt>
        <dd>Extract and organize <code>DNAStringSets</code>s.</dd>
      </dl><dl><dt>
          
          <code><a href="FindSets.html">FindSets()</a></code> 
        </dt>
        <dd>Find all single linkage clusters in an undirected pairs list.</dd>
      </dl><dl><dt>
          
          <code><a href="LinkedPairs-class.html">print(<i>&lt;LinkedPairs&gt;</i>)</a></code> 
        </dt>
        <dd>Tables of where syntenic hits link pairs of genes</dd>
      </dl><dl><dt>
          
          <code><a href="NucleotideOverlap.html">NucleotideOverlap()</a></code> 
        </dt>
        <dd>Tabulating Pairs of Genomic Sequences</dd>
      </dl><dl><dt>
          
          <code><a href="PairSummaries.html">PairSummaries()</a></code> 
        </dt>
        <dd>Summarize connected pairs in a LinkedPairs object</dd>
      </dl><dl><dt>
          
          <code><a href="BlockExpansion.html">BlockExpansion()</a></code> 
        </dt>
        <dd>Attempt to expand blocks of paired features in a <code>PairSummaries</code> object.</dd>
      </dl></div><div class="section level2">
      <h2 id="constructing-phylogenies">Constructing Phylogenies<a class="anchor" aria-label="anchor" href="#constructing-phylogenies"></a></h2>
      
      <p class="section-desc"></p><p>Functions for phylogenetic reconstruction of genomes or COGs</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="Cophenetic.html">Cophenetic()</a></code> 
        </dt>
        <dd>Compute cophenetic distances on dendrogram objects</dd>
      </dl><dl><dt>
          
          <code><a href="DistanceMatrix.html">DistanceMatrix()</a></code> 
        </dt>
        <dd>Calculate the Distances Between Sequences</dd>
      </dl><dl><dt>
          
          <code><a href="IdClusters.html">IdClusters()</a></code> 
        </dt>
        <dd>Cluster Sequences By Similarity</dd>
      </dl><dl><dt>
          
          <code><a href="TreeLine.html">TreeLine()</a></code> 
        </dt>
        <dd>Construct a Phylogenetic Tree</dd>
      </dl><dl><dt>
          
          <code><a href="WriteDendrogram.html">WriteDendrogram()</a></code> 
        </dt>
        <dd>Write a Dendrogram to Newick Format</dd>
      </dl></div><div class="section level2">
      <h2 id="finding-co-evolving-gene-collectives">Finding Co-evolving Gene Collectives<a class="anchor" aria-label="anchor" href="#finding-co-evolving-gene-collectives"></a></h2>
      
      <p class="section-desc"></p><p>Functions for finding co-evolutionary signal between COGs</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="ProtWeaver.html">ProtWeaver()</a></code> 
        </dt>
        <dd>ProtWeaver: Predicting Protein Functional Association Networks</dd>
      </dl><dl><dt>
          
          <code><a href="predict.ProtWeaver.html">predict(<i>&lt;ProtWeaver&gt;</i>)</a></code> 
        </dt>
        <dd>Make predictions with ProtWeaver objects</dd>
      </dl><dl><dt>
          
          <code><a href="plot.ProtWeb.html">plot(<i>&lt;ProtWeb&gt;</i>)</a></code> 
        </dt>
        <dd>Plot predictions in a ProtWeb object</dd>
      </dl><dl><dt>
          
          <code><a href="getProtWebData.ProtWeb.html">GetProtWebData(<i>&lt;ProtWeb&gt;</i>)</a></code> 
        </dt>
        <dd>Extract information from a ProtWeb object</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>

