<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ CompGenomicsBioc2022</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/vapor/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CompGenomicsBioc2022</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/workshop_syllabus.html">Comparative Genomics with SynExtend and DECIPHER</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ahl27/CompGenomicsBioc2022" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="loading-genome-data">Loading Genome Data <a href="#loading-genome-data" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions for loading and visualizing genomic data</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="AlignSeqs.html">AlignSeqs()</a></code> </p>
        </td>
        <td><p>Align a Set of Unaligned Sequences</p></td>
      </tr><tr><td>
          <p><code><a href="AlignTranslation.html">AlignTranslation()</a></code> </p>
        </td>
        <td><p>Align Sequences By Their Amino Acid Translation</p></td>
      </tr><tr><td>
          <p><code><a href="BrowseSeqs.html">BrowseSeqs()</a></code> </p>
        </td>
        <td><p>View Sequences in a Web Browser</p></td>
      </tr><tr><td>
          <p><code><a href="BrowseDB.html">BrowseDB()</a></code> </p>
        </td>
        <td><p>View a Database Table in a Web Browser</p></td>
      </tr><tr><td>
          <p><code><a href="SearchDB.html">SearchDB()</a></code> </p>
        </td>
        <td><p>Obtain Specific Sequences from a Database</p></td>
      </tr><tr><td>
          <p><code><a href="Seqs2DB.html">Seqs2DB()</a></code> </p>
        </td>
        <td><p>Add Sequences from Text File to Database</p></td>
      </tr><tr><td>
          <p><code><a href="DB2Seqs.html">DB2Seqs()</a></code> </p>
        </td>
        <td><p>Export Database Sequences to a FASTA or FASTQ File</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="gene-calling-and-annotation">Gene Calling and Annotation <a href="#gene-calling-and-annotation" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions for calling and annotating genetic regions</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="FindGenes.html">FindGenes()</a></code> </p>
        </td>
        <td><p>Find Genes in a Genome</p></td>
      </tr><tr><td>
          <p><code><a href="FindNonCoding.html">FindNonCoding()</a></code> </p>
        </td>
        <td><p>Find Non-Coding RNAs in a Genome</p></td>
      </tr><tr><td>
          <p><code><a href="LearnNonCoding.html">LearnNonCoding()</a></code> </p>
        </td>
        <td><p>Learn a Non-Coding RNA Model</p></td>
      </tr><tr><td>
          <p><code><a href="IdTaxa.html">IdTaxa()</a></code> </p>
        </td>
        <td><p>Assign Sequences a Taxonomic Classification</p></td>
      </tr><tr><td>
          <p><code><a href="Taxa-class.html">plot(<i>&lt;Taxa&gt;</i>)</a></code> <code><a href="Taxa-class.html">print(<i>&lt;Taxa&gt;</i>)</a></code> <code><a href="Taxa-class.html">`[`(<i>&lt;Taxa&gt;</i>)</a></code> </p>
        </td>
        <td><p>Taxa training and testing objects</p></td>
      </tr><tr><td>
          <p><code><a href="LearnTaxa.html">LearnTaxa()</a></code> </p>
        </td>
        <td><p>Train a Classifier for Assigning Taxonomy</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="constructing-cogs">Constructing COGs <a href="#constructing-cogs" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions to construct COGs from annotated genetic regions</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="DisjointSet.html">DisjointSet()</a></code> </p>
        </td>
        <td><p>Return single linkage clusters from <code>PairSummaries</code> objects.</p></td>
      </tr><tr><td>
          <p><code><a href="ExtractBy.html">ExtractBy()</a></code> </p>
        </td>
        <td><p>Extract and organize <code>DNAStringSets</code>s.</p></td>
      </tr><tr><td>
          <p><code><a href="FindSets.html">FindSets()</a></code> </p>
        </td>
        <td><p>Find all single linkage clusters in an undirected pairs list.</p></td>
      </tr><tr><td>
          <p><code><a href="LinkedPairs-class.html">print(<i>&lt;LinkedPairs&gt;</i>)</a></code> </p>
        </td>
        <td><p>Tables of where syntenic hits link pairs of genes</p></td>
      </tr><tr><td>
          <p><code><a href="NucleotideOverlap.html">NucleotideOverlap()</a></code> </p>
        </td>
        <td><p>Tabulating Pairs of Genomic Sequences</p></td>
      </tr><tr><td>
          <p><code><a href="PairSummaries.html">PairSummaries()</a></code> </p>
        </td>
        <td><p>Summarize connected pairs in a LinkedPairs object</p></td>
      </tr><tr><td>
          <p><code><a href="BlockExpansion.html">BlockExpansion()</a></code> </p>
        </td>
        <td><p>Attempt to expand blocks of paired features in a <code>PairSummaries</code> object.</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="constructing-phylogenies">Constructing Phylogenies <a href="#constructing-phylogenies" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions for phylogenetic reconstruction of genomes or COGs</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="Cophenetic.html">Cophenetic()</a></code> </p>
        </td>
        <td><p>Compute cophenetic distances on dendrogram objects</p></td>
      </tr><tr><td>
          <p><code><a href="DistanceMatrix.html">DistanceMatrix()</a></code> </p>
        </td>
        <td><p>Calculate the Distances Between Sequences</p></td>
      </tr><tr><td>
          <p><code><a href="IdClusters.html">IdClusters()</a></code> </p>
        </td>
        <td><p>Cluster Sequences By Similarity</p></td>
      </tr><tr><td>
          <p><code><a href="TreeLine.html">TreeLine()</a></code> </p>
        </td>
        <td><p>Construct a Phylogenetic Tree</p></td>
      </tr><tr><td>
          <p><code><a href="WriteDendrogram.html">WriteDendrogram()</a></code> </p>
        </td>
        <td><p>Write a Dendrogram to Newick Format</p></td>
      </tr></tbody><tbody><tr><th colspan="2">
          <h2 id="finding-co-evolving-gene-collectives">Finding Co-evolving Gene Collectives <a href="#finding-co-evolving-gene-collectives" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p><p>Functions for finding co-evolutionary signal between COGs</p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="ProtWeaver.html">ProtWeaver()</a></code> </p>
        </td>
        <td><p>ProtWeaver: Predicting Protein Functional Association Networks</p></td>
      </tr><tr><td>
          <p><code><a href="predict.ProtWeaver.html">predict(<i>&lt;ProtWeaver&gt;</i>)</a></code> </p>
        </td>
        <td><p>Make predictions with ProtWeaver objects</p></td>
      </tr><tr><td>
          <p><code><a href="plot.ProtWeb.html">plot(<i>&lt;ProtWeb&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot predictions in a ProtWeb object</p></td>
      </tr><tr><td>
          <p><code><a href="getProtWebData.ProtWeb.html">GetProtWebData(<i>&lt;ProtWeb&gt;</i>)</a></code> </p>
        </td>
        <td><p>Extract information from a ProtWeb object</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer></div>

  


  

  </body></html>

