<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Train a Classifier for Assigning Taxonomy — LearnTaxa • CompGenomicsBioc2022</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Train a Classifier for Assigning Taxonomy — LearnTaxa"><meta property="og:description" content="Trains a classifier based on a reference taxonomy containing sequence representatives assigned to taxonomic groups."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CompGenomicsBioc2022</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/workshop_syllabus.html">Comparative Genomics with SynExtend and DECIPHER</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ahl27/CompGenomicsBioc2022" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Train a Classifier for Assigning Taxonomy</h1>
    
    <div class="hidden name"><code>LearnTaxa.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Trains a classifier based on a reference taxonomy containing sequence representatives assigned to taxonomic groups.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">LearnTaxa</span><span class="op">(</span><span class="va">train</span>,
          <span class="va">taxonomy</span>,
          rank <span class="op">=</span> <span class="cn">NULL</span>,
          K <span class="op">=</span> <span class="cn">NULL</span>,
          N <span class="op">=</span> <span class="fl">500</span>,
          minFraction <span class="op">=</span> <span class="fl">0.01</span>,
          maxFraction <span class="op">=</span> <span class="fl">0.06</span>,
          maxIterations <span class="op">=</span> <span class="fl">10</span>,
          multiplier <span class="op">=</span> <span class="fl">100</span>,
          maxChildren <span class="op">=</span> <span class="fl">200</span>,
          alphabet <span class="op">=</span> <span class="va">AA_REDUCED</span><span class="op">[[</span><span class="fl">139</span><span class="op">]</span><span class="op">]</span>,
          verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>train</dt>
<dd><p>An <code>AAStringSet</code>, <code>DNAStringSet</code>, or <code>RNAStringSet</code> of unaligned sequences.</p></dd>
<dt>taxonomy</dt>
<dd><p>Character string providing the reference taxonomic assignment for each sequence in <code>train</code>.  Taxonomic ranks are separated by semicolons (``;'') beginning with ``Root''.</p></dd>
<dt>rank</dt>
<dd><p>Optionally, a <code>data.frame</code> with 5 named columns giving the ``Index'' (i.e., 0 to the number of unique taxa), ``Name'' (i.e., taxon name), ``Parent'' (i.e., ``Index'' of the parent taxon), ``Level'' (i.e., integer rank level), and ``Rank'' (e.g., ``genus'') of each taxonomic rank.  This information is often provided in a separate ``taxid'' file along with publicly available training sequence sets.</p></dd>
<dt>K</dt>
<dd><p>Integer specifying the k-mer size or <code>NULL</code> (the default) to calculate the k-mer size automatically.  The default value of <code>K</code> is such that matches between sequences are found by chance every <code>N</code> k-mers.</p></dd>
<dt>N</dt>
<dd><p>Numeric indicating the approximate number of k-mers that can be randomly selected before one is found by chance on average. For example, the default value of <code>500</code> will set <code>K</code> (when <code>K</code> is unspecified) so that every 500th k-mer is expected to match by chance.</p></dd>
<dt>minFraction</dt>
<dd><p>Numeric giving the minimum fraction of k-mers to sample during the initial tree descent phase of the classification algorithm.  (See details section below.)</p></dd>
<dt>maxFraction</dt>
<dd><p>Numeric giving the maximum fraction of k-mers to sample during the initial tree descent phase of the classification algorithm.  (See details section below.)</p></dd>
<dt>maxIterations</dt>
<dd><p>Integer specifying the maximum number of iterations to attempt re-classification of a training sequence before declaring it a ``problem sequence''.  (See details section below.)</p></dd>
<dt>multiplier</dt>
<dd><p>Numeric indicating the degree to which individual sequences have control over the fraction of k-mers sampled at any edge during the initial tree descent phase of the classification algorithm.  (See details section below.)</p></dd>
<dt>maxChildren</dt>
<dd><p>Integer giving the maximum number of child taxa of any taxon at which to consider further descending the taxonomic tree.  A value of <code>1</code> will prevent use of the tree descent algorithm altogether.  Lower values may decrease classification speed, but result in output objects that require less memory.</p></dd>
<dt>alphabet</dt>
<dd><p>Character vector of amino acid groupings used to reduce the 20 standard amino acids into smaller groups.  Alphabet reduction helps to find more distant homologies between sequences.  A non-reduced amino acid alphabet can be used by setting <code>alphabet</code> equal to <code>AA_STANDARD</code>.</p></dd>
<dt>verbose</dt>
<dd><p>Logical indicating whether to display progress.</p></dd>
</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>Learning about the training data is a two part process consisting of (i) forming a taxonomic tree and then (ii) ensuring that the <code>train</code>ing sequences can be correctly reclassified.  The latter step relies on reclassifying the sequences in <code>train</code> by descending the taxonomic tree, a process termed ``tree descent''.  Ultimately, the goal of tree descent is to quickly and accurately narrow the selection of groups where a sequence may belong.  During the learning process, tree descent is tuned so that it performs well when classifying new sequences.</p>
<p>The process of training the classifier first involves learning the taxonomic tree spanning all of the reference sequences in <code>train</code>.  Typically, reference taxonomic classifications are provided by an authoritative source, oftentimes along with a ``taxid'' file containing taxonomic <code>rank</code> information.  The taxonomic tree may contain any number of levels (e.g., Root, Phylum, Class, Order, Family, Genus) as long as they are hierarchically nested and always begin with ``Root''.</p>
<p>The second phase of training the classifier, tree descent, involves learning the optimal set of k-mers for discerning between the different sub-groups under each edge.  Here a fraction of the k-mers with the greatest discerning power are matched to a training sequence, and this process is repeated with 100 random subsamples to decide on the set of possible taxonomic groups to which a training sequence may belong.</p>
<p>The learning process works by attempting to correctly re-classify each training sequence in the taxonomy.  Initially, <code>maxFraction</code> of informative k-mers are repeatedly sampled at each edge during tree descent.  Training sequences that are incorrectly classified at an edge will lower the fraction of k-mers that are sampled by an amount that is proportional to <code>multiplier</code>.  As the fraction of sampled k-mers decreases, the tree descent process terminates at higher rank levels.</p>
<p>A major advantage of tree descent is that it both speeds up the classification process and indicates where the training set likely contains mislabeled sequences or incorrectly-placed taxonomic groups.  Training sequences that are not correctly classified within <code>maxIterations</code> are marked as ``problem sequences'', because it is likely that they are mislabeled.  If enough sequences have difficulty being correctly classified at an edge that the fraction drops below <code>minFraction</code>, then the edge is recorded as a ``problem group''.</p>
<p>The final result is an object that can be used for classification with <code><a href="IdTaxa.html">IdTaxa</a></code>, as well as information about <code>train</code> that could be used to help correct any errors in the taxonomy.</p>
    </div>
    <div id="value">
    <h2>Value</h2>
    <p>An object of class <code>Taxa</code> and subclass Train, which is stored as a list with components:</p>
<dl><dt>taxonomy</dt>
<dd><p>A character vector containing all possible groups in the taxonomy.</p></dd>
<dt>taxa</dt>
<dd><p>A character vector containing the basal taxon in each taxonomy.</p></dd>
<dt>ranks</dt>
<dd><p>A character vector of rank names for each taxon, or <code>NULL</code> if <code>rank</code> information was not supplied.</p></dd>
<dt>levels</dt>
<dd><p>Integer giving the rank level of each taxon.</p></dd>
<dt>children</dt>
<dd><p>A list containing the index of all children in the taxonomy for each taxon.</p></dd>
<dt>parents</dt>
<dd><p>An integer providing the index of the parent for each taxon.</p></dd>
<dt>fraction</dt>
<dd><p>A numeric between <code>minFraction</code> and <code>maxFraction</code> that represents the learned fraction of informative k-mers to sample for each taxon during the initial tree descent phase of the classification algorithm.  Problem groups are marked by a fraction of <code>NA</code>.</p></dd>
<dt>sequences</dt>
<dd><p>List containing the integer indices of sequences in <code>train</code> belonging to each taxon.</p></dd>
<dt>kmers</dt>
<dd><p>List containing the unique sorted k-mers (converted to integers) belonging to each sequence in <code>train</code>.</p></dd>
<dt>crossIndex</dt>
<dd><p>Integer indicating the index in taxonomy of each sequence's taxonomic label.</p></dd>
<dt>K</dt>
<dd><p>The value of <code>K</code> provided as input.</p></dd>
<dt>IDFweights</dt>
<dd><p>Numeric vector of length <code>4^K</code> providing the inverse document frequency weight for each k-mer.</p></dd>
<dt>decisionKmers</dt>
<dd><p>List of informative k-mers and their associated relative frequencies for each internal edge in the taxonomy.</p></dd>
<dt>problemSequences</dt>
<dd><p>A <code>data.frame</code> providing the ``Index'', ``Expected'' label, and ``Predicted'' taxon for sequences that could not be correctly classified during the initial tree descent phase of the algorithm.</p></dd>
<dt>problemGroups</dt>
<dd><p>Character vector containing any taxonomic groups that repeatedly had problems with correctly re-classifying sequences in <code>train</code> during the initial tree descent phase of the classification algorithm.  Problem groups likely indicate that a number of the sequences (or an entire group of sequences) assigned to the problem group are incorrectly placed in the taxonomic tree.</p></dd>
<dt>alphabet</dt>
<dd><p>The <code>alphabet</code> when <code>train</code> is an ``AAStringSet'', otherwise <code>NULL</code>.</p></dd>
</dl></div>
    <div id="references">
    <h2>References</h2>
    <p>Murali, A., et al. (2018). IDTAXA: a novel approach for accurate taxonomic classification of microbiome sequences. Microbiome, 6, 140. https://doi.org/10.1186/s40168-018-0521-5</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Erik Wright <a href="mailto:eswright@pitt.edu">eswright@pitt.edu</a></p>
    </div>
    <div id="note">
    <h2>Note</h2>
    <p>If <code>K</code> is <code>NULL</code>, the automatically determined value of <code>K</code> might be too large for some computers, resulting in an error.  In such cases it is recommended that <code>K</code> be manually set to a smaller value.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="IdTaxa.html">IdTaxa</a></code>, <code><a href="Taxa-class.html">Taxa-class</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co"># import training sequences</span></span>
<span class="r-in"><span class="va">fas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"50S_ribosomal_protein_L2.fas"</span>, package<span class="op">=</span><span class="st">"DECIPHER"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">dna</span> <span class="op">&lt;-</span> <span class="fu">readDNAStringSet</span><span class="op">(</span><span class="va">fas</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># parse the headers to obtain a taxonomy</span></span>
<span class="r-in"><span class="va">s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">dna</span><span class="op">)</span>, <span class="st">" "</span><span class="op">)</span></span>
<span class="r-in"><span class="va">genus</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">s</span>, <span class="va">`[`</span>, <span class="fl">1</span><span class="op">)</span></span>
<span class="r-in"><span class="va">species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">s</span>, <span class="va">`[`</span>, <span class="fl">2</span><span class="op">)</span></span>
<span class="r-in"><span class="va">taxonomy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Root"</span>, <span class="va">genus</span>, <span class="va">species</span>, sep<span class="op">=</span><span class="st">"; "</span><span class="op">)</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">taxonomy</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Root; Rickettsia; prowazekii"    "Root; Porphyromonas; gingivalis"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [3] "Root; Porphyromonas; gingivalis" "Root; Porphyromonas; gingivalis"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [5] "Root; Pasteurella; multocida"    "Root; Pasteurella; multocida"   </span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># train the classifier</span></span>
<span class="r-in"><span class="va">trainingSet</span> <span class="op">&lt;-</span> <span class="fu">LearnTaxa</span><span class="op">(</span><span class="va">dna</span>, <span class="va">taxonomy</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 5.39 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="va">trainingSet</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   A training set of class 'Taxa'</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    * K-mer size: 9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    * Number of rank levels: 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    * Total number of sequences: 317</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    * Number of groups: 118</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    * Number of problem groups: 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    * Number of problem sequences: 0</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># view information about the classifier</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">trainingSet</span><span class="op">)</span></span>
<span class="r-plt img"><img src="LearnTaxa-1.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span>
<span class="r-in"><span class="co"># train the classifier with amino acid sequences</span></span>
<span class="r-in"><span class="va">aa</span> <span class="op">&lt;-</span> <span class="fu">translate</span><span class="op">(</span><span class="va">dna</span><span class="op">)</span></span>
<span class="r-in"><span class="va">trainingSetAA</span> <span class="op">&lt;-</span> <span class="fu">LearnTaxa</span><span class="op">(</span><span class="va">aa</span>, <span class="va">taxonomy</span><span class="op">)</span></span>
<span class="r-in"><span class="va">trainingSetAA</span></span>
<span class="r-in"><span class="op">}</span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer></div>

  


  

  </body></html>

