<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Takes in a ``LinkedPairs'' object and gene calls, and returns a data.frame of paired features."><title>Summarize connected pairs in a LinkedPairs object — PairSummaries • CompGenomicsBioc2022</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Summarize connected pairs in a LinkedPairs object — PairSummaries"><meta property="og:description" content="Takes in a ``LinkedPairs'' object and gene calls, and returns a data.frame of paired features."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Coevolution%20Networks.html">Coevolutionary Networks</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com" aria-label="My Website">
    <span class="fa fa-potato"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Summarize connected pairs in a LinkedPairs object</h1>
      
      <div class="d-none name"><code>PairSummaries.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Takes in a ``LinkedPairs'' object and gene calls, and returns a data.frame of paired features.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">PairSummaries</span><span class="op">(</span><span class="va">SyntenyLinks</span>,
              <span class="va">DBPATH</span>,
              PIDs <span class="op">=</span> <span class="cn">FALSE</span>,
              Score <span class="op">=</span> <span class="cn">FALSE</span>,
              IgnoreDefaultStringSet <span class="op">=</span> <span class="cn">FALSE</span>,
              Verbose <span class="op">=</span> <span class="cn">FALSE</span>,
              Model <span class="op">=</span> <span class="st">"Generic"</span>,
              DefaultTranslationTable <span class="op">=</span> <span class="st">"11"</span>,
              AcceptContigNames <span class="op">=</span> <span class="cn">TRUE</span>,
              OffSetsAllowed <span class="op">=</span> <span class="cn">NULL</span>,
              Storage <span class="op">=</span> <span class="fl">1</span>,
              <span class="va">...</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>SyntenyLinks</dt>
<dd><p>A <code>LinkedPairs</code> object. In previous versions of this function, a <code>GeneCalls</code> object was also required, but this object is now carried forward from <code>NucleotideOverlap</code> inside the <code>LinkedPairs</code> object.</p></dd>
<dt>DBPATH</dt>
<dd><p>A SQLite connection object or a character string specifying the path to the database file constructed from DECIPHER's <code>Seqs2DB</code> function. This path is always required as ``PairsSummaries'' always computes the tetramer distance between paired sequences.</p></dd>
<dt>PIDs</dt>
<dd><p>Logical indicating whether to provide a PID for each pair. If <code>TRUE</code> all pairs will be aligned using DECIPHER's <code>AlignProfiles</code>. This step can be time consuming, especially for large numbers of pairs. Default is <code>FALSE</code>.</p></dd>
<dt>Score</dt>
<dd><p>Logical indicating whether to provide a length normalized score with DECIPHER's <code>ScoreAlignment</code> function. If <code>TRUE</code> all pairs will be aligned using DECIPHER's <code>AlignProfiles</code>. This step can be time consuming, especially for large numbers of pairs. Default is <code>FALSE</code>.</p></dd>
<dt>IgnoreDefaultStringSet</dt>
<dd><p>Logical indicating alignment type preferences. If <code>FALSE</code> (the default) pairs that can be aligned in amino acid space will be aligned as an <code>AAStringSet</code>. If <code>TRUE</code> all pairs will be aligned in nucleotide space. For <code>PairSummaries</code> to align the translation of a pair of sequences, both sequences must be tagged as coding in the ``GeneCalls'' object, and be the correct width for translation.</p></dd>
<dt>Verbose</dt>
<dd><p>Logical indicating whether or not to display a progress bar and print the time difference upon completion.</p></dd>
<dt>Model</dt>
<dd><p>A character string specifying a model to use to predict PIDs without performing an alignment. By default this argument is ``Generic'' specifying a generic PID prediction model based on PIDs computed from a randomly selected set of genomes. Currently no other models are included. Users may also supply their own model of type ``glm'' if they so desire in the form of an RData file. This model will need to take in some, or of the columns of statistics per pair that PairSummaries supplies.</p></dd>
<dt>DefaultTranslationTable</dt>
<dd><p>A character used to set the default translation table for <code>translate</code>. Is passed to <code>getGeneticCode</code>. Used when no translation table is specified in the ``GeneCalls'' object.</p></dd>
<dt>AcceptContigNames</dt>
<dd><p>Match names of contigs between gene calls object and synteny object. Where relevant, the first white space and everything following are removed from contig names. If <code>TRUE</code>, PairSummaries assumes that the contigs at each position in the synteny object and ``GeneCalls'' object are in the same order. Is automatically set to <code>TRUE</code> when ``GeneCalls'' are of class ``GRanges''. Is currently <code>TRUE</code> by default.</p></dd>
<dt>OffSetsAllowed</dt>
<dd><p>Defaults to <code>NULL</code>. Supplying an integer vector will indicate gap sizes to attempt to fill. A value of <code>2</code> will attempt to span gaps of size 1. If a vector larger than 1 is provided, i.e. <code>c(2, 3)</code>, will attempt to query all gap sizes implied by the vector, in this case gaps of size 1 and 2.</p></dd>
<dt>Storage</dt>
<dd><p>Numeric indicating the approximate size a user wishes to allow for holding <code>StringSet</code>s in memory to extract gene sequences, in ``Gigabytes''. The lower <code>Storage</code> is set, the more likely that <code>PairSummaries</code> will need to reaccess <code>StringSet</code>s when extracting gene sequences. The higher <code>Storage</code> is set, the more sequences <code>PairSummaries</code> will attempt to hold in memory, avoiding the need to re-access the source database many times. Set to 1 by default, indicating that <code>PairSummaries</code> can store a ``Gigabyte'' of sequences in memory at a time.</p></dd>
<dt>...</dt>
<dd><p>Arguments to be passed to <code>AlignProfiles</code>, and <code>DistanceMatrix</code>.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The <code>LinkedPairs</code> object generated by <code>NucleotideOverlap</code> is a container for raw data that describes possible orthologous relationships, however ultimate assignment of orthology is up to user discretion. <code>PairSummaries</code> generates a clear table with relevant statistics for a user to work with as they choose. The option to align all pairs, though onerous can allow users to apply a hard threshold to predictions by PID, while built in models can allow more expedient thresholding from predicted PIDs.</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A data.frame of class ``data.frame'' and ``PairSummaries'' of paired genes that are connected by syntenic hits. Contains columns describing the k-mers that link the pair. Columns ``p1'' and ``p2'' give the location ids of the the genes in the pair in the form ``DatabaseIdentifier_ContigIdentifier_GeneIdentifier''. ``ExactMatch'' provides an integer representing the exact number of nucleotides contained in the linking k-mers. ``TotalKmers'' provides an integer describing the number of distinct k-mers linking the pair. ``MaxKmer'' provides an integer describing the largest k-mer that links the pair. A column titled ``Consensus'' provides a value between zero and 1 indicating whether the kmers that link a pair of features are in the same position in each feature, with 1 indicating they are in exactly the same position and 0 indicating they are in as different a position as is possible. The ``Adjacent'' column provides an integer value ranging between 0 and 2 denoting whether a feature pair's direct neighbors are also paired. Gap filled pairs neither have neighbors, or are included as neighbors. The ``TetDist'' column provides the euclidean distance between oligonucleotide - of size 4 - frequences between predicted pairs. ``PIDType'' provides a character vector with values of ``NT'' where either of the pair indicates it is not a translatable sequence or ``AA'' where both sequences are translatable. If users choose to perform pairwise alignments there will be a ``PID'' column providing a numeric describing the percent identity between the two sequences. If users choose to predict PIDs using their own, or a provided model, a ``PredictedPID'' column will be provided.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Nicholas Cooley <a href="mailto:npc19@pitt.edu">npc19@pitt.edu</a></p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code>FindSynteny</code>, <code>Synteny-class</code>, <code><a href="NucleotideOverlap.html">NucleotideOverlap</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">DBPATH</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span>
<span class="r-in">                      <span class="st">"Endosymbionts.sqlite"</span>,</span>
<span class="r-in">                      package <span class="op">=</span> <span class="st">"SynExtend"</span><span class="op">)</span></span>
<span class="r-in">                      </span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Endosymbionts_LinkedFeatures"</span>, package <span class="op">=</span> <span class="st">"SynExtend"</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">Pairs</span> <span class="op">&lt;-</span> <span class="fu">PairSummaries</span><span class="op">(</span>SyntenyLinks <span class="op">=</span> <span class="va">Endosymbionts_LinkedFeatures</span>,</span>
<span class="r-in">                       PIDs <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span class="r-in">                       DBPATH <span class="op">=</span> <span class="va">DBPATH</span>,</span>
<span class="r-in">                       Verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Preparing overhead data.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Overhead complete.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Collecting pairs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ================================================================================</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 1.414115 secs</span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>

