<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="ProtWeaver is an S3 class with methods for predicting functional association using
protein or gene data. ProtWeaver implements several methods utilized in the literature, with many more planned for future implementation. For details on predictions, see predict.ProtWeaver."><title>ProtWeaver: Predicting Protein Functional Association Networks — ProtWeaver • CompGenomicsBioc2022</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ProtWeaver: Predicting Protein Functional Association Networks — ProtWeaver"><meta property="og:description" content="ProtWeaver is an S3 class with methods for predicting functional association using
protein or gene data. ProtWeaver implements several methods utilized in the literature, with many more planned for future implementation. For details on predictions, see predict.ProtWeaver."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/workshop_syllabus.html">Comparative Genomics with SynExtend and DECIPHER</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>ProtWeaver: Predicting Protein Functional Association Networks</h1>
      
      <div class="d-none name"><code>ProtWeaver.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>ProtWeaver is an S3 class with methods for predicting functional association using
protein or gene data. ProtWeaver implements several methods utilized in the literature, with many more planned for future implementation. For details on predictions, see <code><a href="predict.ProtWeaver.html">predict.ProtWeaver</a></code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">ProtWeaver</span><span class="op">(</span><span class="va">ListOfData</span>, NoWarn<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>ListOfData</dt>
<dd><p>A list of gene data, where each entry corresponds to information on a particular
gene. List must contain either dendrograms or vectors, and cannot contain a mixture. 
If list is composed of dendrograms, each dendrogram is a gene tree for the corresponding entry. If list is composed of vectors, vectors should be numeric or character vectors denoting the genomes containing that gene.</p></dd>
<dt>NoWarn</dt>
<dd><p>Several algorithms depend on having certain data. When a <code>ProtWeaver</code> object is initialized, it automatically selects which algorithms can be used given the input data. By default, ProtWeaver will notify the user of algorithms that cannot be used with warnings. Setting <code>NoWarn=TRUE</code> will suppress these messages.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>ProtWeaver expects input data to be a list. All entries must be one of the following:</p><ol><li><p><code>ListOfData[[i]] = c('ID#1', 'ID#2', ..., 'ID#k')</code></p></li>
<li><p><code>ListOfData[[i]] = c('i1_d1_p1', 'i2_d2_p2', ..., 'ik_dk_pk') </code></p></li>
<li><p><code>ListOfData[[i]] = dendrogram(...)</code></p></li>
</ol><p>In (1), each ID#i corresponds to the unique identifier for genome #i. For entry #j in the list, the presence of 'ID#i' means genome #i has an ortholog for gene/protein #j.</p>
<p>Case (2) is the same as (1), just with the formatting of names slightly different. Each entry is of the form <code>i_d_p</code>, where <code>i</code> is the unique identifier for the genome, <code>d</code> is which chromosome the ortholog is located, and <code>p</code> is what position the ortholog appears in on that chromosome. <code>p</code> must be a <code>numeric</code>, while the other entries can be any value.</p>
<p>Case (3) expects gene trees for each gene, with labeled leaves corresponding to each source genome. If <code>ListOfData</code> is in this format, taking <code>labels(ListOfData[[i]])</code> should produce a character vector that matches the format of either (2) or (1).</p>
<p><em>See the Examples section for illustrative examples.</em></p>
<p>ProtWeaver requires input of scenario (3) to use MirrorTree or ContextTree, and requires input of scenario (2) (or (3) with leaves labeled according to (2)) for co-localization analyses.</p>
<p>Note that ALL entries must belong to the same category--a combination of character vectors and dendrograms is not allowed.</p>
<p>Prediction of a functional association network is done using <code>predict(ProtWeaverObject)</code>. See <code><a href="predict.ProtWeaver.html">predict.ProtWeaver</a></code> for more information.</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns a ProtWeaver object.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Aidan Lakshman <a href="mailto:ahl27@pitt.edu">ahl27@pitt.edu</a></p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="predict.ProtWeaver.html">predict.ProtWeaver</a></code>,
<code>ExampleStreptomycesData</code>,
<code>BuiltInEnsembles</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co"># I'm using gene to mean either a gene or protein</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Imagine we have the following 4 genomes:</span></span>
<span class="r-in"><span class="co">## (each letter denotes a distinct gene)</span></span>
<span class="r-in"><span class="co">##    Genome 1: a b c d</span></span>
<span class="r-in"><span class="co">##    Genome 2: d c e</span></span>
<span class="r-in"><span class="co">##    Genome 3: b a e </span></span>
<span class="r-in"><span class="co">##    Genome 4: a e</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## We have 5 total genes: (a,b,c,d,e)</span></span>
<span class="r-in"><span class="co">##    a is present in genomes 1, 3, 4</span></span>
<span class="r-in"><span class="co">##    b is present in genomes 1, 3</span></span>
<span class="r-in"><span class="co">##    c is present in genomes 1, 2</span></span>
<span class="r-in"><span class="co">##    d is present in genomes 1, 2</span></span>
<span class="r-in"><span class="co">##    e is present in genomes 2, 3, 4</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Constructing a ProtWeaver object according to (1):</span></span>
<span class="r-in"><span class="va">l</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"><span class="va">l</span><span class="op">[[</span><span class="st">'a'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'1'</span>, <span class="st">'3'</span>, <span class="st">'4'</span><span class="op">)</span> </span>
<span class="r-in"><span class="va">l</span><span class="op">[[</span><span class="st">'b'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'1'</span>, <span class="st">'3'</span><span class="op">)</span> </span>
<span class="r-in"><span class="va">l</span><span class="op">[[</span><span class="st">'c'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'1'</span>, <span class="st">'2'</span><span class="op">)</span> </span>
<span class="r-in"><span class="va">l</span><span class="op">[[</span><span class="st">'d'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'1'</span>, <span class="st">'2'</span><span class="op">)</span> </span>
<span class="r-in"><span class="va">l</span><span class="op">[[</span><span class="st">'e'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'2'</span>, <span class="st">'3'</span>, <span class="st">'4'</span><span class="op">)</span> </span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Each value of the list corresponds to a gene</span></span>
<span class="r-in"><span class="co">## The associated vector shows which genomes have that gene</span></span>
<span class="r-in"><span class="va">pwCase1</span> <span class="op">&lt;-</span> <span class="fu">ProtWeaver</span><span class="op">(</span><span class="va">l</span><span class="op">)</span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Disabling MirrorTree-based algorithms. Input list must include inputs of type "dendrogram" for MT algorithms. Consult the documentation for more info.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Co-localization disabled. Labels must be in the format [GENOME]_[INDEX]_[ORDER] to use co-localization (where ORDER is a numeric). Consult the documentation for more info.</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Constructing a ProtWeaver object according to (2):</span></span>
<span class="r-in"><span class="co">##  Here we need to add in the chromosome and the position</span></span>
<span class="r-in"><span class="co">##  As we only have one chromosome, </span></span>
<span class="r-in"><span class="co">##  we can just set that to 1 for all.</span></span>
<span class="r-in"><span class="co">##  Position can be identified with knowledge, or with</span></span>
<span class="r-in"><span class="co">##  FindGenes(...) from DECIPHER.</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## In this toy case, genomes are small so it's simple.</span></span>
<span class="r-in"><span class="va">l</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"><span class="va">l</span><span class="op">[[</span><span class="st">'a'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'1_1_1'</span>, <span class="st">'3_1_2'</span>, <span class="st">'4_1_1'</span><span class="op">)</span> </span>
<span class="r-in"><span class="va">l</span><span class="op">[[</span><span class="st">'b'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'1_1_2'</span>, <span class="st">'3_1_1'</span><span class="op">)</span> </span>
<span class="r-in"><span class="va">l</span><span class="op">[[</span><span class="st">'c'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'1_1_3'</span>, <span class="st">'2_1_2'</span><span class="op">)</span> </span>
<span class="r-in"><span class="va">l</span><span class="op">[[</span><span class="st">'d'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'1_1_4'</span>, <span class="st">'2_1_1'</span><span class="op">)</span> </span>
<span class="r-in"><span class="va">l</span><span class="op">[[</span><span class="st">'e'</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'2_1_3'</span>, <span class="st">'3_1_3'</span>, <span class="st">'4_1_2'</span><span class="op">)</span> </span>
<span class="r-in"></span>
<span class="r-in"><span class="va">pwCase2</span> <span class="op">&lt;-</span> <span class="fu">ProtWeaver</span><span class="op">(</span><span class="va">l</span><span class="op">)</span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Disabling MirrorTree-based algorithms. Input list must include inputs of type "dendrogram" for MT algorithms. Consult the documentation for more info.</span>
<span class="r-in"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## For Case 3, we just need dendrogram objects for each</span></span>
<span class="r-in"><span class="co"># l[['a']] &lt;- dendrogram(...)</span></span>
<span class="r-in"><span class="co"># l[['b']] &lt;- dendrogram(...)</span></span>
<span class="r-in"><span class="co"># l[['c']] &lt;- dendrogram(...)</span></span>
<span class="r-in"><span class="co"># l[['d']] &lt;- dendrogram(...)</span></span>
<span class="r-in"><span class="co"># l[['e']] &lt;- dendrogram(...)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">## Leaf labels for these will be the same as the </span></span>
<span class="r-in"><span class="co">##  entries in Case 1.</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>

