<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Returns the set of sequences meeting the search criteria."><title>Obtain Specific Sequences from a Database — SearchDB • CompGenomicsBioc2022</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Obtain Specific Sequences from a Database — SearchDB"><meta property="og:description" content="Returns the set of sequences meeting the search criteria."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/CompGenomicsBioc2022.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/CreatingCOGs.html">Finding COGs with `SynExtend`</a>
    <a class="dropdown-item" href="../articles/CreatingTrees.html">Constructing Phylogenies with `DECIPHER`</a>
    <a class="dropdown-item" href="../articles/GeneCallingAndAnnotation.html">Gene Calling and Annotation with `DECIPHER`</a>
    <a class="dropdown-item" href="../articles/Introduction.html">Introduction</a>
    <a class="dropdown-item" href="../articles/ProtWeaver.html">Predicting Functional Association with `SynExtend`</a>
    <a class="dropdown-item" href="../articles/WorkingWithGenomeData.html">Loading in Genome Data with `DECIPHER`</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Obtain Specific Sequences from a Database</h1>
      
      <div class="d-none name"><code>SearchDB.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Returns the set of sequences meeting the search criteria.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">SearchDB</span><span class="op">(</span><span class="va">dbFile</span>,
         tblName <span class="op">=</span> <span class="st">"Seqs"</span>,
         identifier <span class="op">=</span> <span class="st">""</span>,
         type <span class="op">=</span> <span class="st">"XStringSet"</span>,
         limit <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>,
         replaceChar <span class="op">=</span> <span class="cn">NA</span>,
         nameBy <span class="op">=</span> <span class="st">"row_names"</span>,
         orderBy <span class="op">=</span> <span class="st">"row_names"</span>,
         countOnly <span class="op">=</span> <span class="cn">FALSE</span>,
         removeGaps <span class="op">=</span> <span class="st">"none"</span>,
         quality <span class="op">=</span> <span class="st">"Phred"</span>,
         clause <span class="op">=</span> <span class="st">""</span>,
         processors <span class="op">=</span> <span class="fl">1</span>,
         verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>dbFile</dt>
<dd><p>A SQLite connection object or a character string specifying the path to the database file.</p></dd>
<dt>tblName</dt>
<dd><p>Character string specifying the table where the sequences are located.</p></dd>
<dt>identifier</dt>
<dd><p>Optional character string used to narrow the search results to those matching a specific identifier.  If "" (the default) then all identifiers are selected.</p></dd>
<dt>type</dt>
<dd><p>The type of <code>XStringSet</code> (sequences) to return.  This should be (an unambiguous abbreviation of) one of <code>"XStringSet"</code>, <code>"DNAStringSet"</code>, <code>"RNAStringSet"</code>, <code>"AAStringSet"</code>, <code>"BStringSet"</code>, <code>"QualityScaledXStringSet"</code>, <code>"QualityScaledDNAStringSet"</code>, <code>"QualityScaledRNAStringSet"</code>, <code>"QualityScaledAAStringSet"</code>, or <code>"QualityScaledBStringSet"</code>.  If <code>type</code> is <code>"XStringSet"</code> or <code>"QualityScaledXStringSet"</code> then an attempt is made to guess the type of sequences based on their composition.</p></dd>
<dt>limit</dt>
<dd><p>Number of results to display.  The default (<code>-1</code>) does not limit the number of results.</p></dd>
<dt>replaceChar</dt>
<dd><p>Optional character used to replace any characters of the sequence that are not present in the <code>XStringSet</code>'s alphabet.  Not applicable if <code>type=="BStringSet"</code>.  The default (<code>NA</code>) results in an error if an incompatible character exist.  (See details section below.)</p></dd>
<dt>nameBy</dt>
<dd><p>Character string giving the column name for naming the <code>XStringSet</code>.</p></dd>
<dt>orderBy</dt>
<dd><p>Character string giving the column name for sorting the results.  Defaults to the order of entries in the database.  Optionally can be followed by <code>" ASC"</code> or <code>" DESC"</code> to specify ascending (the default) or descending order.</p></dd>
<dt>countOnly</dt>
<dd><p>Logical specifying whether to return only the number of sequences.</p></dd>
<dt>removeGaps</dt>
<dd><p>Determines how gaps ("-" or "." characters) are removed in the sequences.  This should be (an unambiguous abbreviation of) one of <code>"none"</code>, <code>"all"</code> or <code>"common"</code>.</p></dd>
<dt>clause</dt>
<dd><p>An optional character string to append to the query as part of a ``where clause''.</p></dd>
<dt>quality</dt>
<dd><p>The type of quality object to return if <code>type</code> is a <code>QualityScaledXStringSet</code>.  This should be (an unambiguous abbreviation of) one of <code>"Phred"</code>, <code>"Solexa"</code>, or <code>"Illumina"</code>.  Note that recent versions of Illumina software provide <code>"Phred"</code> formatted quality scores.</p></dd>
<dt>processors</dt>
<dd><p>The number of processors to use, or <code>NULL</code> to automatically detect and use all available processors.</p></dd>
<dt>verbose</dt>
<dd><p>Logical indicating whether to display queries as they are sent to the database.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>If <code>type</code> is <code>"DNAStringSet"</code> then all U's are converted to T's before creating the <code>DNAStringSet</code>, and vise-versa if <code>type</code> is <code>"RNAStringSet"</code>.  All remaining characters not in the <code>XStringSet</code>'s alphabet are converted to <code>replaceChar</code> or removed if <code>replaceChar</code> is <code>""</code>.  Note that if <code>replaceChar</code> is <code>NA</code> (the default), it will result in an error when an unexpected character is found.  Quality information is interpreted as <code>PredQuality</code> scores.</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>An <code>XStringSet</code> or <code>QualityScaledXStringSet</code> with the sequences that meet the specified criteria.  The <code>names</code> of the object correspond to the value in the <code>nameBy</code> column of the database.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>ES Wright (2016) "Using DECIPHER v2.0 to Analyze Big Biological Sequence Data in R". The R Journal, <b>8(1)</b>, 352-359.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Erik Wright <a href="mailto:eswright@pitt.edu">eswright@pitt.edu</a></p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="Seqs2DB.html">Seqs2DB</a></code>, <code><a href="DB2Seqs.html">DB2Seqs</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">db</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Bacteria_175seqs.sqlite"</span>, package<span class="op">=</span><span class="st">"DECIPHER"</span><span class="op">)</span></span>
<span class="r-in"><span class="co"># get all sequences in the default table:</span></span>
<span class="r-in"><span class="va">dna</span> <span class="op">&lt;-</span> <span class="fu">SearchDB</span><span class="op">(</span><span class="va">db</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Search Expression:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> select row_names, sequence from _Seqs where row_names in (select row_names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> from Seqs)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet of length: 175</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.01 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="co"># select a random sequence:</span></span>
<span class="r-in"><span class="va">dna</span> <span class="op">&lt;-</span> <span class="fu">SearchDB</span><span class="op">(</span><span class="va">db</span>, orderBy<span class="op">=</span><span class="st">"random()"</span>, limit<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Search Expression:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> select Seqs.row_names, _Seqs.sequence from Seqs join _Seqs on Seqs.row_names =</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _Seqs.row_names where _Seqs.row_names in (select row_names from Seqs) order by</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> random() limit 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet of length: 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.01 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="co"># remove gaps from "Mycobacterium" sequences:</span></span>
<span class="r-in"><span class="va">dna</span> <span class="op">&lt;-</span> <span class="fu">SearchDB</span><span class="op">(</span><span class="va">db</span>, identifier<span class="op">=</span><span class="st">"Mycobacterium"</span>, removeGaps<span class="op">=</span><span class="st">"all"</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Search Expression:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> select row_names, sequence from _Seqs where row_names in (select row_names</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> from Seqs where identifier is "Mycobacterium")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet of length: 22</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="co"># provide a more complex query:</span></span>
<span class="r-in"><span class="va">dna</span> <span class="op">&lt;-</span> <span class="fu">SearchDB</span><span class="op">(</span><span class="va">db</span>, nameBy<span class="op">=</span><span class="st">"description"</span>, orderBy<span class="op">=</span><span class="st">"bases"</span>, removeGaps<span class="op">=</span><span class="st">"common"</span>,</span>
<span class="r-in">                clause<span class="op">=</span><span class="st">"nonbases is 0"</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Search Expression:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> select description, _Seqs.sequence from Seqs join _Seqs on Seqs.row_names =</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _Seqs.row_names where _Seqs.row_names in (select row_names from Seqs where</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> nonbases is 0) order by bases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet of length: 19</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>

