<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Adds sequences to a database."><title>Add Sequences from Text File to Database — Seqs2DB • CompGenomicsBioc2022</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Add Sequences from Text File to Database — Seqs2DB"><meta property="og:description" content="Adds sequences to a database."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">Coevolutionary Networks</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com/tutorials" aria-label="My Website">
    <span class="fa fa-house-chimney"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Add Sequences from Text File to Database</h1>
      
      <div class="d-none name"><code>Seqs2DB.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Adds sequences to a database.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">Seqs2DB</span><span class="op">(</span><span class="va">seqs</span>,
        <span class="va">type</span>,
        <span class="va">dbFile</span>,
        <span class="va">identifier</span>,
        tblName <span class="op">=</span> <span class="st">"Seqs"</span>,
        chunkSize <span class="op">=</span> <span class="fl">1e7</span>,
        replaceTbl <span class="op">=</span> <span class="cn">FALSE</span>,
        fields <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>accession <span class="op">=</span> <span class="st">"ACCESSION"</span>, rank <span class="op">=</span> <span class="st">"ORGANISM"</span><span class="op">)</span>,
        processors <span class="op">=</span> <span class="fl">1</span>,
        verbose <span class="op">=</span> <span class="cn">TRUE</span>,
        <span class="va">...</span><span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>seqs</dt>
<dd><p>A connection object or a character string specifying the file path to the file containing the sequences, an <code>XStringSet</code> object if <code>type</code> is <code>XStringSet</code>, or a <code>QualityScaledXStringSet</code> object if <code>type</code> is <code>QualityScaledXStringSet</code>. Files compressed with <code>gzip</code>, <code>bzip2</code>, <code>xz</code>, or <code>lzma</code> compression are automatically detected and decompressed during import.  Full URL paths (e.g., "http://" or "ftp://") to uncompressed text files or <code>gzip</code> compressed text files can also be used.</p></dd>

  <dt>type</dt>
<dd><p>The type of the sequences (<code>seqs</code>) being imported.  This should be (an unambiguous abbreviation of) one of <code>"FASTA"</code>, <code>"FASTQ"</code>, <code>"GenBank"</code>, <code>"XStringSet"</code>, or <code>"QualityScaledXStringSet"</code>.</p></dd>

  <dt>dbFile</dt>
<dd><p>A SQLite connection object or a character string specifying the path to the database file.  If the <code>dbFile</code> does not exist then a new database is created at this location.</p></dd>

  <dt>identifier</dt>
<dd><p>Character string specifying the <code>"id"</code> to give the imported sequences in the database.</p></dd>

  <dt>tblName</dt>
<dd><p>Character string specifying the table in which to add the sequences.</p></dd>

  <dt>chunkSize</dt>
<dd><p>Number of characters to read at a time.</p></dd>

  <dt>replaceTbl</dt>
<dd><p>Logical indicating whether to overwrite the entire table in the database.  If <code>FALSE</code> (the default) then the sequences are appended to any already existing in the <code>tblName</code>.  If <code>TRUE</code> the entire table is dropped, removing any existing sequences before adding any new sequences.</p></dd>

  <dt>fields</dt>
<dd><p>Named character vector providing the fields to import from a <code>"GenBank"</code> formatted file as text columns in the database (not applicable for other <code>"type"</code>s).  The default is to import the <code>"ACCESSION"</code> field as a column named <code>"accession"</code> and the <code>"ORGANISM"</code> field as a column named <code>"rank"</code>.  Other uppercase fields, such as <code>"LOCUS"</code> or <code>"VERSION"</code>, can be specified in similar manner.  Note that the <code>"DEFINITION"</code> field is automatically imported as a column named <code>"description"</code> in the database.</p></dd>

  <dt>processors</dt>
<dd><p>The number of processors to use, or <code>NULL</code> to automatically detect and use all available processors.</p></dd>

  <dt>verbose</dt>
<dd><p>Logical indicating whether to display each query as it is sent to the database.</p></dd>

  <dt>...</dt>
<dd><p>Further arguments to be passed directly to <code>Codec</code> for compressing sequence information.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Sequences are imported into the database in chunks of lines specified by <code>chunkSize</code>.  The sequences can then be identified by searching the database for the <code>identifier</code> provided.  Sequences are added to the database verbatim, so that no sequence information is lost when the sequences are exported from the database.  The sequence (record) names are recorded into a column named ``description'' in the database.</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>The total number of sequences in the database table is returned after import.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>ES Wright (2016) "Using DECIPHER v2.0 to Analyze Big Biological Sequence Data in R". The R Journal, <b>8(1)</b>, 352-359.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Erik Wright <a href="mailto:eswright@pitt.edu">eswright@pitt.edu</a></p>
    </div>
    <div class="section level2">
    <h2 id="warning">Warning<a class="anchor" aria-label="anchor" href="#warning"></a></h2>
    <p>If <code>replaceTbl</code> is <code>TRUE</code> then any sequences already in the table are overwritten, which is equivalent to dropping the entire table.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="BrowseDB.html">BrowseDB</a></code>, <code><a href="SearchDB.html">SearchDB</a></code>, <code><a href="DB2Seqs.html">DB2Seqs</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">gen</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Bacteria_175seqs.gen"</span>, package<span class="op">=</span><span class="st">"DECIPHER"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">dbConn</span> <span class="op">&lt;-</span> <span class="fu">dbConnect</span><span class="op">(</span><span class="fu">SQLite</span><span class="op">(</span><span class="op">)</span>, <span class="st">":memory:"</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">Seqs2DB</span><span class="op">(</span><span class="va">gen</span>, <span class="st">"GenBank"</span>, <span class="va">dbConn</span>, <span class="st">"Bacteria"</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading GenBank file chunk 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 175 total sequences in table Seqs.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.08 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="fu"><a href="BrowseDB.html">BrowseDB</a></span><span class="op">(</span><span class="va">dbConn</span><span class="op">)</span></span>
<span class="r-in"><span class="va">dna</span> <span class="op">&lt;-</span> <span class="fu"><a href="SearchDB.html">SearchDB</a></span><span class="op">(</span><span class="va">dbConn</span>, nameBy<span class="op">=</span><span class="st">"description"</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Search Expression:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> select description, _Seqs.sequence from Seqs join _Seqs on Seqs.row_names =</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> _Seqs.row_names where _Seqs.row_names in (select row_names from Seqs)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DNAStringSet of length: 175</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Time difference of 0.01 secs</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span class="fu">dbDisconnect</span><span class="op">(</span><span class="va">dbConn</span><span class="op">)</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

    </footer></div>

  

  

  </body></html>

