<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="CompGenomicsBioc2022">
<title>Building Coevolution Networks with SynExtend • CompGenomicsBioc2022</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Building Coevolution Networks with SynExtend">
<meta property="og:description" content="CompGenomicsBioc2022">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">Coevolutionary Networks</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html">Conclusion</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com/tutorials" aria-label="Homepage">
    <span class="fa fa-user"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Building Coevolution Networks with SynExtend</h1>
                        <h4 data-toc-skip class="author">Aidan
Lakshman<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;a href="mailto:ahl27@pitt.edu" class="email"&gt;ahl27@pitt.edu&lt;/a&gt;&lt;/p&gt;'><sup>1</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ahl27/CompGenomicsBioc2022/blob/HEAD/vignettes/CoevolutionNetworks.Rmd" class="external-link"><code>vignettes/CoevolutionNetworks.Rmd</code></a></small>
      <div class="d-none name"><code>CoevolutionNetworks.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="coevolutionary-analysis">Coevolutionary Analysis<a class="anchor" aria-label="anchor" href="#coevolutionary-analysis"></a>
</h2>
<p>At this point, we’ve learned how to generate gene trees for a set of
COGs. We’re now ready to look for signals of coevolution, which imply
functional associations. These methods are implemented via the
<code>ProtWeaver</code> class in <code>SynExtend</code>, which includes
many commonly used methods for detecting coevolutionary patterns.</p>
<p>While the previous steps have only utilized a small subsample of the
data, we’ve pre-computed the complete dataset for you to play with. This
dataset is comprised of the 91 <em>Micrococcus</em> genomes available
with assemblies at the Scaffold, Chromosome, or Complete level (<a href="https://www.ncbi.nlm.nih.gov/data-hub/genome/?taxon=1269&amp;annotated_only=true&amp;refseq_annotation=true&amp;assembly_level=1%3A3" class="external-link">link
to query</a>). Note that more genomes may become available after this
conference; these are all that were available at the time.</p>
<p>We ran the complete pipeline of identifying and annotating genes with
<code>DECIPHER</code>, finding COGs with <code>SynExtend</code>, and
then creating gene trees for each COG using <code>DECIPHER</code>. The
complete data consist of 3,407 distinct COGs. All of this analysis is
performed entirely within <code>SynExtend</code> and
<code>DECIPHER</code>; no external packages or data are required aside
from the input genomes.</p>
<p>We now use the new <code>ProtWeaver</code> class to try to find COGs
that show evidence of correlated evolutionary selective pressures, also
referred to as ‘coevolutionary signal’.</p>
<p>Let’s start by loading in our input data. For people downloading the
datafiles, these files are found in <code>CoevNetworks/</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DECIPHER</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SynExtend</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">COGsFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, <span class="st">'CoevNetworks'</span>, </span>
<span>                        <span class="st">'MicrococcusCOGs.RData'</span>,</span>
<span>                        package<span class="op">=</span><span class="st">'CompGenomicsBioc2022'</span><span class="op">)</span></span>
<span><span class="va">AnnotationsFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, <span class="st">'CoevNetworks'</span>, </span>
<span>                               <span class="st">'MicrococcusAnnotations.RData'</span>,</span>
<span>                        package<span class="op">=</span><span class="st">'CompGenomicsBioc2022'</span><span class="op">)</span></span>
<span><span class="va">TreesFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, <span class="st">'CoevNetworks'</span>, </span>
<span>                         <span class="st">'MicrococcusGeneTrees.RData'</span>,</span>
<span>                        package<span class="op">=</span><span class="st">'CompGenomicsBioc2022'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># All of these files are formatted the same way, as lists with each entry</span></span>
<span><span class="co"># corresponding to a COG. COGs entries are identifiers for each gene, CogsAnnot </span></span>
<span><span class="co"># entries are annotations, and CogTrees entries are gene trees</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">COGsFile</span><span class="op">)</span>          <span class="co"># Loads 'COGs'</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">AnnotationsFile</span><span class="op">)</span>   <span class="co"># Loads 'CogsAnnot'</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">TreesFile</span><span class="op">)</span>         <span class="co"># Loads 'CogTrees'</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Let's look at one of the COGs</span></span>
<span><span class="va">COGs</span><span class="op">[[</span><span class="fl">7</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "1_1_919"     "12_44_1601"  "13_5_415"    "2_1_10"      "14_1_274"   </span></span>
<span><span class="co">##  [6] "15_1_14"     "16_132_1767" "17_107_1177" "18_1_1825"   "19_1_1783"  </span></span>
<span><span class="co">## [11] "20_1_958"    "21_223_1464" "22_44_1726"  "23_13_736"   "3_1_130"    </span></span>
<span><span class="co">## [16] "24_11_767"   "25_1_915"    "26_8_258"    "27_53_1567"  "28_10_1132" </span></span>
<span><span class="co">## [21] "29_53_1900"  "30_1_72"     "31_7_808"    "32_7_458"    "33_183_1337"</span></span>
<span><span class="co">## [26] "34_130_2076" "35_197_1194" "36_128_792"  "37_22_1142"  "38_81_1984" </span></span>
<span><span class="co">## [31] "4_1_9"       "39_1_1608"   "40_1_416"    "41_1_925"    "42_1_1032"  </span></span>
<span><span class="co">## [36] "43_1_707"    "44_1_704"    "45_13_755"   "46_63_1530"  "47_1_970"   </span></span>
<span><span class="co">## [41] "48_1_690"    "49_1_779"    "50_1_267"    "51_20_1253"  "52_1_971"   </span></span>
<span><span class="co">## [46] "53_105_395"  "54_109_577"  "55_33_1488"  "56_11_533"   "57_161_2179"</span></span>
<span><span class="co">## [51] "58_111_515"  "59_55_1658"  "5_1_9"       "60_50_1995"  "61_1_78"    </span></span>
<span><span class="co">## [56] "62_8_204"    "63_155_1042" "64_47_712"   "65_96_2144"  "66_11_774"  </span></span>
<span><span class="co">## [61] "67_66_1831"  "68_22_82"    "69_22_1206"  "70_34_781"   "71_96_2235" </span></span>
<span><span class="co">## [66] "72_1_1811"   "73_11_821"   "74_1_221"    "75_15_1093"  "76_1_2110"  </span></span>
<span><span class="co">## [71] "77_1_696"    "78_1_1963"   "79_12_872"   "80_12_855"   "81_48_2136" </span></span>
<span><span class="co">## [76] "82_41_671"   "83_20_1289"  "84_185_549"  "85_42_1351"  "86_8_1238"  </span></span>
<span><span class="co">## [81] "88_7_611"    "89_126_1611" "90_1_980"    "6_21_1244"   "7_40_2015"  </span></span>
<span><span class="co">## [86] "8_1_348"     "9_5_575"     "10_1_382"    "11_1_10"     "11_12_1037" </span></span>
<span><span class="co">## [91] "91_1_985"</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">CogsAnnot</span><span class="op">[[</span><span class="fl">7</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##   A test set of class 'Taxa' with length 91</span></span>
<span><span class="co">##      confidence name                 taxon</span></span>
<span><span class="co">##  [1]       100% 50_1_267             Root; 09120 Genetic Information Processi...</span></span>
<span><span class="co">##  [2]       100% 51_20_1253           Root; 09120 Genetic Information Processi...</span></span>
<span><span class="co">##  [3]       100% 52_1_971             Root; 09120 Genetic Information Processi...</span></span>
<span><span class="co">##  [4]        71% 53_105_395           Root; 09120 Genetic Information Processi...</span></span>
<span><span class="co">##  [5]       100% 54_109_577           Root; 09120 Genetic Information Processi...</span></span>
<span><span class="co">##  ...        ... ...                  ...</span></span>
<span><span class="co">## [87]       100% 45_13_755            Root; 09120 Genetic Information Processi...</span></span>
<span><span class="co">## [88]       100% 46_63_1530           Root; 09120 Genetic Information Processi...</span></span>
<span><span class="co">## [89]       100% 47_1_970             Root; 09120 Genetic Information Processi...</span></span>
<span><span class="co">## [90]       100% 48_1_690             Root; 09120 Genetic Information Processi...</span></span>
<span><span class="co">## [91]       100% 49_1_779             Root; 09120 Genetic Information Processi...</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">CogsAnnot</span><span class="op">[[</span><span class="fl">7</span><span class="op">]</span><span class="op">]</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $taxon</span></span>
<span><span class="co">## [1] "Root"                                                             </span></span>
<span><span class="co">## [2] "09120 Genetic Information Processing"                             </span></span>
<span><span class="co">## [3] "09122 Translation"                                                </span></span>
<span><span class="co">## [4] "03010 Ribosome [PATH:ko03010]"                                    </span></span>
<span><span class="co">## [5] "K02899  RP-L27, MRPL27, rpmA, large subunit ribosomal protein L27"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $confidence</span></span>
<span><span class="co">## [1] 100 100 100 100 100</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">CogTrees</span><span class="op">[[</span><span class="fl">7</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## 'dendrogram' with 2 branches and 91 members total, at height 0.3813559</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Note that tree labels correspond to gene identifiers</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">setequal</a></span><span class="op">(</span><span class="va">COGs</span><span class="op">[[</span><span class="fl">7</span><span class="op">]</span><span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/r/base/labels.html" class="external-link">labels</a></span><span class="op">(</span><span class="va">CogTrees</span><span class="op">[[</span><span class="fl">7</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<p>There is a ton of data here, and we unfortunately don’t have time to
look at all of it. To demonstrate some of the things we can do with
<code>ProtWeaver</code>, we’re going to look at subset of the data that
is easier to investigate.</p>
<p>We’ll subset the COGs to ones that meet the following
characteristics:</p>
<ul>
<li>COGs with no paralogs</li>
<li>COGs not part of the core genome</li>
<li>COGs present in at least 5 genomes (not singletons or super rare
ones)</li>
<li>COGs with at least one high confidence annotation</li>
<li>COGs that imply a coding region</li>
</ul>
<p>These are relatively arbitrary requirements, but they subset the data
to an example that runs quickly and is easily understandable. This
essentially takes a group of COGs that are not essential to the
organisms but tend to appear a lot, that have been characterized
individually before.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Subsetting COGs</span></span>
<span></span>
<span><span class="co"># Cutoff values (91 genomes total!)</span></span>
<span><span class="va">CoreCutoff</span> <span class="op">&lt;-</span> <span class="fl">88</span></span>
<span><span class="va">UnderCutoff</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span></span>
<span><span class="co"># Get assembly identifiers for each COG</span></span>
<span><span class="va">truncCOGs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">COGs</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">'([0-9]*)_.*'</span>, <span class="st">'\\1'</span>, <span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Find COGs without paralogs</span></span>
<span><span class="va">noParas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">truncCOGs</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get non-core genome</span></span>
<span><span class="va">notCoreCOGs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">truncCOGs</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op">&lt;</span> <span class="va">CoreCutoff</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get genes in more than 5 organisms</span></span>
<span><span class="va">notSingles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">truncCOGs</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;</span> <span class="va">UnderCutoff</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Make sure COGs are coding elements</span></span>
<span><span class="va">codingCOGs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">CogsAnnot</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">is</span><span class="op">(</span><span class="va">x</span>, <span class="st">'Taxa'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># At least one high confidence annotation</span></span>
<span><span class="va">highConf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">CogsAnnot</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> </span>
<span>                   <span class="kw">if</span><span class="op">(</span><span class="fu">is</span><span class="op">(</span><span class="va">x</span>, <span class="st">'Taxa'</span><span class="op">)</span><span class="op">)</span> </span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">x</span>, \<span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">$</span><span class="va">confidence</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">y</span><span class="op">$</span><span class="va">confidence</span><span class="op">)</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">50</span></span>
<span>                   <span class="kw">else</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Subset for the workshop</span></span>
<span><span class="va">WorkshopSubset</span> <span class="op">&lt;-</span> <span class="va">noParas</span> <span class="op">&amp;</span> <span class="va">notCoreCOGs</span> <span class="op">&amp;</span> <span class="va">notSingles</span> <span class="op">&amp;</span> <span class="va">codingCOGs</span> <span class="op">&amp;</span> <span class="va">highConf</span></span>
<span></span>
<span><span class="co"># Subset our data</span></span>
<span><span class="va">WCogs</span> <span class="op">&lt;-</span> <span class="va">COGs</span><span class="op">[</span><span class="va">WorkshopSubset</span><span class="op">]</span></span>
<span><span class="va">WAnnots</span> <span class="op">&lt;-</span> <span class="va">CogsAnnot</span><span class="op">[</span><span class="va">WorkshopSubset</span><span class="op">]</span></span>
<span><span class="va">WTrees</span> <span class="op">&lt;-</span> <span class="va">CogTrees</span><span class="op">[</span><span class="va">WorkshopSubset</span><span class="op">]</span></span></code></pre></div>
<p>Let’s also put together a list of consensus annotations for each
COG.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">consAnnots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">vector</a></span><span class="op">(</span><span class="st">'character'</span>, length<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">CogsAnnot</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span> <span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">CogsAnnot</span><span class="op">)</span> <span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">taxaentry</span> <span class="op">&lt;-</span> <span class="va">CogsAnnot</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu">is</span><span class="op">(</span><span class="va">taxaentry</span>, <span class="st">'Taxa'</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="va">consAnnots</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'NONCODING'</span></span>
<span>  <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="va">annots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">taxaentry</span>, \<span class="op">(</span><span class="va">y</span><span class="op">)</span> <span class="va">y</span><span class="op">$</span><span class="va">taxon</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">y</span><span class="op">$</span><span class="va">taxon</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span>    <span class="va">annots</span> <span class="op">&lt;-</span> <span class="va">annots</span><span class="op">[</span><span class="va">annots</span><span class="op">!=</span><span class="st">'unclassified_Root'</span><span class="op">]</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">annots</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span></span>
<span>      <span class="va">consAnnots</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">'PNACT'</span></span>
<span>    <span class="kw">else</span></span>
<span>      <span class="va">consAnnots</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">annots</span><span class="op">)</span>, decreasing<span class="op">=</span><span class="cn">T</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>Now we can make our <code>ProtWeaver</code> object.
<code>ProtWeaver</code> has multiple input options, either a list
formatted like <code>COGs</code> (list with gene identifiers) or a list
like <code>CogTrees</code> (list with gene trees).</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SynExtend/man/ProtWeaver.html" class="external-link">ProtWeaver</a></span><span class="op">(</span><span class="va">WTrees</span><span class="op">)</span></span></code></pre></div>
<p>The <code>ProtWeaver</code> constuctor automatically detects the type
of data you have and adjusts available predictors accordingly. While it
functions best with a list of dendrograms for each COG, it can also run
with simple presence/absence patterns. See the documentation file for
<code>ProtWeaver</code> for more information on this functionality.</p>
<p>We’re now ready to make predictions. Predicting functional
associations is done with the <code>predict.ProtWeaver</code> S3 method.
Let’s examine possible functional associations between the COGs we
have.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">preds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span>
<span></span>
<span><span class="co">## Warning in cor(v1, v2, use = "na.or.complete", method = "pearson"): the standard</span></span>
<span><span class="co">## deviation is zero</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">preds</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## a ProtWeb object.</span></span>
<span><span class="co">##  Method used: Ensemble </span></span>
<span><span class="co">##  Number of genes: 88 </span></span>
<span><span class="co">##  Number of predictions: 3916</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="viewing-our-results">Viewing our results<a class="anchor" aria-label="anchor" href="#viewing-our-results"></a>
</h2>
<p>Notice that <code>preds</code> is a <code>ProtWeb</code> object. This
is just a simple S3 class with a pretty print method wrapping a matrix
of pairwise association scores. We can get the raw data with
<code><a href="https://rdrr.io/pkg/SynExtend/man/getProtWebData.ProtWeb.html" class="external-link">GetProtWebData()</a></code>:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Subset so the output is actually readable</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SynExtend/man/getProtWebData.ProtWeb.html" class="external-link">GetProtWebData</a></span><span class="op">(</span><span class="va">preds</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">7</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##           1         2         3         4         5         6         7</span></span>
<span><span class="co">## 1 1.0000000 0.3663701 0.2026004 0.2212769 0.2583244 0.2565627 0.1959871</span></span>
<span><span class="co">## 2 0.3663701 1.0000000 0.1677659 0.2811801 0.2310440 0.2988038 0.2406802</span></span>
<span><span class="co">## 3 0.2026004 0.1677659 1.0000000 0.2175542 0.1921812 0.2210058 0.2324306</span></span>
<span><span class="co">## 4 0.2212769 0.2811801 0.2175542 1.0000000 0.2177652 0.2634777 0.1868761</span></span>
<span><span class="co">## 5 0.2583244 0.2310440 0.1921812 0.2177652 1.0000000 0.3015111 0.1798632</span></span>
<span><span class="co">## 6 0.2565627 0.2988038 0.2210058 0.2634777 0.3015111 1.0000000 0.1762112</span></span>
<span><span class="co">## 7 0.1959871 0.2406802 0.2324306 0.1868761 0.1798632 0.1762112 1.0000000</span></span></code></pre>
<p>The <code>ProtWeb</code> class will be updated next release cycle to
include more methods, including a custom plotting function. The current
<code>plot.ProtWeb</code> S3 method implements a force-directed
embedding of the pairwise scores, but it’s a big work-in-progress. Stay
tuned for the next release cycle for more functionality regarding
<code>ProtWeb</code>.</p>
<p>In the meantime, we can use the <code>igraph</code> package to find
clusters of coevolving COGs.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org" class="external-link">igraph</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'igraph'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:Biostrings':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     union</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:XVector':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     path</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:IRanges':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     union</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:S4Vectors':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     union</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:BiocGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     normalize, path, union</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     decompose, spectrum</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     union</span></span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">adjMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SynExtend/man/getProtWebData.ProtWeb.html" class="external-link">GetProtWebData</a></span><span class="op">(</span><span class="va">preds</span><span class="op">)</span></span>
<span><span class="va">COGnames</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">WTrees</span><span class="op">)</span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_from_adjacency_matrix.html" class="external-link">graph_from_adjacency_matrix</a></span><span class="op">(</span><span class="va">adjMatrix</span>, weighted<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                                 mode<span class="op">=</span><span class="st">'undirected'</span>, diag<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">clusters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/cluster_fast_greedy.html" class="external-link">cluster_fast_greedy</a></span><span class="op">(</span><span class="va">g</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Getting the clusters &amp; identifying COGs by consensus annotation</span></span>
<span><span class="va">clusterLabels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">vector</a></span><span class="op">(</span><span class="st">'list'</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span> <span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">clusterLabels</span><span class="op">)</span> <span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">cluster</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/communities.html" class="external-link">communities</a></span><span class="op">(</span><span class="va">clusters</span><span class="op">)</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span></span>
<span>  <span class="va">labs</span> <span class="op">&lt;-</span> <span class="va">consAnnots</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="va">cluster</span><span class="op">)</span><span class="op">]</span></span>
<span>  <span class="va">clusterLabels</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">labs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">labs</span>, \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">' '</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="op">}</span></span>
<span><span class="va">clusterLabels</span></span></code></pre></div>
<pre><code><span><span class="co">## [[1]]</span></span>
<span><span class="co">## [1] "K00526  E1.17.4.1B, nrdB, nrdF, ribonucleoside-diphosphate reductase beta chain [EC:1.17.4.1]"</span></span>
<span><span class="co">## [2] "K00265  gltB, glutamate synthase (NADPH) large chain [EC:1.4.1.13]"                           </span></span>
<span><span class="co">## [3] "K00266  gltD, glutamate synthase (NADPH) small chain [EC:1.4.1.13]"                           </span></span>
<span><span class="co">## [4] "K05569  mnhE, mrpE, multicomponent Na+:H+ antiporter subunit E"                               </span></span>
<span><span class="co">## [5] "K05570  mnhF, mrpF, multicomponent Na+:H+ antiporter subunit F"                               </span></span>
<span><span class="co">## [6] "K01695  trpA, tryptophan synthase alpha chain [EC:4.2.1.20]"                                  </span></span>
<span><span class="co">## [7] "PNACT"                                                                                        </span></span>
<span><span class="co">## [8] "NONCODING"                                                                                    </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## [[2]]</span></span>
<span><span class="co">## [1] "unclassified_03060 Protein export [PATH:ko03060]"               </span></span>
<span><span class="co">## [2] "K00927  PGK, pgk, phosphoglycerate kinase [EC:2.7.2.3]"         </span></span>
<span><span class="co">## [3] "K01803  TPI, tpiA, triosephosphate isomerase (TIM) [EC:5.3.1.1]"</span></span>
<span><span class="co">## [4] "NONCODING"                                                      </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## [[3]]</span></span>
<span><span class="co">##  [1] "K00153  E1.1.1.306, S-(hydroxymethyl)mycothiol dehydrogenase [EC:1.1.1.306]"                   </span></span>
<span><span class="co">##  [2] "K00525  E1.17.4.1A, nrdA, nrdE, ribonucleoside-diphosphate reductase alpha chain [EC:1.17.4.1]"</span></span>
<span><span class="co">##  [3] "K11754  folC, dihydrofolate synthase / folylpolyglutamate synthase [EC:6.3.2.12 6.3.2.17]"     </span></span>
<span><span class="co">##  [4] "K00036  G6PD, zwf, glucose-6-phosphate 1-dehydrogenase [EC:1.1.1.49 1.1.1.363]"                </span></span>
<span><span class="co">##  [5] "K02500  hisF, imidazole glycerol-phosphate synthase subunit HisF [EC:4.3.2.10]"                </span></span>
<span><span class="co">##  [6] "K01496  hisI, phosphoribosyl-AMP cyclohydrolase [EC:3.5.4.19]"                                 </span></span>
<span><span class="co">##  [7] "K06975  K06975, uncharacterized protein"                                                       </span></span>
<span><span class="co">##  [8] "K16869  lipL, octanoyl-[GcvH]:protein N-octanoyltransferase [EC:2.3.1.204]"                    </span></span>
<span><span class="co">##  [9] "K02072  metI, D-methionine transport system permease protein"                                  </span></span>
<span><span class="co">## [10] "K00969  nadD, nicotinate-nucleotide adenylyltransferase [EC:2.7.7.18]"                         </span></span>
<span><span class="co">## [11] "K00940  ndk, NME, nucleoside-diphosphate kinase [EC:2.7.4.6]"                                  </span></span>
<span><span class="co">## [12] "K06191  nrdH, glutaredoxin-like protein NrdH"                                                  </span></span>
<span><span class="co">## [13] "K03647  nrdI, protein involved in ribonucleotide reduction"                                    </span></span>
<span><span class="co">## [14] "K03979  obgE, cgtA, MTG2, GTPase [EC:3.6.5.-]"                                                 </span></span>
<span><span class="co">## [15] "K00147  proA, glutamate-5-semialdehyde dehydrogenase [EC:1.2.1.41]"                            </span></span>
<span><span class="co">## [16] "K22305  psp, phosphoserine phosphatase [EC:3.1.3.3]"                                           </span></span>
<span><span class="co">## [17] "K08300  rne, ribonuclease E [EC:3.1.26.12]"                                                    </span></span>
<span><span class="co">## [18] "K02888  RP-L21, MRPL21, rplU, large subunit ribosomal protein L21"                             </span></span>
<span><span class="co">## [19] "K03500  rsmB, sun, 16S rRNA (cytosine967-C5)-methyltransferase [EC:2.1.1.176]"                 </span></span>
<span><span class="co">## [20] "K09710  ybeB, ribosome-associated protein"                                                     </span></span>
<span><span class="co">## [21] "PNACT"                                                                                         </span></span>
<span><span class="co">## [22] "PNACT"                                                                                         </span></span>
<span><span class="co">## [23] "PNACT"                                                                                         </span></span>
<span><span class="co">## [24] "PNACT"                                                                                         </span></span>
<span><span class="co">## [25] "PNACT"                                                                                         </span></span>
<span><span class="co">## [26] "PNACT"                                                                                         </span></span>
<span><span class="co">## [27] "PNACT"                                                                                         </span></span>
<span><span class="co">## [28] "PNACT"                                                                                         </span></span>
<span><span class="co">## [29] "PNACT"                                                                                         </span></span>
<span><span class="co">## [30] "PNACT"                                                                                         </span></span>
<span><span class="co">## [31] "PNACT"                                                                                         </span></span>
<span><span class="co">## [32] "PNACT"                                                                                         </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## [[4]]</span></span>
<span><span class="co">##  [1] "K02168  betT, betS, choline/glycine/proline betaine transport protein"                               </span></span>
<span><span class="co">##  [2] "K19710  E2.7.7.53, ATP adenylyltransferase [EC:2.7.7.53]"                                            </span></span>
<span><span class="co">##  [3] "K01523  hisE, phosphoribosyl-ATP pyrophosphohydrolase [EC:3.6.1.31]"                                 </span></span>
<span><span class="co">##  [4] "K00765  hisG, ATP phosphoribosyltransferase [EC:2.4.2.17]"                                           </span></span>
<span><span class="co">##  [5] "K09762  K09762, uncharacterized protein"                                                             </span></span>
<span><span class="co">##  [6] "K02073  metQ, D-methionine transport system substrate-binding protein"                               </span></span>
<span><span class="co">##  [7] "K05571  mnhG, mrpG, multicomponent Na+:H+ antiporter subunit G"                                      </span></span>
<span><span class="co">##  [8] "K00604  MTFMT, fmt, methionyl-tRNA formyltransferase [EC:2.1.2.9]"                                   </span></span>
<span><span class="co">##  [9] "K00995  pgsA, PGS1, CDP-diacylglycerol---glycerol-3-phosphate 3-phosphatidyltransferase [EC:2.7.8.5]"</span></span>
<span><span class="co">## [10] "K03811  pnuC, nicotinamide mononucleotide transporter"                                               </span></span>
<span><span class="co">## [11] "K02335  polA, DNA polymerase I [EC:2.7.7.7]"                                                         </span></span>
<span><span class="co">## [12] "K06958  rapZ, RNase adapter protein RapZ"                                                            </span></span>
<span><span class="co">## [13] "K00794  ribH, RIB4, 6,7-dimethyl-8-ribityllumazine synthase [EC:2.5.1.78]"                           </span></span>
<span><span class="co">## [14] "K02899  RP-L27, MRPL27, rpmA, large subunit ribosomal protein L27"                                   </span></span>
<span><span class="co">## [15] "K02945  RP-S1, rpsA, small subunit ribosomal protein S1"                                             </span></span>
<span><span class="co">## [16] "K01783  rpe, RPE, ribulose-phosphate 3-epimerase [EC:5.1.3.1]"                                       </span></span>
<span><span class="co">## [17] "K00564  rsmC, 16S rRNA (guanine1207-N2)-methyltransferase [EC:2.1.1.172]"                            </span></span>
<span><span class="co">## [18] "K04564  SOD2, superoxide dismutase, Fe-Mn family [EC:1.15.1.1]"                                      </span></span>
<span><span class="co">## [19] "K00598  tam, trans-aconitate 2-methyltransferase [EC:2.1.1.144]"                                     </span></span>
<span><span class="co">## [20] "K01868  TARS, thrS, threonyl-tRNA synthetase [EC:6.1.1.3]"                                           </span></span>
<span><span class="co">## [21] "K01609  trpC, indole-3-glycerol phosphate synthase [EC:4.1.1.48]"                                    </span></span>
<span><span class="co">## [22] "K01011  TST, MPST, sseA, thiosulfate/3-mercaptopyruvate sulfurtransferase [EC:2.8.1.1 2.8.1.2]"      </span></span>
<span><span class="co">## [23] "K00876  udk, UCK, uridine kinase [EC:2.7.1.48]"                                                      </span></span>
<span><span class="co">## [24] "K03702  uvrB, excinuclease ABC subunit B"                                                            </span></span>
<span><span class="co">## [25] "K03703  uvrC, excinuclease ABC subunit C"                                                            </span></span>
<span><span class="co">## [26] "K03976  ybaK, ebsC, Cys-tRNA(Pro)/Cys-tRNA(Cys) deacylase [EC:3.1.1.-]"                              </span></span>
<span><span class="co">## [27] "K06916  zapE, cell division protein ZapE"                                                            </span></span>
<span><span class="co">## [28] "PNACT"                                                                                               </span></span>
<span><span class="co">## [29] "PNACT"                                                                                               </span></span>
<span><span class="co">## [30] "PNACT"                                                                                               </span></span>
<span><span class="co">## [31] "PNACT"                                                                                               </span></span>
<span><span class="co">## [32] "PNACT"                                                                                               </span></span>
<span><span class="co">## [33] "NONCODING"                                                                                           </span></span>
<span><span class="co">## [34] "NONCODING"                                                                                           </span></span>
<span><span class="co">## [35] "NONCODING"                                                                                           </span></span>
<span><span class="co">## [36] "PNACT"                                                                                               </span></span>
<span><span class="co">## [37] "PNACT"                                                                                               </span></span>
<span><span class="co">## [38] "PNACT"                                                                                               </span></span>
<span><span class="co">## [39] "PNACT"                                                                                               </span></span>
<span><span class="co">## [40] "PNACT"                                                                                               </span></span>
<span><span class="co">## [41] "PNACT"                                                                                               </span></span>
<span><span class="co">## [42] "PNACT"                                                                                               </span></span>
<span><span class="co">## [43] "PNACT"                                                                                               </span></span>
<span><span class="co">## [44] "PNACT"</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="methods-implemented-in-protweaver">Methods Implemented in ProtWeaver<a class="anchor" aria-label="anchor" href="#methods-implemented-in-protweaver"></a>
</h2>
<p>By default, <code>predict.ProtWeaver</code> makes an ensemble
prediction using as many individual models as it can run with the data
provided. However, users are free to use any of the individual models
without the ensemble predictor. The methods implemented are the
following:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># PHYLOGENETIC PROFILING METHODS:</span></span>
<span>  <span class="co">## P/A = Presence/Absence Profiles</span></span>
<span>  <span class="co">## Jaccard distance of P/A</span></span>
<span><span class="va">Jaccard</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'Jaccard'</span><span class="op">)</span> </span>
<span></span>
<span>  <span class="co">## Hamming distance of P/A</span></span>
<span><span class="va">Hamming</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'Hamming'</span><span class="op">)</span> </span>
<span></span>
<span>  <span class="co">## MutualInformation of P/A</span></span>
<span><span class="va">MutualInf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'MutualInformation'</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co">## Direct Coupling Analysis of P/A</span></span>
<span><span class="va">ProfDCA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'ProfDCA'</span><span class="op">)</span> </span>
<span></span>
<span>  <span class="co">## Correlation of gain/loss events on phylogeny, requires Species Tree</span></span>
<span><span class="va">Behdenna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'Behdenna'</span>, mySpeciesTree<span class="op">=</span><span class="va">exSpeciesTree</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># CO-LOCALIZATION METHODS:</span></span>
<span><span class="va">Colocalization</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'Coloc'</span><span class="op">)</span> <span class="co"># Co-localization analysis</span></span>
<span></span>
<span><span class="co"># DISTANCE MATRIX METHDOS:</span></span>
<span><span class="va">MirrorTree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'MirrorTree'</span><span class="op">)</span></span>
<span><span class="va">ContextTree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'ContextTree'</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Residue Methods: (ONLY AVAILABLE IN DEV VERSION)</span></span>
<span><span class="co">#   ## MutualInf of residues</span></span>
<span><span class="co">#ResidueMI &lt;- predict(pw, method='ResidueMI') </span></span></code></pre></div>
<style type="text/css">
.pagination {
    display: flex;
    display: -webkit-flex;
    padding-left: 0;
    list-style: none;
    justify-content: center
}
</style>
<div class="center">
<ul class="pagination pagination-lg">
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">«</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">1</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">2</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">3</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">4</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">5</a>
</li>
<li class="page-item active">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">6</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html">7</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html">»</a>
</li>
</ul>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
