<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="CompGenomicsBioc2022">
<title>Building Coevolution Networks with SynExtend • CompGenomicsBioc2022</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Building Coevolution Networks with SynExtend">
<meta property="og:description" content="CompGenomicsBioc2022">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">Coevolutionary Networks</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com/tutorials" aria-label="Homepage">
    <span class="fa fa-user"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Building Coevolution Networks with SynExtend</h1>
                        <h4 data-toc-skip class="author">Aidan
Lakshman<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;a href="mailto:ahl27@pitt.edu" class="email"&gt;ahl27@pitt.edu&lt;/a&gt;&lt;/p&gt;'><sup>1</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ahl27/CompGenomicsBioc2022/blob/HEAD/vignettes/CoevolutionNetworks.Rmd" class="external-link"><code>vignettes/CoevolutionNetworks.Rmd</code></a></small>
      <div class="d-none name"><code>CoevolutionNetworks.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="coevolutionary-analysis">Coevolutionary Analysis<a class="anchor" aria-label="anchor" href="#coevolutionary-analysis"></a>
</h2>
<p>At this point, we’ve learned how to generate gene trees for a set of
COGs. We’re now ready to look for signals of coevolution, which imply
functional associations. These methods are implemented via the
<code>ProtWeaver</code> class in <code>SynExtend</code>, which includes
many commonly used methods for detecting coevolutionary patterns. While
the previous step only generated a few gene trees, I’ve included an
<code>RData</code> file with the results if we had generated gene trees
for all COGs in the dataset.</p>
<p>Generating a <code>ProtWeaver</code> object from this is simple:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DECIPHER</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SynExtend</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">datafile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, <span class="st">'treesFromCOGs.RData'</span>, 
                        package<span class="op">=</span><span class="st">'CompGenomicsBioc2022'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">datafile</span>, verbose<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>

<span class="va">pw</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SynExtend/man/ProtWeaver.html" class="external-link">ProtWeaver</a></span><span class="op">(</span><span class="va">treesFromCOGs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">25</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">pw</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## a ProtWeaver object with 25 groups and 301 genomes.</span></code></pre>
<p>The <code>ProtWeaver</code> constuctor automatically detects the type
of data you have and adjusts available predictors accordingly. While it
functions best with a list of dendrograms for each COG, it can also run
with simple presence/absence patterns. See the documentation file for
<code>ProtWeaver</code> for more information on this functionality.</p>
<p>We’re now ready to make predictions. Predicting functional
associations is done with the <code>predict.ProtWeaver</code> S3 method.
Let’s examine possible functional associations between the first 100
COGs we have.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">preds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">preds</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## a ProtWeb object.</span>
<span class="co">##  Method used: Ensemble </span>
<span class="co">##  Number of genes: 25 </span>
<span class="co">##  Number of predictions: 325</span></code></pre>
</div>
<div class="section level2">
<h2 id="viewing-our-results">Viewing our results<a class="anchor" aria-label="anchor" href="#viewing-our-results"></a>
</h2>
<p>Notice that <code>preds</code> is a <code>ProtWeb</code> object. This
is just a simple S3 class with a pretty print method wrapping a matrix
of pairwise association scores. We can get the raw data with
<code><a href="https://rdrr.io/pkg/SynExtend/man/getProtWebData.ProtWeb.html" class="external-link">GetProtWebData()</a></code>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Subset so the output is actually readable</span>
<span class="fu"><a href="https://rdrr.io/pkg/SynExtend/man/getProtWebData.ProtWeb.html" class="external-link">GetProtWebData</a></span><span class="op">(</span><span class="va">preds</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">7</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span></code></pre></div>
<pre><code><span class="co">##           1         2         3         4         5         6         7</span>
<span class="co">## 1 1.0000000 0.4969302 0.4170682 0.4285390 0.2864437 0.5411136 0.2860832</span>
<span class="co">## 2 0.4969302 1.0000000 0.2709793 0.3777730 0.1762943 0.4110484 0.1888778</span>
<span class="co">## 3 0.4170682 0.2709793 1.0000000 0.9998540 0.1305225 0.2478214 0.1369708</span>
<span class="co">## 4 0.4285390 0.3777730 0.9998540 1.0000000 0.1077506 0.2697997 0.1141384</span>
<span class="co">## 5 0.2864437 0.1762943 0.1305225 0.1077506 1.0000000 0.2423908 0.1367688</span>
<span class="co">## 6 0.5411136 0.4110484 0.2478214 0.2697997 0.2423908 1.0000000 0.2037752</span>
<span class="co">## 7 0.2860832 0.1888778 0.1369708 0.1141384 0.1367688 0.2037752 1.0000000</span></code></pre>
<p>The <code>ProtWeb</code> class will be updated next release cycle to
include more methods, including a custom plotting function. The current
<code>plot.ProtWeb</code> S3 method implements a force-directed
embedding of the pairwise scores, but it’s a big work-in-progress. Stay
tuned for the next release cycle for more functionality regarding
<code>ProtWeb</code>.</p>
</div>
<div class="section level2">
<h2 id="methods-implemented-in-protweaver">Methods Implemented in ProtWeaver<a class="anchor" aria-label="anchor" href="#methods-implemented-in-protweaver"></a>
</h2>
<p>By default, <code>predict.ProtWeaver</code> makes an ensemble
prediction using as many individual models as it can run with the data
provided. However, users are free to use any of the individual models
without the ensemble predictor. The methods implemented are the
following:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># PHYLOGENETIC PROFILING METHODS:</span>
  <span class="co">## P/A = Presence/Absence Profiles</span>
  <span class="co">## Jaccard distance of P/A</span>
<span class="va">Jaccard</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'Jaccard'</span><span class="op">)</span> 

  <span class="co">## Hamming distance of P/A</span>
<span class="va">Hamming</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'Hamming'</span><span class="op">)</span> 

  <span class="co">## MutualInformation of P/A</span>
<span class="va">MutualInf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'MutualInformation'</span><span class="op">)</span>

  <span class="co">## Direct Coupling Analysis of P/A</span>
<span class="va">ProfDCA</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'ProfDCA'</span><span class="op">)</span> 

  <span class="co">## Correlation of gain/loss events on phylogeny, requires Species Tree</span>
<span class="va">Behdenna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'Behdenna'</span>, mySpeciesTree<span class="op">=</span><span class="va">exSpeciesTree</span><span class="op">)</span>

<span class="co"># CO-LOCALIZATION METHODS:</span>
<span class="va">Colocalization</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'Coloc'</span><span class="op">)</span> <span class="co"># Co-localization analysis</span>

<span class="co"># DISTANCE MATRIX METHDOS:</span>
<span class="va">MirrorTree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'MirrorTree'</span><span class="op">)</span>
<span class="va">ContextTree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">pw</span>, method<span class="op">=</span><span class="st">'ContextTree'</span><span class="op">)</span>

<span class="co"># Residue Methods: (ONLY AVAILABLE IN DEV VERSION)</span>
<span class="co">#   ## MutualInf of residues</span>
<span class="co">#ResidueMI &lt;- predict(pw, method='ResidueMI') </span>
<span class="co">#   ## MutualInf of phyletic changes</span>
<span class="co">#DendroMI &lt;- predict(pw, method='DendroMI')</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="thank-you">Thank you!<a class="anchor" aria-label="anchor" href="#thank-you"></a>
</h2>
<p>If you’ve made it through this entire tutorial, thank you for
following along! I hope this series was informative and useful to your
analyses. All code showcased here is actively being worked on by members
of our lab, especially the <code>ProtWeaver</code> and
<code>ProtWeb</code> functionalities. If you have any comments,
suggestions, or feature requests for <code>ProtWeaver</code>,
<code>ProtWeb</code>, or this tutorial, please feel free to either email
me at <a href="mailto:ahl27@pitt.edu" class="email">ahl27@pitt.edu</a>
or open an issue on GitHub.</p>
<style type="text/css">
.pagination {
    display: flex;
    display: -webkit-flex;
    padding-left: 0;
    list-style: none;
    justify-content: center
}
</style>
<div class="center">
<ul class="pagination pagination-lg">
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">«</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">1</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">2</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">3</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">4</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">5</a>
</li>
<li class="page-item active">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">6</a>
</li>
<li class="page-item disabled">
<a class="page-link" href="#">»</a>
</li>
</ul>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
