<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="CompGenomicsBioc2022">
<title>Conclusion • CompGenomicsBioc2022</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Conclusion">
<meta property="og:description" content="CompGenomicsBioc2022">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">3.3.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">Coevolutionary Networks</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html">Conclusion</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com/tutorials" aria-label="Homepage">
    <span class="fa fa-user"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Conclusion</h1>
                        <h4 data-toc-skip class="author">Aidan
Lakshman<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;a href="mailto:ahl27@pitt.edu" class="email"&gt;ahl27@pitt.edu&lt;/a&gt;&lt;/p&gt;'><sup>1</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ahl27/CompGenomicsBioc2022/blob/HEAD/vignettes/Conclusion.Rmd" class="external-link"><code>vignettes/Conclusion.Rmd</code></a></small>
      <div class="d-none name"><code>Conclusion.Rmd</code></div>
    </div>

    
    
<p><img src="images/PipelineOverview.png" width="100%"></p>
<p>This tutorial has walked through a complete pipeline of comparative
genomic analysis. By using this tool, we’ve been able to generate
hypotheses of functional associations between genetic regions using only
raw sequence data as input. Along the way, we’ve covered the
following:</p>
<ul>
<li>Working with sequencing data, including alignment and
visualization</li>
<li>Finding and annotating genes from sequencing data</li>
<li>Finding Clusters of Orthologous Genes (COGs)</li>
<li>Reconstructing the evolutionary history of COGs</li>
<li>Analyzing coevolutionary signal between COGs</li>
</ul>
<p>This coevolutionary signal allows us to generate hypotheses about
possible unknown functional associations between proteins. In the
previous example, we were able to correctly identify a cluster of
ureases using only sequencing data.</p>
<p>This pipeline works on both genes that have been previously
characterized and those that have not, meaning we can guide future wet
lab investigation with these predictions. Future work will investigate
ways to improve our predictions, so stay tuned to the
<code>SynExtend</code> package as it develops!</p>
<div class="section level2">
<h2 id="implementation-at-scale">Implementation at Scale<a class="anchor" aria-label="anchor" href="#implementation-at-scale"></a>
</h2>
<p>While our analyses are designed to be scalable, they are not yet fast
enough to be able to examine thousands of genomes within a single
workshop. Since we eventually need to make comparisons between every
pair of COGs, the task grows quadratically as more genomes (and thus
more COGs) are analyzed. Because of this, we chose a smaller test set to
showcase performance in this workshop.</p>
<p>However, our methods can scale to many more genomes. All of our
methods are designed to be run in parallel across a supercomputer
system. The majority of the methods have low CPU and memory
requirements, and thus can be run on small (2-4GB memory) nodes. We have
recently completed analysis of all available <em>Streptomyces</em>
genomes, which can complete in as little as an hour given sufficient
compute nodes.</p>
<p>Our pipeline of analysis looks like this:</p>
<p><img src="images/Parallelization.png" width="100%" style="display: block; margin: auto;"></p>
<div class="section level3">
<h3 id="runtime-estimations">Runtime Estimations<a class="anchor" aria-label="anchor" href="#runtime-estimations"></a>
</h3>
<p>The following is a rough estimate of runtime considerations at each
stage of the pipeline. These examples use <em>Streptomyces</em> genomes,
which are roughly 7-9 megabases. Stages marked <code>single</code> are
only run once, whereas <code>parallel</code> are run multiple times.
“Runtime” denotes total runtime if <code>single</code> and runtime per
iteration if <code>parallel</code>. More details on each runtime
component are available on the corresponding pages.</p>
<p>For scaling, <span class="math inline">\(G\)</span> is the number of
Genomes, <span class="math inline">\(g\)</span> is the number of genes,
<span class="math inline">\(C\)</span> is the number of COGs, <span class="math inline">\(g_C\)</span> is the number of genes in a
particular COG.</p>
<p><code>NJ</code>, <code>MP</code>, and <code>ML</code> denote Neighbor
Joining, Maximum Parsimony, and Maximum Likelihood (respectively).
<code>ML</code> is more accurate than <code>MP</code> at the cost of
increased runtime, and the same is true of <code>MP</code> and
<code>NJ</code>. See the section on <a href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Phylogenetics</a>
for more info.</p>
<table class="table">
<colgroup>
<col width="33%">
<col width="17%">
<col width="17%">
<col width="17%">
<col width="14%">
</colgroup>
<thead><tr class="header">
<th align="center">Pipeline Stage</th>
<th align="center">Scaling</th>
<th align="center">Single or Parallel</th>
<th align="center">
<em>Streptomyces</em> Example</th>
<th align="center">Example Runtime</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">Get Genomes</td>
<td align="center"><span class="math inline">\(O(G)\)</span></td>
<td align="center">Single</td>
<td align="center">300 Genomes</td>
<td align="center">~60 sec</td>
</tr>
<tr class="even">
<td align="center">Find Genes</td>
<td align="center"><span class="math inline">\(O(G)\)</span></td>
<td align="center">Parallel</td>
<td align="center">Single Genome</td>
<td align="center">~15 min</td>
</tr>
<tr class="odd">
<td align="center">Annotate Genes</td>
<td align="center"><span class="math inline">\(O(G)\)</span></td>
<td align="center">Parallel</td>
<td align="center">Single Genome</td>
<td align="center">~20 min</td>
</tr>
<tr class="even">
<td align="center">Pairwise Orthology</td>
<td align="center"><span class="math inline">\(O(G^2)\)</span></td>
<td align="center">Parallel</td>
<td align="center">Pair of Genomes</td>
<td align="center">~10 min</td>
</tr>
<tr class="odd">
<td align="center">Ortholog Pairs to COGs</td>
<td align="center"><span class="math inline">\(O(g^2)\)</span></td>
<td align="center">Single</td>
<td align="center">All pairs from 300 Genomes</td>
<td align="center">~5 min</td>
</tr>
<tr class="even">
<td align="center">Align and Make Tree</td>
<td align="center"><span class="math inline">\(O(C)\)</span></td>
<td align="center">Parallel</td>
<td align="center">Single COG</td>
<td align="center">&lt;1 sec (NJ) <br> ~0.5 hr (MP) <br> ~13 hr
(ML)</td>
</tr>
<tr class="odd">
<td align="center">Calculate Coevolution</td>
<td align="center"><span class="math inline">\(O(C^2)\)</span></td>
<td align="center">Parallel</td>
<td align="center">Pair of COGs</td>
<td align="center">~5 sec</td>
</tr>
<tr class="even">
<td align="center">(Optimal) Total Runtime</td>
<td align="center">—</td>
<td align="center">—</td>
<td align="center">300 Genomes</td>
<td align="center">~1 hr (NJ) <br> ~1.5 hr (MP) <br> ~14 hr (ML)</td>
</tr>
</tbody>
</table>
<p>For this example, 300 <em>Streptomyces</em> genomes resulted in ~2.2
million genes, resulting in ~65k COGs. Overall runtime depends on the
phylogenetic reconstruction algorithm used. Tree construction is
performed using amino acid alignments.</p>
<p>Users interested in deploying these analyses at scale are encouraged
to contact our lab for more information!</p>
</div>
</div>
<div class="section level2">
<h2 id="thank-you">Thank you!<a class="anchor" aria-label="anchor" href="#thank-you"></a>
</h2>
<p>If you’ve made it through this entire tutorial, thank you for
following along! I hope this series was informative and useful to your
analyses. All code showcased here is actively being worked on by members
of our lab, especially the <code>ProtWeaver</code> and
<code>ProtWeb</code> functionalities. If you have any comments,
suggestions, or feature requests for <code>ProtWeaver</code>,
<code>ProtWeb</code>, or this tutorial, please feel free to either email
me at <a href="mailto:ahl27@pitt.edu" class="email">ahl27@pitt.edu</a>
or open an issue on GitHub.</p>
</div>
<div class="section level2">
<h2 id="other-resources">Other Resources<a class="anchor" aria-label="anchor" href="#other-resources"></a>
</h2>
<p>If you’re interested in learning more about me, our lab, or
phylogenetics, check out these resources:</p>
<ul>
<li><a href="https://ahl27.com" class="external-link">My website</a></li>
<li><a href="https://wrightlabscience.com" class="external-link">The Wright Lab</a></li>
<li><a href="https://www.ahl27.com/IntroPhylogenetics/" class="external-link">Intro to
Phylogenetics</a></li>
<li><a href="http://www2.decipher.codes/" class="external-link">More DECIPHER
Tutorials</a></li>
</ul>
<style type="text/css">
.pagination {
    display: flex;
    display: -webkit-flex;
    padding-left: 0;
    list-style: none;
    justify-content: center
}
</style>
<div class="center">
<ul class="pagination pagination-lg">
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">«</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">1</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">2</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">3</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">4</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">5</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">6</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html">7</a>
</li>
<li class="page-item disabled">
<a class="page-link" href="#">»</a>
</li>
</ul>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
