<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="CompGenomicsBioc2022">
<title>Constructing Phylogenies with DECIPHER • CompGenomicsBioc2022</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Constructing Phylogenies with DECIPHER">
<meta property="og:description" content="CompGenomicsBioc2022">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">Coevolutionary Networks</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com/tutorials" aria-label="Homepage">
    <span class="fa fa-user"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Constructing Phylogenies with DECIPHER</h1>
                        <h4 data-toc-skip class="author">Aidan
Lakshman<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;a href="mailto:ahl27@pitt.edu" class="email"&gt;ahl27@pitt.edu&lt;/a&gt;&lt;/p&gt;'><sup>1</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ahl27/CompGenomicsBioc2022/blob/HEAD/vignettes/ConstructingPhylogenies.Rmd" class="external-link"><code>vignettes/ConstructingPhylogenies.Rmd</code></a></small>
      <div class="d-none name"><code>ConstructingPhylogenies.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="phylogenetic-reconstruction">Phylogenetic Reconstruction<a class="anchor" aria-label="anchor" href="#phylogenetic-reconstruction"></a>
</h2>
<p>We’ve now learned how to find Clusters of Orthologous Genes (COGs)
from a set of sequences. The last thing we need for our final analysis
are phylogenetic reconstructions of each gene cluster. In this step,
we’ll build phylogenies for our COGs using the new
<code><a href="https://rdrr.io/pkg/DECIPHER/man/TreeLine.html" class="external-link">TreeLine()</a></code> function introduced in the latest update of
<code>DECIPHER</code>.</p>
<p>In the last step, we constructed COGs for a small subset of our data
to learn the algorithmic flow of that portion of the pipeline. Since
analyzing the entire dataset is time-prohibitive for this workshop,
we’ll work with a pre-calculated <code>RData</code> file containing the
results we would have gotten if we had run the same calculation on the
entire dataset.</p>
<p><em>Small note:</em> the setup for this page includes a helper
function to plot the phylogenies as a circular plot. You can plot them
just as well with <code>plot</code>, I just prefer this visualization.
You can also use <code>plottype='unrooted'</code> for a different
visualization, though labels are often hard to read in that
visualization.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DECIPHER</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/source.html" class="external-link">source</a></span><span class="op">(</span><span class="st">"TreeLine.R"</span><span class="op">)</span> <span class="co"># Shouldn't be necessary if using dev version</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">datafile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, <span class="st">'phyloInputData.RData'</span>, 
                        package<span class="op">=</span><span class="st">'CompGenomicsBioc2022'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="va">datafile</span>, verbose<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="co"># Should load 'AlignedSequences'</span></code></pre></div>
<pre><code><span class="co">## Loading objects:</span>
<span class="co">##   AlignedSequences</span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Helper plotting function</span>
<span class="va">plot_tree_unrooted</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">dend</span>, <span class="va">main</span>, <span class="va">plottype</span><span class="op">=</span><span class="st">'fan'</span><span class="op">)</span><span class="op">{</span>
  <span class="kw">if</span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va"><a href="http://ape-package.ird.fr/" class="external-link">ape</a></span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
    <span class="va">tf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/WriteDendrogram.html" class="external-link">WriteDendrogram</a></span><span class="op">(</span><span class="va">dend</span>, file<span class="op">=</span><span class="va">tf</span>, quoteLabels<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
    <span class="va">predTree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html" class="external-link">read.tree</a></span><span class="op">(</span><span class="va">tf</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">predTree</span>, <span class="va">plottype</span>, main<span class="op">=</span><span class="va">main</span><span class="op">)</span>
  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">dend</span>, main<span class="op">=</span><span class="va">main</span><span class="op">)</span>
  <span class="op">}</span>
<span class="op">}</span>

<span class="va">testSeqs</span> <span class="op">&lt;-</span> <span class="va">AlignedSequences</span>

<span class="va">treeCOG1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/TreeLine.html" class="external-link">TreeLine</a></span><span class="op">(</span><span class="va">testSeqs</span>, method<span class="op">=</span><span class="st">'MP'</span>, reconstruct<span class="op">=</span><span class="cn">TRUE</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="fu">plot_tree_unrooted</span><span class="op">(</span><span class="va">treeCOG1</span>, main<span class="op">=</span><span class="st">'MP'</span><span class="op">)</span></code></pre></div>
<p><img src="ConstructingPhylogenies_files/figure-html/unnamed-chunk-2-1.png" width="1000px"></p>
<p>That’s all we need to construct a quick phylogenenetic tree in R!
I’ve set <code>reconstruct=TRUE</code> to have <code>TreeLine</code>
automatically reconstruct ancestral states at each node.</p>
<p>However, <code>TreeLine</code> incorporates a wealth of functionality
past what is detailed here. In fact, this tree isn’t even the best tree
we can make! Let’s take a look at all the new features included in the
<code><a href="https://rdrr.io/pkg/DECIPHER/man/TreeLine.html" class="external-link">TreeLine()</a></code> function.</p>
</div>
<div class="section level2">
<h2 id="tree-building-methods">Tree-Building Methods<a class="anchor" aria-label="anchor" href="#tree-building-methods"></a>
</h2>
<p>Our first example used <code>method='MP'</code>, meaning it
constructed a phyletic tree using a Maximum Parsimony method. However,
many more methods are available. <code><a href="https://rdrr.io/pkg/DECIPHER/man/TreeLine.html" class="external-link">TreeLine()</a></code> implements
Maximum Parsimony (<code>MP</code>), Neighbor-Joining (<code>NJ</code>),
Ultrametric (<code>complete</code>, <code>single</code>,
<code>UPGMA</code>, <code>WPGMA</code>), and Maximum Likelihood
(<code>ML</code>) methods. Each of these have different strengths,
weaknesses, and assumptions. While an in-depth look at the difference
between these methods is outside the scope of this tutorial, I recently
published <a href="https://ahl27.com/IntroPhylogenetics" class="external-link">another
tutorial</a> that goes into the mathematics of how these methods
work.</p>
<p>Example code for each of these:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Maximum Parsimony</span>
<span class="va">treeMP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/TreeLine.html" class="external-link">TreeLine</a></span><span class="op">(</span>myXStringSet<span class="op">=</span><span class="va">testSeqs</span>, method<span class="op">=</span><span class="st">'MP'</span><span class="op">)</span>
<span class="fu">plot_tree_unrooted</span><span class="op">(</span><span class="va">treeMP</span>, main<span class="op">=</span><span class="st">'MP'</span><span class="op">)</span></code></pre></div>
<p><img src="ConstructingPhylogenies_files/figure-html/unnamed-chunk-3-1.png" width="1000px"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Neighbor-Joining</span>
<span class="va">distMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/DistanceMatrix.html" class="external-link">DistanceMatrix</a></span><span class="op">(</span><span class="va">testSeqs</span>, correction<span class="op">=</span><span class="st">'Jukes-Cantor'</span><span class="op">)</span>
<span class="va">treeNJ</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/TreeLine.html" class="external-link">TreeLine</a></span><span class="op">(</span>myDistMatrix<span class="op">=</span><span class="va">distMatrix</span>, method<span class="op">=</span><span class="st">'NJ'</span><span class="op">)</span>
<span class="fu">plot_tree_unrooted</span><span class="op">(</span><span class="va">treeNJ</span>, main<span class="op">=</span><span class="st">'NJ'</span><span class="op">)</span></code></pre></div>
<p><img src="ConstructingPhylogenies_files/figure-html/unnamed-chunk-4-1.png" width="1000px"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># UPGMA tree</span>
<span class="va">distMatrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/DistanceMatrix.html" class="external-link">DistanceMatrix</a></span><span class="op">(</span><span class="va">testSeqs</span>, correction<span class="op">=</span><span class="st">'Jukes-Cantor'</span><span class="op">)</span>
<span class="va">treeUltra</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/TreeLine.html" class="external-link">TreeLine</a></span><span class="op">(</span>myDistMatrix<span class="op">=</span><span class="va">distMatrix</span>, method<span class="op">=</span><span class="st">'UPGMA'</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">treeUltra</span>, main<span class="op">=</span><span class="st">'UPGMA'</span><span class="op">)</span></code></pre></div>
<p><img src="ConstructingPhylogenies_files/figure-html/unnamed-chunk-5-1.png" width="1000px"></p>
<p>Maximum-Likelihood trees are the most accurate, but also the slowest
to create. This method iteratively maximizes the likelihood of the tree
under a given sequence evolution model for a set of aligned sequences.
In the interest of time, this demo will set the <code>maxTime</code>
argument to prevent the algorithm from taking too long.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Maximum Likehood</span>
<span class="co">#   - Max runtime is set here to 30sec, default is as long as it takes</span>
<span class="co">#   - maxTime expresses time in HOURS, not sec/min</span>
<span class="co">#   - Note that method='ML' is the default setting</span>
<span class="va">treeML</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/TreeLine.html" class="external-link">TreeLine</a></span><span class="op">(</span><span class="va">testSeqs</span>, maxTime<span class="op">=</span><span class="op">(</span><span class="fl">30</span><span class="op">/</span><span class="fl">3600</span><span class="op">)</span><span class="op">)</span>
<span class="fu">plot_tree_unrooted</span><span class="op">(</span><span class="va">treeML</span>, main<span class="op">=</span><span class="st">'Maximum-Likelihood'</span><span class="op">)</span></code></pre></div>
<p><img src="ConstructingPhylogenies_files/figure-html/unnamed-chunk-6-1.png" width="1000px"></p>
</div>
<div class="section level2">
<h2 id="sequence-evolution-models">Sequence Evolution Models<a class="anchor" aria-label="anchor" href="#sequence-evolution-models"></a>
</h2>
<p>One question you’re probably asking is, “What is this ‘given sequence
evolution model’?” That’s an excellent question–by default,
<code><a href="https://rdrr.io/pkg/DECIPHER/man/TreeLine.html" class="external-link">TreeLine()</a></code> will test a variety of sequence evolution models
and pick the one that works best for your data. This means that you
shouldn’t typically have to worry about which model to use.</p>
<p>However, what if we wanted to explicitly pick a certain model? What
if we wanted to exclude a handful of models? Or what if we’re just
curious what models are even being tested?</p>
<p>Fret not, for there is a solution. Models are passed to
<code><a href="https://rdrr.io/pkg/DECIPHER/man/TreeLine.html" class="external-link">TreeLine()</a></code> as a list with two named entries,
<code>$Nucleotide</code> and <code>$Protein</code>. To look at the
default models tested, simply print out the <code>MODELS</code> variable
included from <code>DECIPHER</code>:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">DECIPHER</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/DECIPHER/man/MODELS.html" class="external-link">MODELS</a></span></code></pre></div>
<pre><code><span class="co">## $Nucleotide</span>
<span class="co">##  [1] "JC69"     "JC69+G4"  "K80"      "K80+G4"   "F81"      "F81+G4"  </span>
<span class="co">##  [7] "HKY85"    "HKY85+G4" "T92"      "T92+G4"   "TN93"     "TN93+G4" </span>
<span class="co">## [13] "SYM"      "SYM+G4"   "GTR"      "GTR+G4"  </span>
<span class="co">## </span>
<span class="co">## $Protein</span>
<span class="co">##  [1] "AB"               "AB+G4"            "BLOSUM62"         "BLOSUM62+G4"     </span>
<span class="co">##  [5] "cpREV"            "cpREV+G4"         "cpREV64"          "cpREV64+G4"      </span>
<span class="co">##  [9] "Dayhoff"          "Dayhoff+G4"       "DCMut-Dayhoff"    "DCMut-Dayhoff+G4"</span>
<span class="co">## [13] "DCMut-JTT"        "DCMut-JTT+G4"     "DEN"              "DEN+G4"          </span>
<span class="co">## [17] "FLAVI"            "FLAVI+G4"         "FLU"              "FLU+G4"          </span>
<span class="co">## [21] "gcpREV"           "gcpREV+G4"        "HIVb"             "HIVb+G4"         </span>
<span class="co">## [25] "HIVw"             "HIVw+G4"          "JTT"              "JTT+G4"          </span>
<span class="co">## [29] "LG"               "LG+G4"            "MtArt"            "MtArt+G4"        </span>
<span class="co">## [33] "mtDeu"            "mtDeu+G4"         "mtInv"            "mtInv+G4"        </span>
<span class="co">## [37] "mtMam"            "mtMam+G4"         "mtMet"            "mtMet+G4"        </span>
<span class="co">## [41] "mtOrt"            "mtOrt+G4"         "mtREV"            "mtREV+G4"        </span>
<span class="co">## [45] "mtVer"            "mtVer+G4"         "MtZoa"            "MtZoa+G4"        </span>
<span class="co">## [49] "PMB"              "PMB+G4"           "Q.bird"           "Q.bird+G4"       </span>
<span class="co">## [53] "Q.insect"         "Q.insect+G4"      "Q.LG"             "Q.LG+G4"         </span>
<span class="co">## [57] "Q.mammal"         "Q.mammal+G4"      "Q.pfam"           "Q.pfam+G4"       </span>
<span class="co">## [61] "Q.plant"          "Q.plant+G4"       "Q.yeast"          "Q.yeast+G4"      </span>
<span class="co">## [65] "rtREV"            "rtREV+G4"         "stmtREV"          "stmtREV+G4"      </span>
<span class="co">## [69] "VT"               "VT+G4"            "WAG"              "WAG+G4"          </span>
<span class="co">## [73] "WAGstar"          "WAGstar+G4"</span></code></pre>
<p>Nucleotide models include classic names like Jukes-Cantor
(<code>JC69</code>) and Felsenstein 1981 (<code>F81</code>), as well as
many others. The amino acid substitution models contain a mixture of
general models (e.g. <code>BLOSUM62</code>, <code>Dayhoff</code>), as
well as models tailored towards specific organisms
(e.g. <code>Q.insect</code>, <code>HIVb</code>).</p>
<p>To use a single model, simply create a list matching the structure of
<code>MODELS</code>, just with the models you want to include. To
exclude certain models, copy <code>MODELS</code> and remove the entries
you don’t want.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Using a specific set of models</span>
<span class="va">mySpecificModel</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Nucleotide<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'JC69'</span>, <span class="st">'HKY85'</span><span class="op">)</span><span class="op">)</span>
<span class="va">specTree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/TreeLine.html" class="external-link">TreeLine</a></span><span class="op">(</span><span class="va">testSeqs</span>, model<span class="op">=</span><span class="va">mySpecificModel</span>, maxTime<span class="op">=</span><span class="op">(</span><span class="fl">30</span><span class="op">/</span><span class="fl">3600</span><span class="op">)</span><span class="op">)</span>
<span class="fu">plot_tree_unrooted</span><span class="op">(</span><span class="va">specTree</span>, main<span class="op">=</span><span class="st">'Specific Set ML'</span><span class="op">)</span></code></pre></div>
<p><img src="ConstructingPhylogenies_files/figure-html/unnamed-chunk-8-1.png" width="1000px"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Excluding Specific Models</span>
<span class="va">myExcludedModel</span> <span class="op">&lt;-</span> <span class="fu">DECIPHER</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/DECIPHER/man/MODELS.html" class="external-link">MODELS</a></span>
<span class="va">myExcludedModel</span><span class="op">$</span><span class="va">Protein</span> <span class="op">&lt;-</span> <span class="cn">NULL</span> <span class="co"># Remove all protein models</span>
<span class="va">myExcludedModel</span><span class="op">$</span><span class="va">Nucleotide</span> <span class="op">&lt;-</span> <span class="va">myExcludedModel</span><span class="op">$</span><span class="va">Nucleotide</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span> <span class="co"># Remove JC69</span>
<span class="va">exclTree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/TreeLine.html" class="external-link">TreeLine</a></span><span class="op">(</span><span class="va">testSeqs</span>, model<span class="op">=</span><span class="va">myExcludedModel</span>, maxTime<span class="op">=</span><span class="op">(</span><span class="fl">30</span><span class="op">/</span><span class="fl">3600</span><span class="op">)</span><span class="op">)</span>
<span class="fu">plot_tree_unrooted</span><span class="op">(</span><span class="va">exclTree</span>, main<span class="op">=</span><span class="st">'Excluded Set ML'</span><span class="op">)</span></code></pre></div>
<p><img src="ConstructingPhylogenies_files/figure-html/unnamed-chunk-8-2.png" width="1000px"></p>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>That’s all we need to know how to do to generate phylogenies for our
dataset. In order to conduct our final coevolutionary analysis, we’re
going to need to build a gene tree for all of our COGs. I’ve precomputed
these trees for us, so we can load those in in the next step without
having to worry about long runtimes.</p>
<p>There are a few other parameters I didn’t mention in this writeup.
The most significant one for our use is <code>reconstruct=TRUE</code>,
as this reconstructs ancestral states (which will be important for later
analyses). There are also additional arguments for multiprocessing
(<code>processors=1</code>), using Laguerre quadrature for likelihoods
(<code>quadrature=T/F</code>), switching the type of information
criterion for ML trees
(<code>informationCriterion=c('AICc', 'BIC')</code>), and many others.
See the documentation page for more information on these–for now, we’ll
continue on to our final goal.</p>
<style type="text/css">
.pagination {
    display: flex;
    display: -webkit-flex;
    padding-left: 0;
    list-style: none;
    justify-content: center
}
</style>
<div class="center">
<ul class="pagination pagination-lg">
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">«</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">1</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">2</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">3</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">4</a>
</li>
<li class="page-item active">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">5</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">6</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">»</a>
</li>
</ul>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
