<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="CompGenomicsBioc2022">
<title>Gene Calling and Annotation with DECIPHER • CompGenomicsBioc2022</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Gene Calling and Annotation with DECIPHER">
<meta property="og:description" content="CompGenomicsBioc2022">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">Coevolutionary Networks</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html">Conclusion</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com/tutorials" aria-label="Homepage">
    <span class="fa fa-user"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Gene Calling and Annotation with DECIPHER</h1>
                        <h4 data-toc-skip class="author">Aidan
Lakshman<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;a href="mailto:ahl27@pitt.edu" class="email"&gt;ahl27@pitt.edu&lt;/a&gt;&lt;/p&gt;'><sup>1</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ahl27/CompGenomicsBioc2022/blob/HEAD/vignettes/GeneCallingAnnotation.Rmd" class="external-link"><code>vignettes/GeneCallingAnnotation.Rmd</code></a></small>
      <div class="d-none name"><code>GeneCallingAnnotation.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="gene-calling-and-annotation">Gene Calling and Annotation<a class="anchor" aria-label="anchor" href="#gene-calling-and-annotation"></a>
</h2>
<p>At this point, we’ve learned how to read in some genomic data, and
have gained some basic familiarity working with it. The next step in our
pipeline is to take a set of genomes, identify the coding regions in
them, and predict the function of as many genetic regions as we can.
We’ll start off by identifying the genes themselves.</p>
</div>
<div class="section level2">
<h2 id="finding-genes">Finding Genes<a class="anchor" aria-label="anchor" href="#finding-genes"></a>
</h2>
<p>In the interest of time, we’re going to be using plasmid sequences
from a dataset of Archaeal genomes. Plasmids are small enough that we
can quickly execute our queries. We’ll begin by reading in the data from
a <code>.fasta</code> file, as we did in the previous section.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DECIPHER</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/source.html" class="external-link">source</a></span><span class="op">(</span><span class="st">"TreeLine.R"</span><span class="op">)</span> <span class="co"># Shouldn't be necessary if using dev version</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load a single complete Micrococcus genome</span></span>
<span><span class="va">datafile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, <span class="st">'GeneCalling'</span>, <span class="st">'SingleSeq.fa.gz'</span>, </span>
<span>                        package<span class="op">=</span><span class="st">'CompGenomicsBioc2022'</span><span class="op">)</span></span>
<span><span class="va">dnaGenome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readDNAStringSet</a></span><span class="op">(</span><span class="va">datafile</span><span class="op">)</span></span>
<span><span class="va">aaGenome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/translate.html" class="external-link">translate</a></span><span class="op">(</span><span class="va">dnaGenome</span><span class="op">)</span> </span></code></pre></div>
<p>Next, we’re going to identify the genes in the first sequence in our
sequence set. We’ll mark this as our ‘genome’ (although it’s really a
plasmid), and predict genes within it using <code><a href="https://rdrr.io/pkg/DECIPHER/man/FindGenes.html" class="external-link">FindGenes()</a></code>
from the <code>DECIPHER</code> package.</p>
<p><code><a href="https://rdrr.io/pkg/DECIPHER/man/FindGenes.html" class="external-link">FindGenes()</a></code> returns a <code>Genes</code> object with
information on where genes start and end in the genome. We can then
extract the sequences corresponding to each gene using the
<code><a href="https://rdrr.io/pkg/DECIPHER/man/ExtractGenes.html" class="external-link">ExtractGenes()</a></code> function.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">geneLocs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/FindGenes.html" class="external-link">FindGenes</a></span><span class="op">(</span><span class="va">dnaGenome</span><span class="op">)</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/ExtractGenes.html" class="external-link">ExtractGenes</a></span><span class="op">(</span><span class="va">geneLocs</span>, <span class="va">dnaGenome</span>, type<span class="op">=</span><span class="st">'AAStringSet'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># `Genes` object</span></span>
<span><span class="va">geneLocs</span></span></code></pre></div>
<pre><code><span><span class="co">## Genes object of size 2,195 specifying:</span></span>
<span><span class="co">## 2,195 protein coding genes from 99 to 4,653 nucleotides.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Index Strand Begin   End TotalScore ... Gene</span></span>
<span><span class="co">## 1     1      0   256  1803     376.25 ...    1</span></span>
<span><span class="co">## 2     1      0  2554  3669     262.68 ...    1</span></span>
<span><span class="co">## 3     1      0  3691  4905     225.54 ...    1</span></span>
<span><span class="co">## 4     1      0  4898  5461      71.87 ...    1</span></span>
<span><span class="co">## 5     1      0  5819  7981     557.28 ...    1</span></span>
<span><span class="co">## 6     1      0  8050 10746     695.28 ...    1</span></span>
<span><span class="co">## ... with 2,189 more rows.</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Sequences corresponding to each gene</span></span>
<span><span class="va">genes</span></span></code></pre></div>
<pre><code><span><span class="co">## AAStringSet object of length 2195:</span></span>
<span><span class="co">##        width seq</span></span>
<span><span class="co">##    [1]   516 MVADQAVLSSWRSVVGSLEDDARVSARLMGFV...RKIRELMAERRTIYNQVTELTNEIKRKQRGA*</span></span>
<span><span class="co">##    [2]   372 MKFTVERDILTDAVSWAARSLSPRPPVPVLSG...SAPKPALLTGVNQEDGVVSDYRYLVMPVRIA*</span></span>
<span><span class="co">##    [3]   405 MYLSHLTVADFRSYRWADLELTPGSTVLLGAN...RQDAEGSEAVAAAAPVEGDIREPRREGGADG*</span></span>
<span><span class="co">##    [4]   188 MAETPAPFEPDRPDLALVQLRRVREAARERGE...TRIEVAGPQAPSWRKGPRTVRGGRGPRDTYG*</span></span>
<span><span class="co">##    [5]   721 MVDAMPENPAEEPTAASAAPNPEAVPDAVGQP...DAVFSMLMGEDVESRRTFIQQNAKDIRFLDV*</span></span>
<span><span class="co">##    ...   ... ...</span></span>
<span><span class="co">## [2191]   195 MSAENVTPAPEAEDAVVETPAGQGSRVAEQDD...KIVHDVVADAGLTSESEGEGARRHVVISADD*</span></span>
<span><span class="co">## [2192]   333 MFEAILSPFRWLMSWLLGAFHSILEFAGLPAD...QAGRGGAVLNGEVVRQSGQRVQPQRKNRRRK*</span></span>
<span><span class="co">## [2193]   123 MTVTPSPFVVLEPSREWGPLRALPSALLAGLL...PPGHRRWPPGRQPRILALNHPPIPPDLPQED*</span></span>
<span><span class="co">## [2194]   133 MLPRDRRVRTPAEFRHLGRTGTRAGRRTVVVS...AEADYALLRRETVGALGKALKPHLPAASEHA*</span></span>
<span><span class="co">## [2195]    46 MTKRTFQPNNRRRARKHGFRARMRTRAGRAILSARRGKNRAELSA*</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="removing-non-coding-regions">Removing Non-Coding Regions<a class="anchor" aria-label="anchor" href="#removing-non-coding-regions"></a>
</h2>
<p><code><a href="https://rdrr.io/pkg/DECIPHER/man/FindGenes.html" class="external-link">FindGenes()</a></code> finds the genes themselves, but these may
not encode coding regions. We’re more interested in the regions that are
actually translated into proteins, since these are what we’ll try to
annotate later. For this, we’ll use the <code><a href="https://rdrr.io/pkg/DECIPHER/man/FindNonCoding.html" class="external-link">FindNonCoding()</a></code>
function.</p>
<p><code><a href="https://rdrr.io/pkg/DECIPHER/man/FindNonCoding.html" class="external-link">FindNonCoding()</a></code> is used with three main datafiles
depending on the data to analyze:</p>
<ul>
<li>
<code>data("NonCodingRNA_Archaea")</code> for Archaeal data</li>
<li>
<code>data("NonCodingRNA_Bacteria")</code> for Bacterial data</li>
<li>
<code>data("NonCodingRNA_Eukarya")</code> for Eukaryotic data</li>
</ul>
<p>These include pretrained models with common non-coding patterns for
the relevant domain of life. If these pretrained models are insuffient,
you can train your own dataset using <code><a href="https://rdrr.io/pkg/DECIPHER/man/LearnNonCoding.html" class="external-link">LearnNonCoding()</a></code>
(though this is outside the scope of this workshop).</p>
<p>Once we’ve run <code><a href="https://rdrr.io/pkg/DECIPHER/man/FindNonCoding.html" class="external-link">FindNonCoding()</a></code>, we can use
<code><a href="https://rdrr.io/pkg/DECIPHER/man/ExtractGenes.html" class="external-link">ExtractGenes()</a></code> as before to pull out the coding regions.
This will result in more accurate gene calling than just running
<code><a href="https://rdrr.io/pkg/DECIPHER/man/FindGenes.html" class="external-link">FindGenes()</a></code> directly.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"NonCodingRNA_Bacteria"</span><span class="op">)</span></span>
<span><span class="va">ncRNA</span> <span class="op">&lt;-</span> <span class="va">NonCodingRNA_Bacteria</span></span>
<span></span>
<span><span class="va">geneticRegions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/FindNonCoding.html" class="external-link">FindNonCoding</a></span><span class="op">(</span><span class="va">ncRNA</span>, <span class="va">dnaGenome</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Find annotations of noncoding regions</span></span>
<span><span class="va">annotations</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">geneticRegions</span>, <span class="st">"annotations"</span><span class="op">)</span></span>
<span><span class="va">geneMatches</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">geneticRegions</span><span class="op">[</span>,<span class="st">"Gene"</span><span class="op">]</span>, <span class="va">annotations</span><span class="op">)</span></span>
<span><span class="va">noncodingAnnots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">annotations</span><span class="op">)</span><span class="op">[</span><span class="va">geneMatches</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Find Genes</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/FindGenes.html" class="external-link">FindGenes</a></span><span class="op">(</span><span class="va">dnaGenome</span>, includeGenes<span class="op">=</span><span class="va">geneticRegions</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># What noncoding regions have we found and successfully annotated?</span></span>
<span><span class="va">noncodingAnnots</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##            5'_ureB-RF02514            Flavo_1-RF01705 </span></span>
<span><span class="co">##                          1                          1 </span></span>
<span><span class="co">##     FMN_Riboswitch-RF00050 Glycine_Riboswitch-RF00504 </span></span>
<span><span class="co">##                          1                          1 </span></span>
<span><span class="co">##    RNase_P_class_A-RF00010           SmallSRP-RF00169 </span></span>
<span><span class="co">##                          1                          1 </span></span>
<span><span class="co">##              tmRNA-RF00023                   tRNA-Asn </span></span>
<span><span class="co">##                          1                          1 </span></span>
<span><span class="co">##                   tRNA-Asp                   tRNA-Cys </span></span>
<span><span class="co">##                          1                          1 </span></span>
<span><span class="co">##                   tRNA-His                   tRNA-Phe </span></span>
<span><span class="co">##                          1                          1 </span></span>
<span><span class="co">##                   tRNA-Trp                   tRNA-Tyr </span></span>
<span><span class="co">##                          1                          1 </span></span>
<span><span class="co">##           rRNA_16S-RF00177           rRNA_23S-RF02541 </span></span>
<span><span class="co">##                          2                          2 </span></span>
<span><span class="co">##            rRNA_5S-RF00001     SAM_Riboswitch-RF00162 </span></span>
<span><span class="co">##                          2                          2 </span></span>
<span><span class="co">##     TPP_Riboswitch-RF00059                   tRNA-Gln </span></span>
<span><span class="co">##                          2                          2 </span></span>
<span><span class="co">##                   tRNA-Lys                   tRNA-Met </span></span>
<span><span class="co">##                          2                          2 </span></span>
<span><span class="co">##                   tRNA-Val                   tRNA-Glu </span></span>
<span><span class="co">##                          2                          3 </span></span>
<span><span class="co">##                   tRNA-Ile                   tRNA-Pro </span></span>
<span><span class="co">##                          3                          3 </span></span>
<span><span class="co">##                   tRNA-Thr                   tRNA-Ala </span></span>
<span><span class="co">##                          3                          4 </span></span>
<span><span class="co">##                   tRNA-Arg                   tRNA-Gly </span></span>
<span><span class="co">##                          4                          4 </span></span>
<span><span class="co">##                   tRNA-Ser                   tRNA-Leu </span></span>
<span><span class="co">##                          4                          5</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Genes in the genome</span></span>
<span><span class="va">genes</span> </span></code></pre></div>
<pre><code><span><span class="co">## Genes object of size 2,274 specifying:</span></span>
<span><span class="co">## 2,216 protein coding genes from 75 to 4,653 nucleotides.</span></span>
<span><span class="co">## 58 non-coding RNAs from 73 to 3,088 nucleotides.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##   Index Strand Begin   End TotalScore ... Gene</span></span>
<span><span class="co">## 1     1      0   256  1803     365.17 ...    1</span></span>
<span><span class="co">## 2     1      0  2554  3669     254.93 ...    1</span></span>
<span><span class="co">## 3     1      0  3691  4905     227.03 ...    1</span></span>
<span><span class="co">## 4     1      0  4898  5461      75.07 ...    1</span></span>
<span><span class="co">## 5     1      0  5819  7981     543.99 ...    1</span></span>
<span><span class="co">## 6     1      0  8050 10746     681.74 ...    1</span></span>
<span><span class="co">## ... with 2,268 more rows.</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="classification-with-idtaxa">Classification with <code>IDTAXA</code><a class="anchor" aria-label="anchor" href="#classification-with-idtaxa"></a>
</h2>
<p>We now have a set of coding regions. Our last step for this section
is to try to annotate their function. This functionality is done with
<code><a href="https://rdrr.io/pkg/DECIPHER/man/IdTaxa.html" class="external-link">IdTaxa()</a></code> from the <code>DECIPHER</code> package.</p>
<p><code><a href="https://rdrr.io/pkg/DECIPHER/man/IdTaxa.html" class="external-link">IdTaxa()</a></code> requires a training set, which can be obtained
in two ways. The first is to download them from <a href="http://www2.decipher.codes/Downloads.html" class="external-link">DECIPHER’s downloads
page</a>, which contains prebuilt training sets for a variety of
organisms. We’ll be using an Actinobacteria dataset obtained from this
website.</p>
<p>The other method is to build a training set yourself using
<code><a href="https://rdrr.io/pkg/DECIPHER/man/LearnTaxa.html" class="external-link">LearnTaxa()</a></code>. This will not be covered in this workshop, but
more information is available on the <code>DECIPHER</code> documentation
for people that are interested.</p>
<p>In the interest of time, we’ll just classify the first 10 genes. Note
that our training set for <code><a href="https://rdrr.io/pkg/DECIPHER/man/IdTaxa.html" class="external-link">IdTaxa()</a></code> is trained on amino
acids, so we have to provide it with amino acid sequences. This is why
we</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Find amino acid sequences corresponding to found genes</span></span>
<span><span class="va">geneSeqs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/ExtractGenes.html" class="external-link">ExtractGenes</a></span><span class="op">(</span><span class="va">genes</span>, <span class="va">dnaGenome</span>, type<span class="op">=</span><span class="st">"DNAStringSet"</span><span class="op">)</span></span>
<span><span class="va">geneSeqs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/translate.html" class="external-link">translate</a></span><span class="op">(</span><span class="va">geneSeqs</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># `trainingSet` is loaded from extdata/GeneCalling/IdTaxaTraining</span></span>
<span><span class="co"># It's broken up into pieces because of file constraints on GH</span></span>
<span><span class="co"># You can rebuild it using the "RecombineIdTaxaTraining.R" file</span></span>
<span><span class="co"># (just substitute the appropriate value for the first line),</span></span>
<span><span class="co"># or you can download it directly from the DECIPHER website.</span></span>
<span></span>
<span><span class="co"># Classify!</span></span>
<span><span class="va">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/IdTaxa.html" class="external-link">IdTaxa</a></span><span class="op">(</span><span class="va">geneSeqs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>, <span class="va">trainingSet</span><span class="op">)</span></span></code></pre></div>
<p>Once we’ve finished calculating, we can either view the annotations
directly, or plot them as a taxonomy.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Looking at all results</span></span>
<span><span class="va">ids</span></span></code></pre></div>
<pre><code><span><span class="co">##   A test set of class 'Taxa' with length 10</span></span>
<span><span class="co">##      confidence taxon</span></span>
<span><span class="co">##  [1]       100% Root; 09130 Environmental Information Processing; 09132 Signa...</span></span>
<span><span class="co">##  [2]       100% Root; 09120 Genetic Information Processing; 09124 Replication...</span></span>
<span><span class="co">##  [3]        99% Root; 09120 Genetic Information Processing; 09124 Replication...</span></span>
<span><span class="co">##  [4]        29% Root; unclassified_Root...                                      </span></span>
<span><span class="co">##  [5]        96% Root; 09180 Brite Hierarchies; 09182 Protein families: geneti...</span></span>
<span><span class="co">##  [6]       100% Root; 09180 Brite Hierarchies; 09182 Protein families: geneti...</span></span>
<span><span class="co">##  [7]        35% Root; unclassified_Root...                                      </span></span>
<span><span class="co">##  [8]        90% Root; unclassified_Root...                                      </span></span>
<span><span class="co">##  [9]        40% Root; unclassified_Root...                                      </span></span>
<span><span class="co">## [10]        28% Root; unclassified_Root...</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Looking at a specific entry</span></span>
<span><span class="va">ids</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## $taxon</span></span>
<span><span class="co">## [1] "Root"                                                   </span></span>
<span><span class="co">## [2] "09130 Environmental Information Processing"             </span></span>
<span><span class="co">## [3] "09132 Signal transduction"                              </span></span>
<span><span class="co">## [4] "02020 Two-component system [PATH:ko02020]"              </span></span>
<span><span class="co">## [5] "K02313  dnaA, chromosomal replication initiator protein"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $confidence</span></span>
<span><span class="co">## [1] 100 100 100 100 100</span></span></code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot the distribution of results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">ids</span>, <span class="va">trainingSet</span><span class="op">)</span></span></code></pre></div>
<p><img src="GeneCallingAnnotation_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<style type="text/css">
.pagination {
    display: flex;
    display: -webkit-flex;
    padding-left: 0;
    list-style: none;
    justify-content: center
}
</style>
<div class="center">
<ul class="pagination pagination-lg">
<li class="page-item">
<p><a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">«</a></p>
</li>
<li class="page-item">
<p><a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">1</a></p>
</li>
<li class="page-item">
<p><a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">2</a></p>
</li>
<li class="page-item active">
<p><a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">3</a></p>
</li>
<li class="page-item">
<p><a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">4</a></p>
</li>
<li class="page-item">
<p><a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">5</a></p>
</li>
<li class="page-item">
<p><a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">6</a></p>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Conclusion.html">7</a>
</li>
<li class="page-item">
<p><a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">»</a></p>
</li>
</ul>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
