<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="CompGenomicsBioc2022">
<title>Loading in Genome Data with DECIPHER â€¢ CompGenomicsBioc2022</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Loading in Genome Data with DECIPHER">
<meta property="og:description" content="CompGenomicsBioc2022">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">CompGenomicsBioc2022</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-workshop">Workshop</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-workshop">
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CompGenomicsBioc2022.html">Overview</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Setup</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">Working with Genomic Data</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Gene Calling and Annotation</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">Finding COGs</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">Building Phylogenies</a>
    <a class="dropdown-item" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">Coevolutionary Networks</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.ahl27.com/tutorials" aria-label="Homepage">
    <span class="fa fa-user"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ahl27/CompGenomicsBioc2022" aria-label="GitHub">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Loading in Genome Data with DECIPHER</h1>
                        <h4 data-toc-skip class="author">Aidan
Lakshman<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;a href="mailto:ahl27@pitt.edu" class="email"&gt;ahl27@pitt.edu&lt;/a&gt;&lt;/p&gt;'><sup>1</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ahl27/CompGenomicsBioc2022/blob/HEAD/vignettes/LoadingGenomeData.Rmd" class="external-link"><code>vignettes/LoadingGenomeData.Rmd</code></a></small>
      <div class="d-none name"><code>LoadingGenomeData.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="reading-in-sequencing-data">Reading in Sequencing Data<a class="anchor" aria-label="anchor" href="#reading-in-sequencing-data"></a>
</h2>
<p>In order to work with sequencing data, we first have to get it into R
in a format that allows us to work with it. The most commonly used
format for genomic data is the <code>XString</code> and
<code>XStringSet</code> classes, which are available through the
<code>Biostrings</code> package.</p>
<p><code>XString</code>s come in 4 distinct flavors, depending on the
characters allowed: - <code>DNAString</code>, for DNA data (ATGC, plus
gaps and ambiguity codes) - <code>RNAString</code>, for RNA data (AUGC,
plus gaps and ambiguity codes) - <code>AAString</code>, for amino acid
data (20 amino acids plus ambiguity codes, U, O, and unknown/gaps) -
<code>BString</code>, for any combination of any letters</p>
<p>When <code>XString</code> objects of the same type are combined, they
become an <code>XStringSet</code>. This provides an easy way to store
and work with genomics data. Below is an example of manually creating an
<code>XStringSet</code>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DECIPHER</span><span class="op">)</span> <span class="co"># Auto-imports Biostrings</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sequences</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'AGACT'</span>,
               <span class="st">'AGACG'</span>,
               <span class="st">'TCATT'</span>,
               <span class="st">'TGCTG'</span>,
               <span class="st">'AGCTG'</span><span class="op">)</span>

<span class="va">sequenceSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">DNAStringSet</a></span><span class="op">(</span><span class="va">sequences</span><span class="op">)</span>
<span class="va">sequenceSet</span></code></pre></div>
<pre><code><span class="co">## DNAStringSet object of length 5:</span>
<span class="co">##     width seq</span>
<span class="co">## [1]     5 <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span></span>
<span class="co">## [2]     5 <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## [3]     5 <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">TT</span></span>
<span class="co">## [4]     5 <span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span>
<span class="co">## [5]     5 <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span></span></code></pre>
<p>Manually typing in sequences obviously isnâ€™t a great system. Most
sequencing data are downloadable as <code>.fasta</code> or
<code>.fastq</code> files, so letâ€™s look at a more realistic use-case
that reads in a data from a <code>.fasta</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">exampledata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>, <span class="st">'EX_Strepto_COG.fasta'</span>, 
                            package<span class="op">=</span><span class="st">'CompGenomicsBioc2022'</span><span class="op">)</span>

<span class="co"># note that we can also use format='fastq' for FASTQ datasets</span>
<span class="va">exStringSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readDNAStringSet</a></span><span class="op">(</span><span class="va">exampledata</span>, format<span class="op">=</span><span class="st">"fasta"</span><span class="op">)</span>
<span class="va">exStringSet</span> <span class="op">&lt;-</span> <span class="va">exStringSet</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span></code></pre></div>
<p>Success! Now we have a large example dataset to work with.</p>
<p>If we wanted to visualize these sequences, we can open them in a web
browser using <code><a href="https://rdrr.io/pkg/DECIPHER/man/BrowseSeqs.html" class="external-link">BrowseSeqs()</a></code> from <code>DECIPHER</code>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/BrowseSeqs.html" class="external-link">BrowseSeqs</a></span><span class="op">(</span><span class="va">exStringSet</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="aligning-sequences">Aligning Sequences<a class="anchor" aria-label="anchor" href="#aligning-sequences"></a>
</h2>
<p>Now that we have some sequences, letâ€™s explore some of the ways we
can manipulate them. A complete demo of <code>Biostrings</code> is
outside the scope of this workshop, so weâ€™ll just focus on functionality
added via <code>DECIPHER</code>. A common analysis in bioinformatics is
aligning sequences. This is easily achievable with either the
<code><a href="https://rdrr.io/pkg/DECIPHER/man/AlignSeqs.html" class="external-link">AlignSeqs()</a></code> function or the <code><a href="https://rdrr.io/pkg/DECIPHER/man/AlignTranslation.html" class="external-link">AlignTranslation()</a></code>
functions from <code>DECIPHER</code>. <code><a href="https://rdrr.io/pkg/DECIPHER/man/AlignTranslation.html" class="external-link">AlignTranslation()</a></code>
aligns sequences based on their translated amino acid sequences, and is
significantly more accurate for DNA or RNA sequences. This code block
will also use <code><a href="https://rdrr.io/pkg/DECIPHER/man/RemoveGaps.html" class="external-link">RemoveGaps()</a></code> from <code>DECIPHER</code> to
remove gaps that are shared by all sequences, in an effort to make the R
printout a little more informative.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ali1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/AlignSeqs.html" class="external-link">AlignSeqs</a></span><span class="op">(</span><span class="va">exStringSet</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/RemoveGaps.html" class="external-link">RemoveGaps</a></span><span class="op">(</span><span class="va">ali1</span>, removeGaps<span class="op">=</span><span class="st">'common'</span><span class="op">)</span> <span class="co"># Better visualization</span></code></pre></div>
<pre><code><span class="co">## DNAStringSet object of length 50:</span>
<span class="co">##      width seq                                              names               </span>
<span class="co">##  [1]  1089 -----------------------...---------------------- 55</span>
<span class="co">##  [2]  1089 <span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">CCC</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">CC</span><span style="color: #000000; background-color: #FFAFAF;">AA</span><span style="color: #000000; background-color: #FFD7AF;">T</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #FFAFAF;">A</span><span style="color: #000000; background-color: #AFFFAF;">C</span><span style="color: #000000; background-color: #AFFFFF;">G</span>--...---------------------- 12664</span>
<span class="co">##  [3]  1089 -----------------------...---------------------- 83844</span>
<span class="co">##  [4]  1089 -----------------------...---------------------- 99294</span>
<span class="co">##  [5]  1089 -----------------------...---------------------- 106117</span>
<span class="co">##  ...   ... ...</span>
<span class="co">## [46]  1089 -----------------------...---------------------- 1077521</span>
<span class="co">## [47]  1089 -----------------------...---------------------- 1092227</span>
<span class="co">## [48]  1089 -----------------------...---------------------- 1099095</span>
<span class="co">## [49]  1089 -----------------------...---------------------- 1101364</span>
<span class="co">## [50]  1089 -----------------------...---------------------- 1114177</span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ali2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/AlignTranslation.html" class="external-link">AlignTranslation</a></span><span class="op">(</span><span class="va">exStringSet</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/RemoveGaps.html" class="external-link">RemoveGaps</a></span><span class="op">(</span><span class="va">ali2</span>, removeGaps<span class="op">=</span><span class="st">'common'</span><span class="op">)</span> <span class="co"># Better visualization</span></code></pre></div>
<pre><code><span class="co">## DNAStringSet object of length 50:</span>
<span class="co">##      width seq                                              names               </span>
<span class="co">##  [1]  1224 -----------------------...---------------------- 55</span>
<span class="co">##  [2]  1224 -----------------------...---------------------- 12664</span>
<span class="co">##  [3]  1224 -----------------------...---------------------- 83844</span>
<span class="co">##  [4]  1224 -----------------------...---------------------- 99294</span>
<span class="co">##  [5]  1224 -----------------------...---------------------- 106117</span>
<span class="co">##  ...   ... ...</span>
<span class="co">## [46]  1224 -----------------------...---------------------- 1077521</span>
<span class="co">## [47]  1224 -----------------------...---------------------- 1092227</span>
<span class="co">## [48]  1224 -----------------------...---------------------- 1099095</span>
<span class="co">## [49]  1224 -----------------------...---------------------- 1101364</span>
<span class="co">## [50]  1224 -----------------------...---------------------- 1114177</span></code></pre>
<p>Alignments tend to start with a lot of gaps, and as a result viewing
them through R isnâ€™t always super informative. As before, we can
visualize this alignment in a much better way using
<code><a href="https://rdrr.io/pkg/DECIPHER/man/BrowseSeqs.html" class="external-link">BrowseSeqs()</a></code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/BrowseSeqs.html" class="external-link">BrowseSeqs</a></span><span class="op">(</span><span class="va">ali2</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="alignment-free-analysis">Alignment-Free Analysis<a class="anchor" aria-label="anchor" href="#alignment-free-analysis"></a>
</h2>
<p>Another common analysis is to look at syntenic hits between sequences
without aligning them. The result allows us to easily visualize
differences between two sequences without having to align them. Letâ€™s
look at an example of finding syntenic hits between the first 6
sequences in our set.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">to_select</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">6</span>,<span class="fl">8</span>,<span class="fl">9</span><span class="op">)</span>
<span class="va">synData</span> <span class="op">&lt;-</span> <span class="va">exStringSet</span><span class="op">[</span><span class="va">to_select</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">synData</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">to_select</span><span class="op">)</span>
<span class="va">dbConn</span> <span class="op">&lt;-</span> <span class="fu">dbConnect</span><span class="op">(</span><span class="fu"><a href="https://rsqlite.r-dbi.org/reference/SQLite.html" class="external-link">SQLite</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">":memory:"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/Seqs2DB.html" class="external-link">Seqs2DB</a></span><span class="op">(</span><span class="va">synData</span>, <span class="st">"XStringSet"</span>, <span class="va">dbConn</span>, identifier<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">to_select</span><span class="op">)</span><span class="op">)</span>

<span class="va">syn</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DECIPHER/man/FindSynteny.html" class="external-link">FindSynteny</a></span><span class="op">(</span><span class="va">dbConn</span>, verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="fu">dbDisconnect</span><span class="op">(</span><span class="va">dbConn</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/pairs.html" class="external-link">pairs</a></span><span class="op">(</span><span class="va">syn</span><span class="op">)</span></code></pre></div>
<p><img src="LoadingGenomeData_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>What are we looking at with this plot? Each plot compares two
genomes, and each dot is a syntenic hit between the two genomes. The X
position of the dot is its location on the first genome, and the Y
position is its position on the second genome. If we had identical
genomes, we would expect to see a diagonal line <span class="math inline">\(y=x\)</span>, indicating that all bases occur at
the same place in both genomes. Gaps indicate areas that do not match,
and points off the diagonal indicate matching areas in different places
on each genomes.</p>
<p>You may also notice that this example creates a SQLite table prior to
running <code><a href="https://rdrr.io/pkg/DECIPHER/man/FindSynteny.html" class="external-link">FindSynteny()</a></code>. <code>DECIPHER</code> is built
around working with sequences via SQLite tables to allow for analysis at
scale. When working with hundreds of thousands of genomes, we canâ€™t
always just store the files within memory on our computer. This wonâ€™t be
covered in this tutorial, but advanced users can refer to the
<code>DECIPHER</code> docs to find out more about using database files.
Iâ€™ve also included more information on these in the Function Reference
page for this section.</p>
<style type="text/css">
.pagination {
    display: flex;
    display: -webkit-flex;
    padding-left: 0;
    list-style: none;
    justify-content: center
}
</style>
<div class="center">
<ul class="pagination pagination-lg">
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">Â«</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/Setup.html">1</a>
</li>
<li class="page-item active">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/LoadingGenomeData.html">2</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">3</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/FindingCOGs.html">4</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/ConstructingPhylogenies.html">5</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/CoevolutionNetworks.html">6</a>
</li>
<li class="page-item">
<a class="page-link" href="https://www.ahl27.com/CompGenomicsBioc2022/articles/GeneCallingAnnotation.html">Â»</a>
</li>
</ul>
</div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Aidan Lakshman.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
